---
title: "08-ASXL1_Redo_all"
author: "Maggie"
date: "11/16/2021"
output: html_document
---

# Setup

```{r setup, include=FALSE}
# Work directory
# setwd("/Projects/ASXL1/")
setwd("/mnt/scratch/KoborLab/Personal_Folders/mfu/Projects/ASXL1")
home <- "/mnt/scratch/KoborLab/Personal_Folders/mfu/"
.libPaths(c(.libPaths(),"/mnt/scratch/KoborLab/R_Libs/4.2"))

# CRAN packages
library(reshape2)
library(plyr)
library(dplyr)
library(ggplot2)
library(GEOmetadb)
library(limma)
library(methylclock)
library(minfi)
library(methylumi)
library(lumi)
library(wateRmelon)
library(nsprcomp)
library(impute)
library(sva)
library(doParallel)
library(whatsex)
library(EpiDISH)
library(pheatmap)
library(Metrics)
library(gridExtra)
library(pbapply)
source(paste0(home, "Functions/locfdr_LMM.R"))
source(paste0(home, "Functions/BetaPlot.R"))
source(paste0(home, "Functions/ProbeFilter.R"))
source(paste0(home, "Functions/DetPvalTest.R"))
source(paste0(home, "Functions/BloodPlot.R"))
source(paste0(home, "Functions/heat_scree_plot.R"))
source(paste0(home, "Functions/Permutation_Source_Code.R"))
```

```{r}
BetaPlot <- function(bt, mt, var){
    if(nrow(bt) > 20000){
        n <- 20000
    } else {
        n <- nrow(bt)
    }
    bt.m <- reshape2::melt(bt[sample(1:nrow(bt),n),])
    #remove NAs before plotting (otherwise get many non-inifnite warnings)
    plt <- bt.m[which(!(is.na(bt.m$value))),]
        
    # Add meta
    colnames(plt) <- c("CpG","ID","Beta")
    plt <- merge(plt, mt, by.x = "ID", by.y = 0) %>% 
        as.data.frame()
        
    ggplot(plt, aes(Beta, group = ID, color = get(var))) +
        geom_density() +
        theme_bw() +
        labs(x = "DNAm Beta Value")
}

BloodPlot <- function(mt, varname = NULL, vartype = c("categorical", "continuous"), 
                      tissue = c("blood", "bloodextended", "cord"), ct = NULL) {
    
    require(dplyr)
    require(ggpubr)
    require(gridExtra)
    require(data.table)
    
    if (!is.null(ct)) {
        ct <- ct
    } else if (tissue == "blood"){
        ct <- c("CD8T", "CD4T", "NK", "Bcell", "Mono", "Gran")
    } else if (tissue == "bloodextended"){
        ct <- c("Bas", "Bmem", "Bnv", "CD4mem", "CD4nv",
                "CD8mem", "CD8nv", "Eos", "Mono", "Neu", "NK", "Treg")
    } else if (tissue == "cord"){
        ct <- c("CD8T", "CD4T", "NK", "Bcell", "Mono", "Gran", "nRBC")
    }
    mt$Sample_Name <- rownames(mt)
    counts <- mt[, c("Sample_Name", ct)] %>% as.data.table()
    counts_plot <- melt(counts, id.vars = "Sample_Name")
    covar <- mt[, c("Sample_Name", varname)]
    plt <- merge(counts_plot, covar, by = "Sample_Name") %>% 
        as.data.frame()
    
    p1 <- ggplot(plt, aes(get(varname), value, fill = get(varname))) +
        geom_boxplot(alpha = 0.2) +
        geom_point(shape = 21, aes(group = Sample_Name), 
                   position = position_jitter(w = 0.1)) +
        xlab(NULL) +
        ylab("Cell Proportion") +
        theme_bw() + 
        theme(axis.text = element_text(size = 12, color = "black"),
              axis.title = element_text(size = 14),
              legend.text = element_text(size = 11),
              legend.title = element_text(size = 14), 
              strip.text = element_text(size = 14),
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank()) +
        facet_wrap(~ variable)
    
    # if (vartype == "categorical") {
    #     p1 <- p1 + stat_compare_means(method = "t.test")
    # } 
    plot(p1)
}
```


# Blood Reference

## Get reference dataset

Given that we previously found that the horvath clock 

```{r cars}
getSQLiteFile(destdir = "~/KoborLab/kobor_space/mfu/GEO")
con <- dbConnect(SQLite(), "~/KoborLab/kobor_space/mfu/GEO/GEOmetadb.sqlite")
dbListFields(con, "gsm")

x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL13534' and  organism_ch1 = 'Homo sapiens'"))
sampl <- x
x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL21145' and  organism_ch1 = 'Homo sapiens'"))
sampl2 <- x
x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL16304' and  organism_ch1 = 'Homo sapiens'"))
sampl3 <- x
x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL23976' and  organism_ch1 = 'Homo sapiens'"))
sampl4 <- x

allsamp <- rbind(sampl, sampl2)
allsamp2 <- rbind(sampl3, sampl4)
allsamp <- rbind(allsamp, allsamp2)

bldID <- c(grep("[Bb]lood|[Pp][Bb]|[Ll]euko|[Pp]eripheral|[Cc]ord|[Bb]uffy", allsamp$characteristics_ch1), 
           grep("[Bb]lood|[Pp][Bb]|[Ll]euko|[Pp]eripheral|[Cc]ord|[Bb]uffy", allsamp$source_name_ch1),
           grep("[Bb]lood|[Pp][Bb]|[Ll]euko|[Pp]eripheral|[Cc]ord|[Bb]uffy", allsamp$title),
           grep("[Bb]lood|[Pp][Bb]|[Ll]euko|[Pp]eripheral|[Cc]ord|[Bb]uffy", allsamp$description))
bldID <- unique(bldID)
bldsamp <- allsamp[bldID, ]
anno <- grep("[Aa]ge", bldsamp$characteristics_ch1, value = T)
annotable <- strsplit2(anno, ";")
anno.f <- sapply(annotable, function(x) {
    x[grep("[Aa]ge", x)]
}) %>% unlist(.)

bldGEO <- table(bldsamp$series_id) %>% sort(decreasing = T) %>% names()
bldGEO <- sapply(bldGEO, function(x) {
    y <- strsplit(x, ",") %>% unlist()
    return(y[1])
}) %>% as.vector()
# bldGEO <- strsplit(paste0(bldGEO, collapse = ","), ",")[[1]]
# bldped <- bldsamp[grep("age: 1[012]|age at sample collection \\(years\\): 1[012]|age \\(y\\): 1[012]|age \\(years\\): 1[012]", bldsamp$characteristics_ch1), ]
```

## Download all samples

```{r}
Sys.setenv("VROOM_CONNECTION_SIZE" = 13107200)
getOption("download.file.method.GEOquery")
options(download.file.method.GEOquery = "libcurl")
dest <- "~/KoborLab/kobor_space/mfu/GEO/Blood"

bldGEO <- bldGEO[!bldGEO %in% c("GSE168406", "GSE64950", "GSE93992")]
BLD1 <- sapply(bldGEO[1:20], function (x) getGEO(x, destdir = dest))
BLD2 <- sapply(bldGEO[21:50], function (x) getGEO(x, destdir = dest))
BLD3 <- sapply(bldGEO[51:100], function (x) getGEO(x, destdir = dest))
BLD4 <- sapply(bldGEO[101:150], function (x) getGEO(x, destdir = dest))
BLD5 <- sapply(bldGEO[151:200], function (x) getGEO(x, destdir = dest))
BLD6 <- sapply(bldGEO[201:300], function (x) getGEO(x, destdir = dest))
BLD7 <- sapply(bldGEO[301:400], function (x) getGEO(x, destdir = dest))
BLD8 <- sapply(bldGEO[401:500], function (x) getGEO(x, destdir = dest))

save(BLD1, BLD2, file = "~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD1_2.RData")
save(BLD3, BLD4, file = "~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD3_4.RData")
save(BLD5, BLD6, file = "~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD5_6.RData")
save(BLD7, BLD8, file = "~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD7_8.RData")

DNAmage1 <- lapply(BLD1, function(x) {
    out <- lapply(x, function(y){
        bt <- exprs(y)
        if(nrow(bt) > 0) {
            bt <- cbind(rownames(bt), bt)
            out <- DNAmAge(bt, clocks = c("Horvath", "Hannum"))
        } else {
            out <- NULL
        }
        return(out)
    })
    return(out)
})
names(DNAmage) <- names(BLD)
beta <- cbind(rownames(beta), beta)
DNAmage <- DNAmAge(beta, clocks = c("Horvath", "Hannum"))
meta <- meta[, colnames(meta) != "Cell_Types"] %>% 
    cbind(., DNAmage[, c("Horvath", "Hannum")])
DNAmage <- meta[, c("Horvath", "Hannum", "Age", "Age_Group2", "Cohort")]
```


```{r}
load("~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD1_2.RData")

### "GSE55763" - 23.7-75
# Samp <- BLD1$GSE55763$GSE55763_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE55763"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE55763.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE55763.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE55763", baseDir = bsdir)
# GSE55763 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE55763/", recursive = T, extended = T)

### "GSE169156" - no one under 18, cancer survivors
# Samp <- BLD1$GSE169156$GSE169156_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE169156"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE169156.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE169156.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE169156", baseDir = bsdir)
# GSE169156 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE169156/", recursive = T, extended = T)

### "GSE145361" - Parkinson's    
# Samp <- BLD1$GSE145361$GSE145361_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE145361"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE145361.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE145361.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE145361", baseDir = bsdir)
# GSE145361 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE145361/", recursive = T, extended = T)

### "GSE157131" - Blood pressure middle age cohort
# Samp <- BLD1$GSE157131$GSE157131_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE157131"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE157131.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE157131.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE157131", baseDir = bsdir)
# GSE157131 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE157131/", recursive = T, extended = T)

### "GSE56046,GSE56047" - age 44 - 83  
# Samp <- BLD1$GSE56046$GSE56046_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE56046"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE56046.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE56046.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE56046", baseDir = bsdir)
# GSE56046 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE56046/", recursive = T, extended = T)

### "GSE147740" - age 26 - 59
# Samp <- BLD1$GSE147740$GSE147740_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE147740"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE147740.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE147740.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE147740", baseDir = bsdir)
# GSE147740 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE147740/", recursive = T, extended = T)

### "GSE157896,GSE158064" - GUSTO maternal sample age 19 - 47
# Samp <- BLD1$GSE157896$GSE157896_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE157896"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE157896.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE157896.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE157896", baseDir = bsdir)
# GSE157896 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE157896/", recursive = T, extended = T)

### "GSE152026" - age 18 - 64
# Samp <- BLD1$GSE152026$GSE152026_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE152026"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE152026.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE152026.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE152026", baseDir = bsdir)
# GSE152026 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE152026/", recursive = T, extended = T)

### "GSE158063,GSE158064" - GUSTO fetal cord tissue
# Samp <- BLD1$GSE158063$GSE158063_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE158063"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE158063.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE158063.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE158063", baseDir = bsdir)
# GSE158063 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE158063/", recursive = T, extended = T)

### "GSE84727" - age 18 - 81
# Samp <- BLD1$GSE84727$GSE84727_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE84727"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "B-Cells"
# pd$selection_markers <- "CD19+"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE84727.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "selection_markers", "disease_status")]
# colnames(GSE84727.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Selection_Markers","Disease_Status")
# getGEOSuppFiles("GSE84727", baseDir = bsdir)
# GSE84727 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE84727/", recursive = T, extended = T)

### "GSE132203" - age 18 - 76
# Samp <- BLD1$GSE132203$GSE132203_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE132203"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE132203.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE132203.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE132203", baseDir = bsdir)
# GSE132203 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE132203/", recursive = T, extended = T)

### "GSE153712" - Alzheimer's cohort
# Samp <- BLD1$GSE153712$GSE153712_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE153712"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE153712.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE153712.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE153712", baseDir = bsdir)
# GSE153712 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE153712/", recursive = T, extended = T)

### "GSE147221" - age 17 - 71
# Samp <- BLD1$GSE147221$GSE147221_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE147221"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE147221.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE147221.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE147221", baseDir = bsdir)
# GSE147221 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE147221/", recursive = T, extended = T)

### "GSE125105" - age 17 - 87
# Samp <- BLD1$GSE125105$GSE125105_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE125105"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE125105.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE125105.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE125105", baseDir = bsdir)
# GSE125105 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE125105/", recursive = T, extended = T)

### "GSE43414" - age 65 - 99 (brain + blood)
# Samp <- BLD1$GSE43414$GSE43414_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE43414"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE43414.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE43414.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE43414", baseDir = bsdir)
# GSE43414 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE43414/", recursive = T, extended = T)

### "GSE42861" - age 18 - 70
# Samp <- BLD1$GSE42861$GSE42861_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE42861"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE42861.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE42861.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE42861", baseDir = bsdir)
# GSE42861 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE42861/", recursive = T, extended = T)

### "GSE116339" - age 23 - 88.46
# Samp <- BLD2$GSE116339$GSE116339_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE116339"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE116339.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE116339.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE116339", baseDir = bsdir)
# GSE116339 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE116339/", recursive = T, extended = T)

### "GSE80417" - age 18 - 891???
# Samp <- BLD2$GSE80417$GSE80417_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE80417"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE80417.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE80417.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE80417", baseDir = bsdir)
# GSE80417 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE80417/", recursive = T, extended = T)

### "GSE40279" - age 19 - 101
# Samp <- BLD2$GSE40279$GSE40279_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE40279"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE40279.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE40279.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE40279", baseDir = bsdir)
# GSE40279 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE40279/", recursive = T, extended = T)     

### "GSE117859,GSE117861" - age 25 - 75
# Samp <- BLD2$GSE117859$GSE117859_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE117859"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE117859.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE117859.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE117859", baseDir = bsdir)
# GSE117859 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE117859/", recursive = T, extended = T)

### "GSE69138" - stroke cohort (average 70 yr)
# Samp <- BLD2$GSE69138$GSE69138_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE69138"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE69138.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE69138.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE69138", baseDir = bsdir)
# GSE69138 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE69138/", recursive = T, extended = T)

### "GSE111629" - age 35 - 92
# Samp <- BLD2$GSE111629$GSE111629_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE111629"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE111629.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE111629.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE111629", baseDir = bsdir)
# GSE111629 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE111629/", recursive = T, extended = T)

### "GSE141065" - cord blood
# Samp <- BLD2$GSE141065$GSE141065_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE141065"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE141065.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE141065.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE141065", baseDir = bsdir)
# GSE141065 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE141065/", recursive = T, extended = T)

### "GSE128235" - age 18 - 87
# Samp <- BLD2$GSE128235$GSE128235_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE128235"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE128235.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE128235.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE128235", baseDir = bsdir)
# GSE128235 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE128235/", recursive = T, extended = T)

### "GSE59685" - age 65 - 99            
# Samp <- BLD2$GSE59685$GSE59685_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE59685"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE59685.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE59685.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE59685", baseDir = bsdir)
# GSE59685 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE59685/", recursive = T, extended = T)

### "GSE117860,GSE117861" - age 25 - 75
# Samp <- BLD2$GSE117860$GSE117860_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE117860"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE117860.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE117860.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE117860", baseDir = bsdir)
# GSE117860 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE117860/", recursive = T, extended = T)

### "GSE51032" - age 34 - 72
# Samp <- BLD2$GSE51032$GSE51032_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE51032"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE51032.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE51032.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE51032", baseDir = bsdir)
# GSE51032 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE51032/", recursive = T, extended = T)

### "GSE72774" - age 35 - 91
# Samp <- BLD2$GSE72774$GSE72774_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72774"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72774.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72774.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72774", baseDir = bsdir)
# GSE72774 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72774/", recursive = T, extended = T)

### "GSE163970" - paget’s disease cohort (average 70yr)
# Samp <- BLD2$GSE163970$GSE163970_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE163970"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE163970.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE163970.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE163970", baseDir = bsdir)
# GSE163970 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE163970/", recursive = T, extended = T)

### "GSE151042" - cord blood
# Samp <- BLD2$GSE151042$GSE151042_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE151042"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE151042.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE151042.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE151042", baseDir = bsdir)
# GSE151042 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE151042/", recursive = T, extended = T)

### "GSE121633" - 19 - 81
# Samp <- BLD2$GSE121633$GSE121633_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE121633"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE121633.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE121633.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE121633", baseDir = bsdir)
# GSE121633 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE121633/", recursive = T, extended = T)

### "GSE100227" - age 40–78
# Samp <- BLD2$GSE100227$GSE100227_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE100227"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE100227.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE100227.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE100227", baseDir = bsdir)
# GSE100227 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE100227/", recursive = T, extended = T)

### "GSE115278" - type II diabetes cohort (average 47.0 yr)
# Samp <- BLD2$GSE115278$`GSE115278-GPL21145_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE115278"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE115278.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE115278.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE115278", baseDir = bsdir)
# GSE115278 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE115278/", recursive = T, extended = T)

### "GSE154566" - age 19 - 66
# Samp <- BLD2$GSE154566$GSE154566_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE154566"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE154566.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE154566.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE154566", baseDir = bsdir)
# GSE154566 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE154566/", recursive = T, extended = T)

### "GSE50660" - age 38 - 67  
# Samp <- BLD2$GSE50660$GSE50660_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE50660"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE50660.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE50660.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE50660", baseDir = bsdir)
# GSE50660 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE50660/", recursive = T, extended = T)

### "GSE72680" - age 18 - 77
# Samp <- BLD2$GSE72680$GSE72680_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72680"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72680.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72680.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72680", baseDir = bsdir)
# GSE72680 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72680/", recursive = T, extended = T)

### "GSE168739" - age 19 - 61
# Samp <- BLD2$GSE168739$GSE168739_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE168739"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE168739.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE168739.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE168739", baseDir = bsdir)
# GSE168739 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE168739/", recursive = T, extended = T)

### "GSE107080,GSE107082" - age 25  - 75
# Samp <- BLD2$GSE107080$GSE107080_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE107080"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE107080.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE107080.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE107080", baseDir = bsdir)
# GSE107080 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE107080/", recursive = T, extended = T)
```

```{r}
load("~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD3_4.RData")

### "GSE100264,GSE107082" - HIV cohort age 	51.38 ± 4.80
# Samp <- BLD3$GSE100264$GSE100264_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE100264"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE100264.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE100264.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE100264", baseDir = bsdir)
# GSE100264 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE100264/", recursive = T, extended = T)

### "GSE105109" - Alzheimer's disease cohort , age 58 - 99
# Samp <- BLD3$GSE105109$GSE105109_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE105109"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE105109.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE105109.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE105109", baseDir = bsdir)
# GSE105109 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE105109/", recursive = T, extended = T)

### "GSE53740" - Progressive supranuclear palsy cohort
# Samp <- BLD3$GSE53740$GSE53740_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE53740"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE53740.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE53740.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE53740", baseDir = bsdir)
# GSE53740 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE53740/", recursive = T, extended = T)

### "GSE97628" - CHAMACOS cord blood
# Samp <- BLD3$GSE97628$GSE97628_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE97628"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE97628.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE97628.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE97628", baseDir = bsdir)
# GSE97628 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE97628/", recursive = T, extended = T)

### "GSE77696" - HIV cohort age 25 - 76
# Samp <- BLD3$GSE77696$GSE77696_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE77696"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE77696.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE77696.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE77696", baseDir = bsdir)
# GSE77696 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE77696/", recursive = T, extended = T)

### "GSE131989" - age 18 - 82
# Samp <- BLD3$GSE131989$GSE131989_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE131989"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE131989.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE131989.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE131989", baseDir = bsdir)
# GSE131989 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE131989/", recursive = T, extended = T)

### "GSE60275" - XCI cohort, age 30 - 75
# Samp <- BLD3$GSE60275$GSE60275_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE60275"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE60275.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE60275.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE60275", baseDir = bsdir)
# GSE60275 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE60275/", recursive = T, extended = T)

### "GSE72775" - age 37 - 91
# Samp <- BLD3$GSE72775$GSE72775_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72775"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72775.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72775.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72775", baseDir = bsdir)
# GSE72775 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72775/", recursive = T, extended = T)

### "GSE51032,GSE51057" - age 34 - 72
# Samp <- BLD3$GSE51032$GSE51032_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE51032"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE51032.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE51032.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE51032", baseDir = bsdir)
# GSE51032 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE51032/", recursive = T, extended = T)

### "GSE61496"- age 30 - 74
# Samp <- BLD3$GSE61496$GSE61496_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE61496"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE61496.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE61496.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE61496", baseDir = bsdir)
# GSE61496 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE61496/", recursive = T, extended = T)

### "GSE72773" - age 35 - 92
# Samp <- BLD3$GSE72773$GSE72773_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72773"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72773.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72773.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72773", baseDir = bsdir)
# GSE72773 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72773/", recursive = T, extended = T)

### "GSE54882" - RA cohort 
# Samp <- BLD3$GSE54882$GSE54882_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE54882"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE54882.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE54882.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE54882", baseDir = bsdir)
# GSE54882 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE54882/", recursive = T, extended = T)

### "GSE144858" - Alzheimer's disease cohort (age 52 - 90)
# Samp <- BLD3$GSE144858$GSE144858_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE144858"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE144858.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE144858.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE144858", baseDir = bsdir)
# GSE144858 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE144858/", recursive = T, extended = T)

### "GSE106648" - age 16 - 66
# Samp <- BLD3$GSE106648$GSE106648_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE106648"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE106648.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE106648.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE106648", baseDir = bsdir)
# GSE106648 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE106648/", recursive = T, extended = T)

### "GSE174422" - age 37 - 75
# Samp <- BLD3$GSE174422$GSE174422_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE174422"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE174422.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE174422.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE174422", baseDir = bsdir)
# GSE174422 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE174422/", recursive = T, extended = T)

### "GSE85210" - smoking cohort, age 19 - 63
# Samp <- BLD3$GSE85210$GSE85210_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE85210"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE85210.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE85210.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE85210", baseDir = bsdir)
# GSE85210 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE85210/", recursive = T, extended = T)

### "GSE53816,GSE54446" - cord blood
# Samp <- BLD3$GSE53816$GSE53816_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE53816"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE53816.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE53816.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE53816", baseDir = bsdir)
# GSE53816 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE53816/", recursive = T, extended = T)

### "GSE156994" - age 41 - 85  
# Samp <- BLD3$GSE156994$GSE156994_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE156994"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE156994.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE156994.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE156994", baseDir = bsdir)
# GSE156994 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE156994/", recursive = T, extended = T)

### "GSE56047,GSE56581" - age 44 - 83
# Samp <- BLD3$GSE56047$`GSE56047-GPL13534_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE56047"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE56047.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE56047.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE56047", baseDir = bsdir)
# GSE56047 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE100264/", recursive = T, extended = T)

### "GSE104942" - Breast cancer cohort
# Samp <- BLD3$GSE104942$GSE104942_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE104942"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE104942.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE104942.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE104942", baseDir = bsdir)
# GSE104942 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE104942/", recursive = T, extended = T)

### "GSE113725" - age 19 - 61
# Samp <- BLD3$GSE113725$GSE113725_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE113725"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE113725.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE113725.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE113725", baseDir = bsdir)
# GSE113725 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE113725/", recursive = T, extended = T)

### "GSE49031,GSE49032" - B-ALL cohort, bone marrow
# Samp <- BLD3$GSE49031$GSE49031_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE49031"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE49031.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE49031.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE49031", baseDir = bsdir)
# GSE49031 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE49031/", recursive = T, extended = T)

### "GSE61151" - age 35 - 83
# Samp <- BLD3$GSE61151$GSE61151_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE61151"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE61151.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE61151.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE61151", baseDir = bsdir)
# GSE61151 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE61151/", recursive = T, extended = T)

### "GSE67705" - age 25 - 67
# Samp <- BLD3$GSE67705$GSE67705_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE67705"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE67705.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE67705.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE67705", baseDir = bsdir)
# GSE67705 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE67705/", recursive = T, extended = T)

### "GSE69270" - age 40 - 49
# Samp <- BLD3$GSE69270$GSE69270_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE69270"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE69270.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE69270.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE69270", baseDir = bsdir)
# GSE69270 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE69270/", recursive = T, extended = T)

### "GSE122408" - pregnancy cohort
# Samp <- BLD3$GSE122408$GSE122408_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE122408"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE122408.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE122408.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE122408", baseDir = bsdir)
# GSE122408 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE122408/", recursive = T, extended = T)

### "GSE73115" - age 73 - 82
# Samp <- BLD3$GSE73115$GSE73115_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE73115"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE73115.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE73115.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE73115", baseDir = bsdir)
# GSE73115 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE73115/", recursive = T, extended = T)

### "GSE68379" - cell lines
# Samp <- BLD3$GSE68379$GSE68379_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE68379"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE68379.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE68379.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE68379", baseDir = bsdir)
# GSE68379 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE68379/", recursive = T, extended = T)

### "GSE74548" - age 65 - 75
# Samp <- BLD3$GSE74548$GSE74548_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE74548"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE74548.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE74548.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE74548", baseDir = bsdir)
# GSE74548 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE74548/", recursive = T, extended = T)

### "GSE102468" - PD cohort
# Samp <- BLD3$GSE102468$GSE102468_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE102468"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE102468.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE102468.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE102468", baseDir = bsdir)
# GSE102468 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE102468/", recursive = T, extended = T)

### "GSE122086" - cord blood
# Samp <- BLD3$GSE122086$GSE122086_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE122086"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE122086.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE122086.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE122086", baseDir = bsdir)
# GSE122086 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE122086/", recursive = T, extended = T)

### "GSE89218" - PTSD veteran cohort
# Samp <- BLD3$GSE89218$GSE89218_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE89218"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE89218.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE89218.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE89218", baseDir = bsdir)
# GSE89218 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE89218/", recursive = T, extended = T)

### "GSE74414" - age 21 - 71
# Samp <- BLD3$GSE74414$GSE74414_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE74414"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE74414.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE74414.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE74414", baseDir = bsdir)
# GSE74414 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE74414/", recursive = T, extended = T)

### "GSE116379,GSE116380" - schizophrenia cohort, age 45 - 51
# Samp <- BLD3$GSE116379$GSE116379_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE116379"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE116379.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE116379.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE116379", baseDir = bsdir)
# GSE116379 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE116379/", recursive = T, extended = T)

### "GSE69176" - cord blood
# Samp <- BLD3$GSE69176$GSE69176_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE69176"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE69176.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE69176.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE69176", baseDir = bsdir)
# GSE69176 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE69176/", recursive = T, extended = T)

### "GSE174201,GSE174202" - not blood
# Samp <- BLD3$GSE174201$GSE174201_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE174201"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE174201.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE174201.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE174201", baseDir = bsdir)
# GSE174201 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE174201/", recursive = T, extended = T)

# "GSE103541" - sorted cells
# Samp <- BLD3$GSE103541$GSE103541_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE103541"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE103541.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE103541.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE103541", baseDir = bsdir)
# GSE103541 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE103541/", recursive = T, extended = T)

# "GSE140038" - age 36 - 77
# Samp <- BLD3$GSE140038$GSE140038_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE140038"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE140038.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE140038.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE140038", baseDir = bsdir)
# GSE140038 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE140038/", recursive = T, extended = T)

# "GSE67530" - age 22 - 93
# Samp <- BLD4$GSE67530$GSE67530_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE67530"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE67530.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE67530.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE67530", baseDir = bsdir)
# GSE67530 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE67530/", recursive = T, extended = T)

# "GSE164056" - age 19 - 50
# Samp <- BLD4$GSE164056$GSE164056_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE164056"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE164056.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE164056.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE164056", baseDir = bsdir)
# GSE164056 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE164056/", recursive = T, extended = T)

# "GSE58888" - nonagenarian cohort
# Samp <- BLD4$GSE58888$GSE58888_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE58888"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE58888.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE58888.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE58888", baseDir = bsdir)
# GSE58888 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE58888/", recursive = T, extended = T)

# "GSE71955,GSE71957" - age 26 - 87
# Samp <- BLD4$GSE71955$GSE71955_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE71955"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE71955.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE71955.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE71955", baseDir = bsdir)
# GSE71955 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE71955/", recursive = T, extended = T)

# "GSE157252" - age 19 - 58
# Samp <- BLD4$GSE157252$GSE157252_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE157252"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE157252.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE157252.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE157252", baseDir = bsdir)
# GSE157252 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE157252/", recursive = T, extended = T)

# "GSE88929" - cord blood
# Samp <- BLD4$GSE88929$GSE88929_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE88929"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE88929.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE88929.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE88929", baseDir = bsdir)
# GSE88929 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE88929/", recursive = T, extended = T)

# "GSE59457" - age 26 - 68
# Samp <- BLD4$GSE59457$GSE59457_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE59457"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE59457.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE59457.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE59457", baseDir = bsdir)
# GSE59457 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE59457/", recursive = T, extended = T)

# "GSE174818" - age 21 - 89
# Samp <- BLD4$GSE174818$GSE174818_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE174818"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE174818.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE174818.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE174818", baseDir = bsdir)
# GSE174818 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE174818/", recursive = T, extended = T)

# "GSE107459,GSE107460" - age 18 - 36
# Samp <- BLD4$GSE107459$GSE107459_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE107459"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE107459.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE107459.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE107459", baseDir = bsdir)
# GSE107459 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE107459/", recursive = T, extended = T)

# "GSE68194" - nonagenarians cohort
# Samp <- BLD4$GSE68194$GSE68194_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE68194"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE68194.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE68194.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE68194", baseDir = bsdir)
# GSE68194 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE68194/", recursive = T, extended = T)

# "GSE144894,GSE144901" - circulating chronic lymphocytic leukemia (CLL)
# Samp <- BLD4$GSE144894$GSE144894_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE144894"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE144894.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE144894.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE144894", baseDir = bsdir)
# GSE144894 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE144894/", recursive = T, extended = T)

# "GSE53840" - HIV cohort, age 31 - 68
# Samp <- BLD4$GSE53840$GSE53840_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE53840"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE53840.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE53840.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE53840", baseDir = bsdir)
# GSE53840 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE53840/", recursive = T, extended = T)

# "GSE109914" - age ??? (some pediatric samples present - gen3 - but unable to differentiate)
# Samp <- BLD4$GSE109914$GSE109914_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE109914"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE109914.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE109914.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE109914", baseDir = bsdir)
# GSE109914 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE109914/", recursive = T, extended = T)

# "GSE137594,GSE137634" - age 20 - 92
# Samp <- BLD4$GSE137594$GSE137594_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE137594"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE137594.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE137594.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE137594", baseDir = bsdir)
# GSE137594 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137594/", recursive = T, extended = T)

# "GSE67393" - age 43.6 ± 12.3
# Samp <- BLD4$GSE67393$GSE67393_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE67393"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE67393.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE67393.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE67393", baseDir = bsdir)
# GSE67393 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE67393/", recursive = T, extended = T)

# "GSE107351,GSE107353" - ~ age 60 
# Samp <- BLD4$GSE107351$GSE107351_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE107351"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE107351.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE107351.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE107351", baseDir = bsdir)
# GSE107351 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE107351/", recursive = T, extended = T)

# "GSE129841" - cord blood
# Samp <- BLD4$GSE129841$GSE129841_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE129841"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE129841.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE129841.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE129841", baseDir = bsdir)
# GSE129841 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE129841/", recursive = T, extended = T)

# "GSE52635" - breast cancer cohort
# Samp <- BLD4$GSE52635$GSE52635_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE52635"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE52635.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE52635.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE52635", baseDir = bsdir)
# GSE52635 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE52635/", recursive = T, extended = T)

# "GSE162606" - age 18 -35
# Samp <- BLD4$GSE162606$GSE162606_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE162606"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE162606.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE162606.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE162606", baseDir = bsdir)
# GSE162606 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE162606/", recursive = T, extended = T)

# "GSE128064,GSE128068" - Lynch syndrome cohort, > age 16
# Samp <- BLD4$GSE128064$GSE128064_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE128064"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE128064.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE128064.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE128064", baseDir = bsdir)
# GSE128064 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE128064/", recursive = T, extended = T)

# "GSE53045" - smoking cohort
# Samp <- BLD4$GSE53045$GSE53045_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE53045"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE53045.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE53045.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE53045", baseDir = bsdir)
# GSE53045 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE53045/", recursive = T, extended = T)

# "GSE156792" - age 18 and 62
# Samp <- BLD4$GSE156792$GSE156792_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE156792"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE156792.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE156792.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE156792", baseDir = bsdir)
# GSE156792 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE156792/", recursive = T, extended = T)

# "GSE105123,GSE105124" - age 19 - 23
# Samp <- BLD4$GSE105123$GSE105123_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE105123"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE105123.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE105123.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE105123", baseDir = bsdir)
# GSE105123 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE105123/", recursive = T, extended = T)

# "GSE137593,GSE137634" - age 20 - 92
# Samp <- BLD4$GSE137593$GSE137593_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE137593"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE137593.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE137593.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE137593", baseDir = bsdir)
# GSE137593 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137593/", recursive = T, extended = T)

# "GSE175364" - age 17 - 82
# Samp <- BLD4$GSE175364$`GSE175364-GPL21145_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE175364"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE175364.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE175364.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE175364", baseDir = bsdir)
# GSE175364 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE175364/", recursive = T, extended = T)

# "GSE62992" - age 45 + 
# Samp <- BLD4$GSE62992$GSE62992_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE62992"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE62992.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE62992.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE62992", baseDir = bsdir)
# GSE62992 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE62992/", recursive = T, extended = T)

# "GSE59065" - age 22 -84
# Samp <- BLD4$GSE59065$GSE59065_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE59065"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE59065.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE59065.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE59065", baseDir = bsdir)
# GSE59065 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE59065/", recursive = T, extended = T)

# "GSE154683" - age 19 - 74
# Samp <- BLD4$GSE154683$GSE154683_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE154683"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE154683.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE154683.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE154683", baseDir = bsdir)
# GSE154683 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE154683/", recursive = T, extended = T)

# "GSE43975" - age 27 ± 3.3
# Samp <- BLD4$GSE43975$GSE43975_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE43975"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE43975.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE43975.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE43975", baseDir = bsdir)
# GSE43975 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE43975/", recursive = T, extended = T)

# "GSE56553" - age 19 - 47
# Samp <- BLD4$GSE56553$GSE56553_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE56553"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE56553.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE56553.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE56553", baseDir = bsdir)
# GSE56553 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE56553/", recursive = T, extended = T)

# "GSE78904" - hepatocellular carcinoma cohort, onset ~ 50
# Samp <- BLD4$GSE78904$GSE78904_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE78904"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE78904.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE78904.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE78904", baseDir = bsdir)
# GSE78904 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE78904/", recursive = T, extended = T)

# "GSE41169" - age 18 - 65
# Samp <- BLD4$GSE41169$GSE41169_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE41169"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE41169.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE41169.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE41169", baseDir = bsdir)
# GSE41169 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE41169/", recursive = T, extended = T)

# "GSE110043" - drinking cohort
# Samp <- BLD4$GSE110043$GSE110043_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE110043"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE110043.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE110043.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE110043", baseDir = bsdir)
# GSE110043 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE110043/", recursive = T, extended = T)

# "GSE67751" - age 24 - 68
# Samp <- BLD4$GSE67751$GSE67751_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE67751"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE67751.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE67751.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE67751", baseDir = bsdir)
# GSE67751 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE67751/", recursive = T, extended = T)

# "GSE89093" - Cancer cohort, age 38 - 79
# Samp <- BLD4$GSE89093$GSE89093_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE89093"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE89093.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE89093.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE89093", baseDir = bsdir)
# GSE89093 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE89093/", recursive = T, extended = T)

# "GSE69633,GSE69636" - cord blood
# Samp <- BLD4$GSE69633$GSE69633_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE69633"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE69633.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE69633.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE69633", baseDir = bsdir)
# GSE69633 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE69633/", recursive = T, extended = T)

# "GSE142536" - age 62 - 91
# Samp <- BLD4$GSE142536$GSE142536_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE142536"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE142536.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE142536.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE142536", baseDir = bsdir)
# GSE142536 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE142536/", recursive = T, extended = T)

# "GSE149318" - age 22 - 51
# Samp <- BLD4$GSE149318$GSE149318_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE149318"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE149318.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE149318.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE149318", baseDir = bsdir)
# GSE149318 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE149318/", recursive = T, extended = T)
```

```{r}
load("~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD5_6.RData")

# "GSE152380" - cord blood
# Samp <- BLD5$GSE152380.GSE152380_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE152380"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE152380.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE152380.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE152380", baseDir = bsdir)
# GSE152380 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE152380/", recursive = T, extended = T)

# "GSE96879" - age 22 - 84
# Samp <- BLD5$GSE96879.GSE96879_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE96879"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE96879.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE96879.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE96879", baseDir = bsdir)
# GSE96879 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE96879/", recursive = T, extended = T)

# "GSE155426" - age 21 - 60
# Samp <- BLD5$GSE155426.GSE155426_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE155426"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE155426.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE155426.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE155426", baseDir = bsdir)
# GSE155426 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE155426/", recursive = T, extended = T)

# "GSE98203" - age 15 - 65
# Samp <- BLD5$GSE98203.GSE98203_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE98203"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE98203.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE98203.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE98203", baseDir = bsdir)
# GSE98203 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE98203/", recursive = T, extended = T)

# "GSE176325" - cord blood
# Samp <- BLD5$GSE176325.GSE176325_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE176325"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE176325.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE176325.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE176325", baseDir = bsdir)
# GSE176325 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE176325/", recursive = T, extended = T)

# "GSE77445" age 18 - 69
# Samp <- BLD5$GSE77445.GSE77445_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE77445"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE77445.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE77445.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE77445", baseDir = bsdir)
# GSE77445 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE77445/", recursive = T, extended = T)

# "GSE72776" - age 35.6 - 86.1
# Samp <- BLD5$GSE72776.GSE72776_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72776"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72776.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72776.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72776", baseDir = bsdir)
# GSE72776 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72776/", recursive = T, extended = T)

# "GSE70453,GSE70494" - placenta
# Samp <- BLD5$GSE70453.GSE70453_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE70453"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE70453.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE70453.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE70453", baseDir = bsdir)
# GSE70453 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE70453/", recursive = T, extended = T)

# "GSE138653,GSE138747" - rheumatoid arthritis cohort
# Samp <- BLD5$GSE138653.GSE138653_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE138653"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE138653.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE138653.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE138653", baseDir = bsdir)
# GSE138653 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE138653/", recursive = T, extended = T)

# "GSE76285,GSE76399" - type II diabetes cohort
# Samp <- BLD5$GSE76285.GSE76285_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE76285"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE76285.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE76285.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE76285", baseDir = bsdir)
# GSE76285 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE76285/", recursive = T, extended = T)

# "GSE36064" - 12 - 193 months
# Samp <- BLD5$GSE36064.GSE36064_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE36064"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE36064.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE36064.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE36064", baseDir = bsdir)
# GSE36064 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE36064/", recursive = T, extended = T)

# "GSE146917,GSE147004" - age 33.75 - 59
# Samp <- BLD5$GSE146917.GSE146917_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE146917"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE146917.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE146917.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE146917", baseDir = bsdir)
# GSE146917 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE146917/", recursive = T, extended = T)

# "GSE161020" - Hep B vaccination cohort, age 44 - 73
# Samp <- BLD5$GSE161020.GSE161020_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE161020"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE161020.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE161020.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE161020", baseDir = bsdir)
# GSE161020 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE161020/", recursive = T, extended = T)

# "GSE93266" - age 49.4 ± 1.9
# Samp <- BLD5$GSE93266.GSE93266_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE93266"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE93266.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE93266.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE93266", baseDir = bsdir)
# GSE93266 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE93266/", recursive = T, extended = T)

# "GSE73412" - age ???
# Samp <- BLD5$GSE73412.GSE73412_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE73412"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE73412.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE73412.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE73412", baseDir = bsdir)
# GSE73412 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE73412/", recursive = T, extended = T)

# "GSE80553" - age 18 - 66
# Samp <- BLD5$GSE80553.GSE80553_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE80553"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE80553.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE80553.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE80553", baseDir = bsdir)
# GSE80553 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE80553/", recursive = T, extended = T)

# "GSE104778" - cord blood
# Samp <- BLD5$GSE104778.GSE104778_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE104778"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE104778.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE104778.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE104778", baseDir = bsdir)
# GSE104778 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE104778/", recursive = T, extended = T)

# "GSE107205" - age 55 - 80
# Samp <- BLD5$GSE107205.GSE107205_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE107205"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE107205.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE107205.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE107205", baseDir = bsdir)
# GSE107205 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE107205/", recursive = T, extended = T)

# "GSE85042" - cord blood
# Samp <- BLD5$GSE85042.GSE85042_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE85042"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE85042.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE85042.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE85042", baseDir = bsdir)
# GSE85042 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE85042/", recursive = T, extended = T)

# "GSE140800" - age 21 - 56
# Samp <- BLD5$GSE140800.GSE140800_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE140800"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE140800.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE140800.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE140800", baseDir = bsdir)
# GSE140800 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE140800/", recursive = T, extended = T)

# "GSE151278" - age 20 - 86
# Samp <- BLD5$GSE151278.GSE151278_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE151278"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE151278.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE151278.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE151278", baseDir = bsdir)
# GSE151278 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE151278/", recursive = T, extended = T)

# "GSE62003" - age 67 - 88
# Samp <- BLD5$GSE62003.GSE62003_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE62003"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE62003.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE62003.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE62003", baseDir = bsdir)
# GSE62003 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE62003/", recursive = T, extended = T)

# "GSE109905" - ASD cohort, age 28.6 +- 6.5
# Samp <- BLD5$GSE109905.GSE109905_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE109905"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE109905.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE109905.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE109905", baseDir = bsdir)
# GSE109905 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE109905/", recursive = T, extended = T)

# "GSE123914" - age 51 - 64
# Samp <- BLD5$GSE123914.GSE123914_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE123914"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE123914.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE123914.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE123914", baseDir = bsdir)
# GSE123914 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE123914/", recursive = T, extended = T)

# "GSE67170" - Hep B / C cohort - adult
# Samp <- BLD5$GSE67170.GSE67170_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE67170"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE67170.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE67170.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE67170", baseDir = bsdir)
# GSE67170 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE67170/", recursive = T, extended = T)

# "GSE72867" - cord blood
# Samp <- BLD5$GSE72867.GSE72867_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72867"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72867.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72867.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72867", baseDir = bsdir)
# GSE72867 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72867/", recursive = T, extended = T)

# "GSE98056" - age 18 - 50
# Samp <- BLD5$GSE98056.GSE98056_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE98056"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE98056.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE98056.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE98056", baseDir = bsdir)
# GSE98056 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE98056/", recursive = T, extended = T)

# "GSE99755" - age 24 - 59
# Samp <- BLD5$GSE99755.GSE99755_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE99755"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE99755.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE99755.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE99755", baseDir = bsdir)
# GSE99755 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE99755/", recursive = T, extended = T)

# "GSE64244" - uniparental disomy (UPD) and Turner syndrome cohort, age???
# Samp <- BLD5$GSE64244.GSE64244_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE64244"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE64244.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE64244.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE64244", baseDir = bsdir)
# GSE64244 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE64244/", recursive = T, extended = T)

# "GSE87640,GSE87650" - age 18.3 - 63.7
# Samp <- BLD5$GSE87640.GSE87640_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE87640"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE87640.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE87640.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE87640", baseDir = bsdir)
# GSE87640 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE87640/", recursive = T, extended = T)

# "GSE116378,GSE116380" - age 18 - 71
# Samp <- BLD5$GSE116378.GSE116378_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE116378"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE116378.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE116378.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE116378", baseDir = bsdir)
# GSE116378 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE116378/", recursive = T, extended = T)

# "GSE134429" - age 23 - 81
# Samp <- BLD5$GSE134429.GSE134429_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE134429"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE134429.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE134429.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE134429", baseDir = bsdir)
# GSE134429 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE134429/", recursive = T, extended = T)

# "GSE146092,GSE146093" - CD4T
# Samp <- BLD5$GSE146092.GSE146092_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE146092"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE146092.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE146092.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE146092", baseDir = bsdir)
# GSE146092 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE146092/", recursive = T, extended = T)

# "GSE179759" - Primary Intracerebral Hemorrhage cohort, age 18 + 
# Samp <- BLD5$GSE179759.GSE179759_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE179759"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE179759.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE179759.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE179759", baseDir = bsdir)
# GSE179759 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE179759/", recursive = T, extended = T)

# "GSE61653" - age 18 + 
# Samp <- BLD5$GSE61653.GSE61653_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE61653"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE61653.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE61653.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE61653", baseDir = bsdir)
# GSE61653 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE61653/", recursive = T, extended = T)

# "GSE76169,GSE76171" - age 39 - 63
# Samp <- BLD5$GSE76169.GSE76169_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE76169"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE76169.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE76169.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE76169", baseDir = bsdir)
# GSE76169 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE76169/", recursive = T, extended = T)

# "GSE41273" - age 1 - 48
# Samp <- BLD5$GSE41273.GSE41273_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE41273"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE41273.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE41273.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE41273", baseDir = bsdir)
# GSE41273 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE41273/", recursive = T, extended = T)

# "GSE68456" - cord blood
# Samp <- BLD5$GSE68456.GSE68456_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE68456"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE68456.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE68456.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE68456", baseDir = bsdir)
# GSE68456 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE68456/", recursive = T, extended = T)

# "GSE122288" - cord blood
# Samp <- BLD5$GSE122288.GSE122288_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE122288"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE122288.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE122288.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE122288", baseDir = bsdir)
# GSE122288 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE122288/", recursive = T, extended = T)

# "GSE76170,GSE76171" - age 39 - 63
# Samp <- BLD5$GSE76170.GSE76170_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE76170"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE76170.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE76170.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE76170", baseDir = bsdir)
# GSE76170 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE76170/", recursive = T, extended = T)

# "GSE120228" - early adulthood
# Samp <- BLD5$GSE120228.GSE120228_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE120228"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE120228.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE120228.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE120228", baseDir = bsdir)
# GSE120228 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE120228/", recursive = T, extended = T)

# "GSE51388" - age 23 - 74
# Samp <- BLD6$GSE51388$GSE51388_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE51388"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE51388.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE51388.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE51388", baseDir = bsdir)
# GSE51388 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE51388/", recursive = T, extended = T)

# "GSE80468" - polycystic ovarian syndrome cohort
# Samp <- BLD6$GSE80468$GSE80468_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE80468"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE80468.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE80468.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE80468", baseDir = bsdir)
# GSE80468 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE80468/", recursive = T, extended = T)

# "GSE82218,GSE82221" - age 10 - 46
# Samp <- BLD6$GSE82218$GSE82218_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE82218"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE82218.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE82218.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE82218", baseDir = bsdir)
# GSE82218 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE82218/", recursive = T, extended = T)

# "GSE79144,GSE79262" - not blood
# Samp <- BLD6$GSE79144$GSE79144_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE79144"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE79144.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE79144.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE79144", baseDir = bsdir)
# GSE79144 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE79144/", recursive = T, extended = T)

# "GSE43976" - smoking cohort, age 26 - 59
# Samp <- BLD6$GSE43976$GSE43976_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE43976"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE43976.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE43976.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE43976", baseDir = bsdir)
# GSE43976 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE43976/", recursive = T, extended = T)

# "GSE59550" - smoking cohort, age 45 +- 8
# Samp <- BLD6$GSE59550$GSE59550_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE59550"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE59550.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE59550.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE59550", baseDir = bsdir)
# GSE59550 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE59550/", recursive = T, extended = T)

# "GSE103929" - age 19 - 60
# Samp <- BLD6$GSE103929$GSE103929_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE103929"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE103929.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE103929.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE103929", baseDir = bsdir)
# GSE103929 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE103929/", recursive = T, extended = T)

# "GSE110554,GSE110555" - smoking cohort, age 19 - 59
# Samp <- BLD6$GSE110554$GSE110554_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE110554"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE110554.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE110554.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE110554", baseDir = bsdir)
# GSE110554 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE110554/", recursive = T, extended = T)

# "GSE104287" - age 20–35 
# Samp <- BLD6$GSE104287$GSE104287_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE104287"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE104287.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE104287.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE104287", baseDir = bsdir)
# GSE104287 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE104287/", recursive = T, extended = T)

# "GSE133118" - age 29 - 43
# Samp <- BLD6$GSE133118$GSE133118_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE133118"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE133118.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE133118.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE133118", baseDir = bsdir)
# GSE133118 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE133118/", recursive = T, extended = T)

# "GSE143307" - age 60 - 78
# Samp <- BLD6$GSE143307$GSE143307_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE143307"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE143307.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE143307.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE143307", baseDir = bsdir)
# GSE143307 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE143307/", recursive = T, extended = T)

# "GSE173717" - age 18 - 65 
# Samp <- BLD6$GSE173717$GSE173717_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE173717"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE173717.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE173717.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE173717", baseDir = bsdir)
# GSE173717 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE173717/", recursive = T, extended = T)
# "GSE54399" - sample from pregnant moms
# Samp <- BLD6$GSE54399$GSE54399_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE54399"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE54399.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE54399.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE54399", baseDir = bsdir)
# GSE54399 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE54399/", recursive = T, extended = T)

# "GSE71398,GSE71403" cancer tumor cohort
# Samp <- BLD6$GSE71398$GSE71398_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE71398"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE71398.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE71398.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE71398", baseDir = bsdir)
# GSE71398 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE71398/", recursive = T, extended = T)

# "GSE75679" - age 27 - 87
# Samp <- BLD6$GSE75679$GSE75679_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE75679"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE75679.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE75679.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE75679", baseDir = bsdir)
# GSE75679 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE75679/", recursive = T, extended = T)

# "GSE76503" - age 55 – 75
# Samp <- BLD6$GSE76503$GSE76503_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE76503"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE76503.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE76503.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE76503", baseDir = bsdir)
# GSE76503 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE76503/", recursive = T, extended = T)

# "GSE98224" - preeclampsia cohort
# Samp <- BLD6$GSE98224$GSE98224_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE98224"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE98224.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE98224.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE98224", baseDir = bsdir)
# GSE98224 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE98224/", recursive = T, extended = T)

# "GSE99624" - age 49 - 87
# Samp <- BLD6$GSE99624$GSE99624_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE99624"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE99624.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE99624.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE99624", baseDir = bsdir)
# GSE99624 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE99624/", recursive = T, extended = T)

# "GSE114935" - pregnant women cohort
# Samp <- BLD6$GSE114935$GSE114935_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE114935"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE114935.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE114935.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE114935", baseDir = bsdir)
# GSE114935 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE114935/", recursive = T, extended = T)

# "GSE77056" - age 23 - 29
# Samp <- BLD6$GSE77056$GSE77056_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE77056"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE77056.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE77056.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE77056", baseDir = bsdir)
# GSE77056 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE77056/", recursive = T, extended = T)

# "GSE85506" - age 19 - 80
# Samp <- BLD6$GSE85506$GSE85506_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE85506"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE85506.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE85506.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE85506", baseDir = bsdir)
# GSE85506 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE85506/", recursive = T, extended = T)

# "GSE100386" - age 20 - 61
# Samp <- BLD6$GSE100386$GSE100386_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE100386"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE100386.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE100386.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE100386", baseDir = bsdir)
# GSE100386 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE100386/", recursive = T, extended = T)

# "GSE104376" - cord blood
# Samp <- BLD6$GSE104376$GSE104376_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE104376"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE104376.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE104376.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE104376", baseDir = bsdir)
# GSE104376 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE104376/", recursive = T, extended = T)

# "GSE79312" - cell line
# Samp <- BLD6$GSE79312$GSE79312_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE79312"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE79312.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE79312.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE79312", baseDir = bsdir)
# GSE79312 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE79312/", recursive = T, extended = T)

# "GSE111942" - rheumatoid arthritis cohort
# Samp <- BLD6$GSE111942$GSE111942_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE111942"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE111942.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE111942.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE111942", baseDir = bsdir)
# GSE111942 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE111942/", recursive = T, extended = T)

# "GSE53128,GSE53130" - age 47 - 76.5
# Samp <- BLD6$GSE53128$GSE53128_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE53128"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE53128.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE53128.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE53128", baseDir = bsdir)
# GSE53128 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE53128/", recursive = T, extended = T)

# "GSE141682" - age 18 - 62
# Samp <- BLD6$GSE141682$GSE141682_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE141682"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE141682.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE141682.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE141682", baseDir = bsdir)
# GSE141682 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE141682/", recursive = T, extended = T)

# "GSE153219,GSE153221" - cord blood and placenta
# Samp <- BLD6$GSE153219$GSE153219_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE153219"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE153219.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE153219.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE153219", baseDir = bsdir)
# GSE153219 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE153219/", recursive = T, extended = T)

# "GSE54596" - cell line
# Samp <- BLD6$GSE54596$GSE54596_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE54596"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE54596.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE54596.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE54596", baseDir = bsdir)
# GSE54596 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE54596/", recursive = T, extended = T)

# "GSE51245" - cord blood and 90+ blood
# Samp <- BLD6$GSE51245$GSE51245_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE51245"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE51245.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE51245.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE51245", baseDir = bsdir)
# GSE51245 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE51245/", recursive = T, extended = T)

# "GSE81961" - age 21 - 43
# Samp <- BLD6$GSE81961$GSE81961_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE81961"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE81961.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE81961.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE81961", baseDir = bsdir)
# GSE81961 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE81961/", recursive = T, extended = T)

# "GSE118241" - myelofibrosis - age???
# Samp <- BLD6$GSE118241$GSE118241_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE118241"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE118241.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE118241.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE118241", baseDir = bsdir)
# GSE118241 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE118241/", recursive = T, extended = T)

# "GSE62924" - age 18 - 39
# Samp <- BLD6$GSE62924$GSE62924_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE62924"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE62924.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE62924.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE62924", baseDir = bsdir)
# GSE62924 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE62924/", recursive = T, extended = T)

# "GSE85311" - age 20 - 68
# Samp <- BLD6$GSE85311$GSE85311_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE85311"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE85311.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE85311.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE85311", baseDir = bsdir)
# GSE85311 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE85311/", recursive = T, extended = T)

# "GSE117929,GSE117931" - systemic sclerosis cohort
# Samp <- BLD6$GSE117929$GSE117929_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE117929"
# pd$tissue <- "Peripheral Blood"   
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE117929.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE117929.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE117929", baseDir = bsdir)
# GSE117929 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE117929/", recursive = T, extended = T)

# "GSE128067,GSE128068" - not blood
# Samp <- BLD6$GSE128067$GSE128067_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE128067"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE128067.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE128067.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE128067", baseDir = bsdir)
# GSE128067 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE128067/", recursive = T, extended = T)

# "GSE130748" - age 70 - 78
# Samp <- BLD6$GSE130748$GSE130748_series_matrix.txt.gz
# pd <- pData(Samp)
#     pd$cohort <- "GSE130748"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE130748.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE130748.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE130748", baseDir = bsdir)
# GSE130748 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE130748/", recursive = T, extended = T)

# "GSE103413" - not blood
# Samp <- BLD6$GSE103413$GSE103413_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE103413"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE103413.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE103413.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE103413", baseDir = bsdir)
# GSE103413 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE103413/", recursive = T, extended = T)

# "GSE67490,GSE67491" - age 39 – 61
# Samp <- BLD6$GSE67490$GSE67490_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE67490"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE67490.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE67490.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE67490", baseDir = bsdir)
# GSE67490 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE67490/", recursive = T, extended = T)

# "GSE79056" - cord blood
# Samp <- BLD6$GSE79056$GSE79056_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE79056"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE79056.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE79056.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE79056", baseDir = bsdir)
# GSE79056 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE79056/", recursive = T, extended = T)

# "GSE82084" - cord blood
# Samp <- BLD6$GSE82084$GSE82084_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE82084"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE82084.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE82084.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE82084", baseDir = bsdir)
# GSE82084 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE82084/", recursive = T, extended = T)

# "GSE120307" - age 19 - 54
# Samp <- BLD6$GSE120307$GSE120307_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE120307"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE120307.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE120307.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE120307", baseDir = bsdir)
# GSE120307 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE120307/", recursive = T, extended = T)

# "GSE156546,GSE156547" - age > 18
# Samp <- BLD6$GSE156546$GSE156546_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE156546"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE156546.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE156546.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE156546", baseDir = bsdir)
# GSE156546 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE156546/", recursive = T, extended = T)

# "GSE72354" - T cells
# Samp <- BLD6$GSE72354$GSE72354_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72354"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72354.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72354.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72354", baseDir = bsdir)
# GSE72354 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72354/", recursive = T, extended = T)

# "GSE79329" - age 43 - 71
# Samp <- BLD6$GSE79329$GSE79329_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE79329"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE79329.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE79329.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE79329", baseDir = bsdir)
# GSE79329 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE79329/", recursive = T, extended = T)

# "GSE113527" - cord blood
# Samp <- BLD6$GSE113527$GSE113527_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE113527"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE113527.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE113527.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE113527", baseDir = bsdir)
# GSE113527 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE113527/", recursive = T, extended = T)

# "GSE140344" - age 19 - 34
# Samp <- BLD6$GSE140344$GSE140344_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE140344"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE140344.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE140344.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE140344", baseDir = bsdir)
# GSE140344 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE140344/", recursive = T, extended = T)

# "GSE148663" - breast cancer cohort
# Samp <- BLD6$GSE148663$GSE148663_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE148663"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE148663.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE148663.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE148663", baseDir = bsdir)
# GSE148663 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE148663/", recursive = T, extended = T)

# "GSE162559,GSE162560" - young adult
# Samp <- BLD6$GSE162559$GSE162559_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE162559"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE162559.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE162559.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE162559", baseDir = bsdir)
# GSE162559 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE162559/", recursive = T, extended = T)

# "GSE166611" - age 19 - 69
# Samp <- BLD6$GSE166611$GSE166611_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE166611"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE166611.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE166611.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE166611", baseDir = bsdir)
# GSE166611 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE166611/", recursive = T, extended = T)

# "GSE173382" - age 18 - 65
# Samp <- BLD6$GSE173382$GSE173382_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE173382"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE173382.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE173382.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE173382", baseDir = bsdir)
# GSE173382 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE173382/", recursive = T, extended = T)

# "GSE69502" - not blood
# Samp <- BLD6$GSE69502$GSE69502_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE69502"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE69502.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE69502.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE69502", baseDir = bsdir)
# GSE69502 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE69502/", recursive = T, extended = T)

# "GSE90496,GSE109381" - brain tumor
# Samp <- BLD6$GSE90496$GSE90496_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE90496"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE90496.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE90496.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE90496", baseDir = bsdir)
# GSE90496 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE90496/", recursive = T, extended = T)

# "GSE37965" - breast cancer cohort
# Samp <- BLD6$GSE37965$GSE37965_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE37965"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE37965.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE37965.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE37965", baseDir = bsdir)
# GSE37965 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE37965/", recursive = T, extended = T)

# "GSE68134,GSE69103" - not blood
# Samp <- BLD6$GSE68134$GSE68134_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE68134"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE68134.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE68134.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE68134", baseDir = bsdir)
# GSE68134 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE68134/", recursive = T, extended = T)

# "GSE118144" - purified cells
# Samp <- BLD6$GSE118144$GSE118144_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE118144"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE118144.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE118144.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE118144", baseDir = bsdir)
# GSE118144 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE118144/", recursive = T, extended = T)

# "GSE138074" - age 27 - 94
# Samp <- BLD6$GSE138074$GSE138074_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE138074"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE138074.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE138074.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE138074", baseDir = bsdir)
# GSE138074 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE138074/", recursive = T, extended = T)

# "GSE166844" - purified cell type
# Samp <- BLD6$GSE166844$GSE166844_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE166844"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE166844.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE166844.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE166844", baseDir = bsdir)
# GSE166844 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE166844/", recursive = T, extended = T)

# "GSE87177" - cell lines
# Samp <- BLD6$GSE87177$GSE87177_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE87177"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE87177.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE87177.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE87177", baseDir = bsdir)
# GSE87177 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE87177/", recursive = T, extended = T)

# "GSE165081,GSE165083" - PD cohort
# Samp <- BLD6$GSE165081$GSE165081_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE165081"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE165081.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE165081.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE165081", baseDir = bsdir)
# GSE165081 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE165081/", recursive = T, extended = T)

# "GSE57362" - mostly not blood
# Samp <- BLD6$GSE57362$GSE57362_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE57362"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE57362.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE57362.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE57362", baseDir = bsdir)
# GSE57362 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE57362/", recursive = T, extended = T)

# "GSE86648,GSE86650" - not blood
# Samp <- BLD6$GSE86648$GSE86648_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE86648"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE86648.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE86648.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE86648", baseDir = bsdir)
# GSE86648 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE86648/", recursive = T, extended = T)

# "GSE88824" - purified cells
# Samp <- BLD6$GSE88824$GSE88824_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE88824"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE88824.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE88824.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE88824", baseDir = bsdir)
# GSE88824 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE88824/", recursive = T, extended = T)

# "GSE72495,GSE72498" - in vitro activated B-cells
# Samp <- BLD6$GSE72495$GSE72495_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE72495"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE72495.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE72495.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE72495", baseDir = bsdir)
# GSE72495 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72495/", recursive = T, extended = T)

# "GSE102504" - not pediatric
# Samp <- BLD6$GSE102504$GSE102504_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE102504"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE102504.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE102504.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE102504", baseDir = bsdir)
# GSE102504 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE102504/", recursive = T, extended = T)

# "GSE111183" - age 50 +- 2
# Samp <- BLD6$GSE111183$GSE111183_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE111183"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE111183.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE111183.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE111183", baseDir = bsdir)
# GSE111183 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE111183/", recursive = T, extended = T)

# "GSE75537,GSE75540"
# "GSE85828"
# "GSE95040"
# "GSE100940"
# "GSE103006,GSE103008"
# "GSE104471,GSE104472"   
# "GSE107737"
# "GSE127824"
# "GSE140692"
# "GSE154971"
# "GSE161678,GSE161778"
# "GSE40005"
# "GSE52113,GSE52114"
# "GSE53841"
# "GSE54939"
# "GSE55491"
# "GSE59489"
# "GSE63499"
# "GSE71244,GSE71245"
# "GSE75405,GSE75406"
# "GSE80310"
# "GSE95486,GSE95488"
# "GSE99994,GSE99996"
# "GSE130153"
# "GSE145254"
# "GSE149572"
# "GSE155952"
# "GSE66562,GSE66564"
# "GSE85647,GSE85649"
# "GSE87016"
# "GSE153211"
# "GSE66459"
# "GSE76056"
# "GSE98813,GSE98815"
# "GSE108423"
# "GSE118469"
# "GSE74486,GSE74519"
# "GSE87582"
# "GSE103397"
# "GSE112905"
# "GSE131461"
# "GSE151355"
# "GSE36369"
# "GSE49064,GSE49065"
# "GSE54643,GSE54690"
# "GSE64316"
# "GSE76269"
# "GSE84743,GSE84745"  
# [351] "GSE33233"            "GSE76585"            "GSE93208"            "GSE77797"            "GSE103189"           "GSE107143"           "GSE142439"           "GSE153564"           "GSE65638"            "GSE71328"            "GSE78743"            "GSE81846"            "GSE107211"           "GSE118468"          
# [365] "GSE66555"            "GSE66881"            "GSE66210"            "GSE79969"            "GSE118132,GSE118133" "GSE60821,GSE60924"   "GSE100561,GSE100563" "GSE110530,GSE110555" "GSE112696"           "GSE128801"           "GSE134425"           "GSE142554"           "GSE172464"           "GSE34982,GSE56851"  
# [379] "GSE35069"            "GSE40799,GSE40800"   "GSE44430,GSE65214"   "GSE50967"            "GSE61256,GSE61258"   "GSE62929"            "GSE69954"            "GSE71825"            "GSE73745"            "GSE74738"            "GSE84003"            "GSE44830"            "GSE48472"            "GSE85392"           
# [393] "GSE103738"           "GSE105420"           "GSE133774"           "GSE153668"           "GSE178609"           "GSE69634,GSE69636"   "GSE89472,GSE89474"   "GSE164468"           "GSE39565"            "GSE41782"            "GSE59505,GSE59509"   "GSE68344,GSE79648"   "GSE77283"            "GSE78035,GSE78956"  
# [407] "GSE88937,GSE88938"   "GSE110274"           "GSE123268,GSE123271" "GSE128733,GSE128734" "GSE166503"           "GSE34777,GSE34779"   "GSE50837"            "GSE77965"            "GSE93373"            "GSE167230"           "GSE100825"           "GSE101840"           "GSE105066,GSE105067" "GSE107917"          
# [421] "GSE108562,GSE108564" "GSE108785,GSE108815" "GSE110555,GSE112618" "GSE110724,GSE110821" "GSE159479,GSE159480" "GSE34486,GSE34487"   "GSE50475"            "GSE54670,GSE54671"   "GSE55734"            "GSE58651"            "GSE63849"            "GSE68342,GSE68344"   "GSE69229"            "GSE80969"           
# [435] "GSE82234"            "GSE83156,GSE83159"   "GSE95761"            "GSE97497"            "GSE103010"           "GSE105078"           "GSE120428,GSE120558" "GSE37966,GSE38128"   "GSE40790"            "GSE52576,GSE52578"   "GSE60923,GSE60924"   "GSE86829,GSE86833"   "GSE86831,GSE86833"   "GSE106902"          
# [449] "GSE109364"           "GSE116375,GSE144196" "GSE119260"           "GSE149568"           "GSE151456,GSE151457" "GSE152245"           "GSE39141"            "GSE42409"            "GSE59507,GSE59509"   "GSE59796"            "GSE65078,GSE65079"   "GSE66247"            "GSE75310"            "GSE80377"           
# [463] "GSE85293,GSE85296"   "GSE85523,GSE85525"   "GSE90933,GSE90934"   "GSE94326,GSE94368"   "GSE116338"           "GSE116754"           "GSE122126"           "GSE131177"           "GSE40699"            "GSE41933"            "GSE42865"            "GSE46648"            "GSE60816,GSE60924"   "GSE60924,GSE81452"  
# [477] "GSE71837"            "GSE75937,GSE75939"   "GSE76639,GSE76641"   "GSE83458"            "GSE131541"           "GSE30654,GSE31848"   "GSE41114,GSE41117"   "GSE47627"            "GSE59508,GSE59509"   "GSE64491,GSE64511"   "GSE64874,GSE64934"   "GSE101641"           "GSE102822"           "GSE109379,GSE109381"
# [491] "GSE116924"           "GSE119774,GSE119776" "GSE124413"           "GSE124567,GSE124569" "GSE132845"           "GSE166212"           "GSE174625"           "GSE40360"            "GSE51547"            "GSE69633"            "GSE73938"            "GSE74877"            "GSE75196"            "GSE77036"           
# [505] "GSE86409"            "GSE86961"
```

```{r}
cohorts <- c("GSE105018", "GSE82273", "GSE152027", "GSE87571", "GSE89353", "GSE103657", "GSE77716", "GSE131433", "GSE112611", "GSE142512", "GSE132181", "GSE87648", "GSE56105", 
             "GSE89278", "GSE80283", "GSE73103", "GSE99863", "GSE159930", "GSE97362", "GSE56600", "GSE40576", "GSE60132", "GSE110128", "GSE110828", "GSE79257", "GSE113967", 
             "GSE93933", "GSE59592", "GSE74432", "GSE36054", "GSE64495", "GSE47880", "GSE124366", "GSE112987", "GSE64380", "GSE78773", "GSE152204", "GSE52588", "GSE59999", 
             "GSE67444", "GSE119778", "GSE62298", "GSE166531", "GSE83424", "GSE104451", "GSE62219", "GSE118940", "GSE152427", "GSE57204", "GSE44798", "GSE104812", "GSE131752", 
             "GSE152084", "GSE162554", "GSE165170", "GSE175379", "GSE32148", "GSE57107", "GSE84624", "GSE90117", "GSE51180", "GSE72777", "GSE119684", "GSE145233", "GSE66552", 
             "GSE116300", "GSE125367", "GSE133801", "GSE60598", "GSE154446", "GSE30870", "GSE102177", "GSE109430", "GSE113012", "GSE111165", "GSE83334", "GSE116992", "GSE48300")
Sys.setenv("VROOM_CONNECTION_SIZE" = 1310720000)
getOption("download.file.method.GEOquery")
options(download.file.method.GEOquery = "libcurl")
dest <- "~/KoborLab/kobor_space/mfu/GEO/Blood"
BLD <- sapply(cohorts, function(x) getGEO(x, destdir = dest))
save(BLD, file = "~/KoborLab/kobor_space/mfu/GEO/Blood/01-BLD.RData")
lapply(cohorts, function(x) getGEOSuppFiles(x))

### "GSE105018" - age 18 for everyone
Samp <- BLD$GSE105018$GSE105018_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE105018"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$sex <- pd$`gender:ch1`
pd$age <- 18
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE105018.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "familyid:ch1", "twinid:ch1", "zygosity:ch1")]
colnames(GSE105018.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Family_ID", "Twin_ID", "Twin_Zygosity")
#getGEOSuppFiles("GSE105018", baseDir = bsdir)
#GSE105018 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE105018/", recursive = T, extended = T)

### "GSE82273" - newborn infant blood
Samp <- BLD$GSE82273$GSE82273_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE82273"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spots"
pd$age <- 2/365 # hill prick taken 2-3 days after delivery
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
pd$disease_status <- "Healthy Control" # 274 samples have cleft lips and 143 samples have cleft palates, but we will consider them "healthy" for this study
GSE82273.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
colnames(GSE82273.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
#getGEOSuppFiles("GSE82273", baseDir = bsdir)
#GSE82273 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE82273/", recursive = T, extended = T)

### "GSE152027" - age 13 - 72
Samp <- BLD$GSE152027$GSE152027_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE152027"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`ageatbloodcollection:ch1`)
pd$sex <- pd$`gender:ch1`
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`status:ch1` %>% gsub("CON", "Healthy Control", .) %>% gsub("FEP", "First-Episode Psychosis", .) %>% gsub("SCZ", "Schizophrenia", .)
GSE152027.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
colnames(GSE152027.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE152027", baseDir = bsdir)
# GSE152027 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE152027/", recursive = T, extended = T)

### "GSE87571" age 14 - 94
Samp <- BLD$GSE87571$GSE87571_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE87571"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE87571.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status")]
colnames(GSE87571.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE87571", baseDir = bsdir)
# GSE87571 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE87571/", recursive = T, extended = T)

### "GSE89353" - newborn - 50 (average 10yr)
Samp <- BLD$GSE89353$GSE89353_series_matrix.txt.gz
pd <- pData(Samp)
GSE89353_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE89353_appendix.xlsx") %>% as.data.frame()
pd <- left_join(pd, GSE89353_appendix, by = c("individual:ch1" = "Sample_ID"))
pd$cohort <- "GSE89353"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`Age at enrollment in years`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Congenital Disorders"
GSE89353.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "individual:ch1", "Growth", "Head circumference", "Intellectual Disability", "Autism Spectrum Disorder", "Epilepsy/Seizures", "Behaviour/Neurologic abnormalities", "Congenital Heart Defects", "Other Malformations/Anomalies", "Facial Dysmorphism", "Family history")]
colnames(GSE89353.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Individual", "Growth", "Head circumference", "Intellectual Disability", "Autism Spectrum Disorder", "Epilepsy/Seizures", "Behaviour/Neurologic abnormalities", "Congenital Heart Defects", "Other Malformations/Anomalies", "Facial Dysmorphism", "Family history")
# getGEOSuppFiles("GSE89353", baseDir = bsdir)
# GSE89353 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE89353/", recursive = T, extended = T)

### "GSE56105" - age 10 - 75
Samp <- BLD$GSE56105$GSE56105_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE56105"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- pd$`gender:ch1`
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE56105.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "twin:ch1")]
colnames(GSE56105.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Twin_Zygosity")
# getGEOSuppFiles("GSE56105", baseDir = bsdir)
# GSE56105 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE56105/", recursive = T, extended = T)

### "GSE103657" - neonatal dried blood
Samp <- BLD$GSE103657$GSE103657_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE103657"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spots"
pd$age <- 2/365
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control" # Some has insulin resistance at age 5
GSE103657.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
colnames(GSE103657.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE103657", baseDir = bsdir)
# GSE103657 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE103657/", recursive = T, extended = T)

### "GSE77716" - GALAII Latino pediatric cohort - no age need to ask
Samp <- BLD$GSE77716$GSE77716_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE77716"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE77716.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "ethnicity:ch1", "disease_status", "gran_basophils:ch1", "gran_eosinophils:ch1", "gran_neutrophils:ch1", "lymphocytes:ch1", "monocytes:ch1")]
colnames(GSE77716.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Ethnicity", "Disease_Status", "Basophils", "Eosinophils", "Neutrophils", "Lymphocytes", "Monocytes")
# getGEOSuppFiles("GSE77716", baseDir = bsdir)
# GSE77716 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE77716/", recursive = T, extended = T)

### "GSE131433" - neonatal blood
Samp <- BLD$GSE131433$GSE131433_series_matrix.txt.gz
pd <- pData(Samp)
pd <- pd[pd$`sample type:ch1` == "Guthrie blood spot", ]
pd$cohort <- "GSE131433"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spots"
pd$age <- 2/365
pd$sex <- pd$`gender:ch1`
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE131433.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "art status:ch1", "art subtype:ch1")]
colnames(GSE131433.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Assisted_Reproductive_Technology_Status", "Assisted_Reproductive_Technology_Subtype")
# getGEOSuppFiles("GSE131433", baseDir = bsdir)
# GSE131433 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE131433/", recursive = T, extended = T)

### "GSE112611" - age 5 - 21
Samp <- BLD$GSE112611$GSE112611_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE112611"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$individual <- gsub("_36M", "", pd$title) %>% gsub("_At.*", "", .) %>% gsub("_24M", "", .) %>% gsub("_12M", "", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`diagnosis:ch1` %>% gsub("non-IBD control", "Healthy Control", .)
GSE112611.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "baseline vs follow-up:ch1", "individual", "disease stage at at follow up:ch1", "disease stage at diagnosis:ch1")]
colnames(GSE112611.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Time_Point", "Individual", "Disease_Stage_Followup", "Disease_Stage_Diagnosis")
# getGEOSuppFiles("GSE112611", baseDir = bsdir)
# GSE112611 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE112611/", recursive = T, extended = T)       

### "GSE142512" - age 1 - 16 (450K)
Samp <- BLD$GSE142512$`GSE142512-GPL13534_series_matrix.txt.gz`
pd <- pData(Samp)
pd$cohort <- "GSE142512"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`t1d status:ch1` %>% gsub("T1D case", "Type 1 Diabetes", .) %>% gsub(".*control", "Healthy Control", .)
GSE142512_450k.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "subject id:ch1", "disease time point:ch1")]
colnames(GSE142512_450k.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Individual", "Time_Point")
# getGEOSuppFiles("GSE142512", baseDir = bsdir)
# GSE142512 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE142512/", recursive = T, extended = T)

### "GSE142512" - age 0 - 23 (EPIC)
Samp <- BLD$GSE142512$`GSE142512-GPL21145_series_matrix.txt.gz`
pd <- pData(Samp)
pd$cohort <- "GSE142512"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`t1d status:ch1` %>% gsub("T1D case", "Type 1 Diabetes", .) %>% gsub(".*control", "Healthy Control", .)
GSE142512_EPIC.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "subject id:ch1", "disease time point:ch1")]
colnames(GSE142512_EPIC.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Individual", "Time_Point")
# getGEOSuppFiles("GSE142512", baseDir = bsdir)
# GSE142512 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE142512/", recursive = T, extended = T)

### "GSE132181" - age birth or age 7
Samp <- BLD$GSE132181$GSE132181_series_matrix.txt.gz
pd <- pData(Samp)
pd <- pd[pd$`age:ch1` == "age 7", ]
pd$cohort <- "GSE132181"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- 7
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE132181.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "ancestry_pc1:ch1", "ancestry_pc2:ch1", "gestationalage_birth:ch1", "samplecollectionsite:ch1", "selfreportedrace:ch1")]
colnames(GSE132181.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Ancestry_PC1", "Ancestry_PC2", "Gestational_Age_Birth", "Sample_Collection_Site", "Race_Self_Reported")
# getGEOSuppFiles("GSE132181", baseDir = bsdir)
# GSE132181 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE132181/", recursive = T, extended = T)

### "GSE87648,GSE87650" - age 0 - 79
Samp <- BLD$GSE87648$GSE87648_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE87648"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`simplified_diagnosis:ch1` %>% gsub("HL", "Healthy Control", .) %>% gsub("CD|HS|UC", "Inflammatory Bowel Disease", .)  
pd$disease_subtype <- pd$`simplified_diagnosis:ch1` %>% gsub("HL|HS", NA, .) %>% gsub("CD", "Crohn's disease", .) %>% gsub("UC", "Ulcerative colitis", .)
GSE87648.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "smoking status:ch1", "patient_number:ch1", "disease_subtype")]
colnames(GSE87648.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Smoking_Status", "Individual", "Disease_Subtype")
# getGEOSuppFiles("GSE87648", baseDir = bsdir)
# GSE87648 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE87648/", recursive = T, extended = T)

### "GSE89278" - neonatal blood spot
Samp <- BLD$GSE89278$GSE89278_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE89278"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spots"
pd$age <- 2/365
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control" # some took DHA supplement
pd$dha <- pd$title %>% gsub("_CONTROL", "N", .) %>% gsub("_DHA", "Y", .)
GSE89278.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "dha")]
colnames(GSE89278.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "DHA_Supplement")
# getGEOSuppFiles("GSE89278", baseDir = bsdir)
# GSE89278 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE89278/", recursive = T, extended = T)

### "GSE80283" - neonatal blood spot
Samp <- BLD$GSE80283$GSE80283_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE80283"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spots"
pd$age <- 2/365
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Premature or small for gestational age"
GSE80283.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "race:ch1", "ga weeks:ch1")]
colnames(GSE80283.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Race_Self_Reported", "Gestational_Age_Birth")
# getGEOSuppFiles("GSE80283", baseDir = bsdir)
# GSE80283 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE80283/", recursive = T, extended = T)

### "GSE73103" - age 14 - 34
Samp <- BLD$GSE73103$GSE73103_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE73103"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE73103.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "rs1011731:ch1", "rs10150332:ch1", "rs10195252:ch1", "rs10458787:ch1", "rs1055144:ch1", "rs10767664:ch1", "rs10769908:ch1", "rs10783050:ch1", "rs10838738:ch1", "rs10993160:ch1", "rs11142387:ch1", "rs1152846:ch1", "rs12517906:ch1", "rs12597579:ch1", "rs1294421:ch1", "rs13107325:ch1", "rs1443512:ch1", "rs1514175:ch1", "rs17782313:ch1", "rs1878047:ch1", "rs1927702:ch1", "rs206936:ch1", "rs2112347:ch1", "rs2145270:ch1", "rs2206734:ch1", "rs2241423:ch1", "rs2287019:ch1", "rs2383393:ch1", "rs2444217:ch1", "rs261967:ch1", "rs2815752:ch1", "rs3934834:ch1", "rs4377469:ch1", "rs4823006:ch1", "rs516636:ch1", "rs652722:ch1", "rs6784615:ch1", "rs6794092:ch1", "rs6795735:ch1", "rs6861681:ch1", "rs6905288:ch1", "rs713586:ch1", "rs7138803:ch1", "rs718314:ch1", "rs7481311:ch1", "rs7498665:ch1", "rs7647305:ch1", "rs7832552:ch1", "rs824931:ch1", "rs887912:ch1", "rs984222:ch1", "rs9939609:ch1", "weight category:ch1")]
colnames(GSE73103.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "rs1011731", "rs10150332", "rs10195252", "rs10458787", "rs1055144", "rs10767664", "rs10769908", "rs10783050", "rs10838738", "rs10993160", "rs11142387", "rs1152846", "rs12517906", "rs12597579", "rs1294421", "rs13107325", "rs1443512", "rs1514175", "rs17782313", "rs1878047", "rs1927702", "rs206936", "rs2112347", "rs2145270", "rs2206734", "rs2241423", "rs2287019", "rs2383393", "rs2444217", "rs261967", "rs2815752", "rs3934834", "rs4377469", "rs4823006", "rs516636", "rs652722", "rs6784615", "rs6794092", "rs6795735", "rs6861681", "rs6905288", "rs713586", "rs7138803", "rs718314", "rs7481311", "rs7498665", "rs7647305", "rs7832552", "rs824931", "rs887912", "rs984222", "rs9939609", "BMI_Category")
# getGEOSuppFiles("GSE73103", baseDir = bsdir)
# GSE73103 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE73103/", recursive = T, extended = T)

### "GSE99863" - age 2
Samp <- BLD$GSE99863$GSE99863_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE99863"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- 2
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE99863.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "season:ch1")]
colnames(GSE99863.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Season_of_Conception")
# getGEOSuppFiles("GSE99863", baseDir = bsdir)
# GSE99863 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE99863/", recursive = T, extended = T)

### "GSE159930" - age 0 - 36
Samp <- BLD$GSE159930$GSE159930_series_matrix.txt.gz
pd <- pData(Samp)
pd$Sample_ID <- pd$title %>% gsub(" \\[Blood\\]| \\[Heart\\]", "", .)
GSE159930_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE159930_appendix.xlsx", skip = 1)
pd <- left_join(pd, GSE159930_appendix, by = c("Sample_ID" = "Sample ID"))
pd$cohort <- "GSE159930"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$year <- gsub("[0-9]{2}m", "", pd$Age) %>% gsub("[0-9]m", "", .) %>% gsub("y.*", "", .) %>% gsub("Newborn", 2/365, .) %>% as.numeric() 
pd$year[is.na(pd$year)] <- 0
pd$month <- gsub(".*, ", "", pd$Age) %>% gsub("[0-9]{2}y", "", .) %>% gsub("[0-9]y", "", .) %>% gsub("m", "", .) %>% gsub("Newborn", 0, .) %>% as.numeric()
pd$month[is.na(pd$month)] <- 0
pd$age <- pd$year + (pd$month / 12)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
GSE159930.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Gender", "diagnosis:ch1", 
                      "Fyler cardiac diagnose", "Head Abnormalities Present", "Eye Abnormalities Present", "Ear Abnormalities Present", "Nose Abnormalities Present", 
                      "Neck Abnormalities Present", "Mouth Abnormalities Present", "Palate Abnormalities Present", "Airway Abnormalities Present", "Chest Abnormalities Present", 
                      "Pulmonary Abnormalities Present", "Hepatic Abnormalities Present", "Dysmorphic Facies Present", "Abdominal Abnormalities Present", 
                      "Renal Abnormalities Present", "Genitourinary Abnormalities Present", "General Skeletal Abnormalities Present", "Hand Abnormalities Present", 
                      "Feet Abnormalities Present", "Neurological Abnormalities Present", "Dermatological Abnormalities Present", "Endocrinologic Abnormalities Present", 
                      "Hematologic Abnormalities Present", "Genetic Syndromes Present")]
colnames(GSE159930.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status",
                           "Fyler cardiac diagnose", "Head Abnormalities", "Eye Abnormalities", "Ear Abnormalities", "Nose Abnormalities",
                           "Neck Abnormalities", "Mouth Abnormalities", "Palate Abnormalities", "Airway Abnormalities", "Chest Abnormalities",
                           "Pulmonary Abnormalities", "Hepatic Abnormalities", "Dysmorphic Facies", "Abdominal Abnormalities",
                           "Renal Abnormalities", "Genitourinary Abnormalities", "General Skeletal Abnormalities", "Hand Abnormalities",
                           "Feet Abnormalities", "Neurological Abnormalities", "Dermatological Abnormalities", "Endocrinologic Abnormalities",
                           "Hematologic Abnormalities", "Genetic Syndromes")
# getGEOSuppFiles("GSE159930", baseDir = bsdir)
# GSE159930 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE159930/", recursive = T, extended = T)

### "GSE97362" - age 0.01 - 52
Samp <- BLD$GSE97362$GSE97362_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE97362"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age (years):ch1`)
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
GSE97362.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "chip", "position", "tissue", "celltype", "age", "sex", "disease state:ch1", "sample type:ch1")]
colnames(GSE97362.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Sentrix_ID", "Sentrix_Position", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Disease_Subtype")
# getGEOSuppFiles("GSE97362", baseDir = bsdir)
# GSE97362 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE97362/", recursive = T, extended = T)

### "GSE56600,GSE56602" - B-ALL cohort (age 0.3 - 14) - not peripheral blood
# Samp <- BLD$GSE56600$GSE56600_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE56600"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE56600.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE56600.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE56600", baseDir = bsdir)
# GSE56600 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE56600/", recursive = T, extended = T)

### "GSE40576,GSE40736" - age 6 - 12 - ask for age
# Samp <- BLD$GSE40576$GSE40576_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE40576"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "PBMC"
# pd$age <- as.numeric(pd$`age:ch1`)
# # pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# # pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`asthma:ch1` %>% gsub("TRUE", "Asthma", .) %>% gsub("FALSE", "Healthy Control", .)
# GSE40576.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "subtype:ch1")]
# colnames(GSE40576.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Disease_Subtype")
# getGEOSuppFiles("GSE40576", baseDir = bsdir)
# GSE40576 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE40576/", recursive = T, extended = T)

### "GSE60132" - age 6 - 85
Samp <- BLD$GSE60132$GSE60132_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE60132"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE60132.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "ancestry:ch1")]
colnames(GSE60132.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Ancestry")
# getGEOSuppFiles("GSE60132", baseDir = bsdir)
# GSE60132 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE60132/", recursive = T, extended = T)

### "GSE110128" - 90 PBMC at age 11
Samp <- BLD$GSE110128$GSE110128_series_matrix.txt.gz
pd <- pData(Samp)
pd <- pd[pd$`tissue:ch1` == "mononuclear cells", ]
pd$cohort <- "GSE110128"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- 11
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`allergy:ch1` %>% gsub("Control", "Healthy Control", .)
GSE110128.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
colnames(GSE110128.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE110128", baseDir = bsdir)
# GSE110128 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE110128/", recursive = T, extended = T)

### "GSE110828,GSE110829" - 47 postnatal peripheral blood
Samp <- BLD$GSE110828$GSE110828_series_matrix.txt.gz
pd <- pData(Samp)
pd <- pd[pd$source_name_ch1 == "PBMC", ]
pd$cohort <- "GSE110828"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- 2/365
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Premature or small for gestational age"
GSE110828.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "gestational age:ch1", "disease_status", "maternal age:ch1", "maternal bmi:ch1", "parity:ch1", "paternal age:ch1", "paternal bmi:ch1", "placenta previa:ch1", "preeclampsia:ch1", "smoking before pregnancy:ch1", "idiopathic prom without inflammation:ch1", "assisted reproductive technology:ch1", "birthweight sdscore:ch1", "chorioamnionitis:ch1", "delivery:ch1", "batch:ch1")]
colnames(GSE110828.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Gestational_Age_Birth", "Disease_Status", "Maternal_Age", "Maternal_BMI", "Parity", "Paternal_Age", "Paternal_BMI", "Placenta_Previa", "Preeclampsia", "Parental_Smoking_Status", "Idiopathic_PROM_without_Inflammation", "Assisted_Reproductive_Technology", "Birthweight_SD_Score", "Chorioamnionitis", "Delivery_Method", "Batch")
# getGEOSuppFiles("GSE110828", baseDir = bsdir)
# GSE110828 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE110828/", recursive = T, extended = T)

# "GSE79257" - neonatal dried blood
Samp <- BLD$GSE79257$GSE79257_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE79257"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spot"
pd$age <- 2/365
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE79257.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "conception type:ch1", "chip", "position")]
colnames(GSE79257.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Assisted_Reproductive_Technology", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE79257", baseDir = bsdir)
# GSE79257 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE79257/", recursive = T, extended = T)

# "GSE113967" - age 2 - 13
Samp <- BLD$GSE113967$GSE113967_series_matrix.txt.gz
GSE113967_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE113967_appendix.xlsx") %>% as.data.frame()
pd <- pData(Samp)
pd$sample_id <- gsub(" .*", "", pd$title)
pd <- left_join(pd, GSE113967_appendix, by = c("sample_id" = "Sample ID"))
pd$cohort <- "GSE113967"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`Age (years)`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1`
GSE113967.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "Inheritance", "chip", "position")]
colnames(GSE113967.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Variant_Inheritance", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE113967", baseDir = bsdir)
# GSE113967 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE113967/", recursive = T, extended = T)

# "GSE93933" - age ???
# Samp <- BLD$GSE93933$GSE93933_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE93933"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE93933.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE93933.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE93933", baseDir = bsdir)
# GSE93933 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE93933/", recursive = T, extended = T)

# "GSE59592" - 2-8 months 
Samp <- BLD$GSE59592$GSE59592_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE59592"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- 0.3 # 2-8 month, average 3.6
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE59592.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "afb1 exposure:ch1")]
colnames(GSE59592.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "AFB1_Exposure")
# getGEOSuppFiles("GSE59592", baseDir = bsdir)
# GSE59592 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE59592/", recursive = T, extended = T)

# "GSE74432" - age 0.6 – 40 
Samp <- BLD$GSE74432$GSE74432_series_matrix.txt.gz
GSE74432_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE74432_appendix.xlsx") %>% as.data.frame()
GSE74432_appendix <- GSE74432_appendix[GSE74432_appendix$`Disease State` != "Control-Fibroblast", ] %>% .[.$`Disease Subtype` != "Fibroblast derived DNA from Sotos patients", ]
pd <- pData(Samp)
pd$cohort <- "GSE74432"
pd$sampleid <- gsub(" .*", "", pd$title)
pd <- right_join(pd, GSE74432_appendix, by = c("sampleid" = "Sample ID"))
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`Age at sample collection (years)`)
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`Disease State` %>% 
    gsub("Control-blood", "Healthy Control", .)
GSE74432.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "Disease Subtype", "Affected Gene", "Exon", "Mutation", "Protein change", "Inheritance", "Birth weight (kg)", "Gestational age (weeks)", "Birth weight (percentile)", "Age at assessment", "Height (SD)", "Weight (SD)", "OFC (SD)", "CNS anomalies", "Intellectual disability", "Autism Spectrum Disorder", "Seizures", "Advanced bone age", "Renal anomalies", "Cardiovascular anomalies", "chip", "position")]
colnames(GSE74432.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Mutation_Type", "Affected_Gene", "Exon", "Mutation", "Protein_Change", "Inheritance", "Birth_Weight_kg", "Gestational_Age_weeks", "Birth_Weight_percentile", "Age_at_Assessment", "Height_SD", "Weight_SD", "OFC_SD", "CNS_Anomalies", "Intellectual_Disability", "Autism_Spectrum_Disorder", "Seizures", "Advanced_Bone_Age", "Renal_Anomalies", "Cardiovascular_Anomalies", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE74432", baseDir = bsdir)
# GSE74432 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE74432/", recursive = T, extended = T)

# "GSE36054" - age 12 - 203 months
Samp <- BLD$GSE36054$GSE36054_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE36054"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- as.numeric(pd$`age at collection months:ch1`)/12
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE36054.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "ethnicity:ch1", "disease_status", "relation", "sentrix id:ch1", "sentrix position:ch1")]
colnames(GSE36054.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Race_Self_Reported", "Disease_Status", "Sample_Relation", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE36054", baseDir = bsdir)
# GSE36054 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE36054/", recursive = T, extended = T)

# "GSE64495" - age 2 - 74
Samp <- BLD$GSE64495$GSE64495_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE64495"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("female", "F", pd$`Sex:ch1`) %>% gsub("male", "M", .)
pd$disease_status <- pd$`disease status:ch1` %>% gsub("Control", "Healthy Control", .)
GSE64495.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "dna methylation age:ch1", "family number:ch1", "relation to proband:ch1")]
colnames(GSE64495.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "DNAm_Age", "Family", "Sample_Relation")
# getGEOSuppFiles("GSE64495", baseDir = bsdir)
# GSE64495 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE64495/", recursive = T, extended = T)

# "GSE47880" - age???, GSA cohort
# Samp <- BLD$GSE47880$GSE47880_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE47880"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE47880.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE47880.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE47880", baseDir = bsdir)
# GSE47880 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE47880/", recursive = T, extended = T)

# "GSE124366" - age 4 - 13
Samp <- BLD$GSE124366$GSE124366_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE124366"
pd$tissue <- pd$`tissue:ch1` %>% gsub("PBMC", "Peripheral Blood", .) 
pd$celltype <- pd$`tissue:ch1`
pd$age <- as.numeric(pd$`age at sample collection (years):ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE124366.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "chip", "position")]
colnames(GSE124366.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE124366", baseDir = bsdir)
# GSE124366 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE124366/", recursive = T, extended = T)

# "GSE112987,GSE113018" - age 0.02 - 18
Samp <- BLD$GSE112987$GSE112987_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE112987"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("control", "Healthy Control", .)
GSE112987.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "cns:ch1", "facial:ch1", "growth:ch1", "chip", "position")]
colnames(GSE112987.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "FASD_CNS_Score", "FASD_Facial_Score", "FASD_Growth_Score", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE112987", baseDir = bsdir)
# GSE112987 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE112987/", recursive = T, extended = T)

# "GSE64380"- age 4 months-25 years
Samp <- BLD$GSE64380$GSE64380_series_matrix.txt.gz
pd <- pData(Samp)
GSE64380_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE64380_appendix.xlsx") %>% as.data.frame()
GSE64380_appendix$patient <- as.character(GSE64380_appendix$patient)
pd$cohort <- "GSE64380"
pd$sampleid <- gsub(".* ", "", pd$title)
pd <- left_join(pd, GSE64380_appendix, by = c("sampleid" = "patient"))
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age at last evaluation`)
pd$sex <- gsub("female", "F", pd$sex) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`diagnosis:ch1` %>% gsub("intellectual disability (ID)", "Intellectual Disability", .) %>% gsub("normal (control)", "Healthy Control", .)
GSE64380.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "birth parameters", "stature", "head", "dysmorphism", "malformation", "delay", "muscular hypotonia", "obesity", "miscellaneous", "family history", "epilepsy")]
colnames(GSE64380.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Birth_Parameters", "Stature", "Head_Size", "Dysmorphism", "Physical_Malformation", "Developmental_Delay", "Muscular_Hypotonia", "Obesity", "Other_Clinical_Diagnosis", "Family_History", "Epilepsy")
# getGEOSuppFiles("GSE64380", baseDir = bsdir)
# GSE64380 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE64380/", recursive = T, extended = T)

# "GSE78773,GSE78956" - age??? - need access to future medicine
# Samp <- BLD$GSE78773$GSE78773_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE78773"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE78773.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE78773.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE78773", baseDir = bsdir)
# GSE78773 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE78773/", recursive = T, extended = T)

library(stringr)
# "GSE152204" - Oculo-Auriculo-Vertebral Spectrum cohort, age 3 - 14
Samp <- BLD$GSE152204$GSE152204_series_matrix.txt.gz
pd <- pData(Samp)
GSE152204_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE152204_appendix.xlsx") %>% as.data.frame()
GSE152204_appendix <- apply(GSE152204_appendix, 2, function(x) {
    x <- gsub("\\-", "0", x) %>% gsub("\\+", "1", .)
    return(x)
}) %>% as.data.frame()
GSE152204_appendix$`OAVS Code` <- as.numeric(GSE152204_appendix$`OAVS Code`) %>% str_pad(., 2, pad = "0") %>% paste0("OAVS", .)
pd$cohort <- "GSE152204"
pd <- left_join(pd, GSE152204_appendix, by = c("title" = "OAVS Code"))
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`Age (years)`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("control", "Healthy Control", .) 
GSE152204.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex", "disease_status", "Familial OAVS", "Prenatal factors", "Hemifacial microsomia", "Cleft lip/palate", "Facial nerve weakness", "Mandibular hypoplasia", "Frontal bossing", "Other craniofacial features", "Preauricular skin tags/pits", "Atresia of external auditory canal", "Microtia/anotia", "Hearing loss", "Dysplastic ears", "Asymmetric ears", "Epibulbar dermoid", "Micropthalmos", "Coloboma upper eyelid", "Other eye anomalies", "Hemivertebrae", "Fusion", "Scoliosis", "Kyphosis", "Other vertebral anomalies", "Cardiovascular", "Brain", "Developmental delay", "Genitourinary malformation", "Radial defects", "Other organs/systems", "chip", "position")]
colnames(GSE152204.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Familial_OAVS", "Prenatal_Factors", "Hemifacial_Microsomia", "Cleft_Lip_or_Palate", "Facial_Nerve_Weakness", "Mandibular_Hypoplasia", "Frontal_Bossing", "Other_Craniofacial_Features", "Preauricular_Skin_Tags_or_Pits", "Atresia_of_External_Auditory_Canal", "Microtia_or_Anotia", "Hearing_Loss", "Dysplastic_Ears", "Asymmetric_Ears", "Epibulbar_Dermoid", "Micropthalmos", "Coloboma_Upper_Eyelid", "Other_Eye_Anomalies", "Hemivertebrae", "Fusion", "Scoliosis", "Kyphosis", "Other_Vertebral_Anomalies", "Cardiovascular", "Brain", "Developmental_Delay", "Genitourinary_Malformation", "Radial_Defects", "Other_Organs_or_Systems", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE152204", baseDir = bsdir)
# GSE152204 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE152204/", recursive = T, extended = T)

# "GSE52588" - Down syndrome cohort, age 9 - 83
Samp <- BLD$GSE52588$GSE52588_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE52588"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("healthy", "Healthy Control", .)
GSE52588.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "group:ch1", "family:ch1")]
colnames(GSE52588.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sample_Relation", "Family")
# getGEOSuppFiles("GSE52588", baseDir = bsdir)
# GSE52588 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE52588/", recursive = T, extended = T)

# "GSE59999" - age 11 - 15 months - pretend everyone is 12 months old
Samp <- BLD$GSE59999$GSE59999_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE59999"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- 1
pd$sex <- gsub("FEMALE", "F", pd$`gender:ch1`) %>% gsub("MALE", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`challenge outcome:ch1` %>% gsub("nonallergic", "Healthy Control", .)
GSE59999.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "phenotype:ch1")]
colnames(GSE59999.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Disease_Subtype")
# getGEOSuppFiles("GSE59999", baseDir = bsdir)
# GSE59999 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE59999/", recursive = T, extended = T)

# "GSE67444" - age 3 - 60 months
Samp <- BLD$GSE67444$GSE67444_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE67444"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spots"
pd$age <- as.numeric(pd$`age (months):ch1`) / 12
pd$ga <- as.numeric(pd$`gestational age (months):ch1`) / 7
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
pd$disease_subtype <- "Lead exposure"
GSE67444.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "disease_subtype", "ga", "mother's age (months):ch1", "mother's blood lead level:ch1", "sample blood lead level:ch1", "smoking:ch1", "chip", "position")]
colnames(GSE67444.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Disease_Subtype", "Gestational_Age", "Maternal_Age", "Maternal_Lead_Level_Blood", "Lead_Level_Blood", "Maternal_Smoking_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE67444", baseDir = bsdir)
# GSE67444 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE67444/", recursive = T, extended = T)

# "GSE119778" - age 14 - 43
# Samp <- BLD$GSE119778$GSE119778_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE119778"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE119778.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE119778.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE119778", baseDir = bsdir)
# GSE119778 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE119778/", recursive = T, extended = T)

# "GSE62298,GSE62303" - dnmt3a mutation cohort, not pediatric
# Samp <- BLD$GSE62298$GSE62298_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE62298"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE62298.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE62298.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE62298", baseDir = bsdir)
# GSE62298 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE62298/", recursive = T, extended = T)

# "GSE166531" - age 0 - 38
Samp <- BLD$GSE166531$GSE166531_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE166531"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disorder type:ch1` %>% gsub("BWS", "Beckwith-Wiedemann Syndrome", .) %>% gsub("SRS", "Silver-Russel Syndrome", .)
GSE166531.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE166531.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE166531", baseDir = bsdir)
# GSE166531 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE166531/", recursive = T, extended = T)

# "GSE83424" - ASD cohort, age 2 - 14
Samp <- BLD$GSE83424$GSE83424_series_matrix.txt.gz
pd <- pData(Samp)
GSE83424_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE83424_appendix.xls", skip = 1) %>% as.data.frame()
pd <- left_join(pd, GSE83424_appendix, by = c("title" = "Patient"))
pd$cohort <- "GSE83424"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`Age (years)`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Autism Spectrum Disorders"
GSE83424.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Gender", "disease_status", "chip", "position", "Characteristics", "AGRE classification")]
colnames(GSE83424.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Clinical_Diagnosis", "AGRE_Classification")
# getGEOSuppFiles("GSE83424", baseDir = bsdir)
# GSE83424 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE83424/", recursive = T, extended = T)

# "GSE104451" - Silver-Russell syndrome, range from pediatric to adult samples - age unavailable
# Samp <- BLD$GSE104451$GSE104451_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE104451"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE104451.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE104451.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE104451", baseDir = bsdir)
# GSE104451 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE104451/", recursive = T, extended = T)

# "GSE62219" - age 3 - 60 months
Samp <- BLD$GSE62219$GSE62219_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE62219"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age (months):ch1`) / 12
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE62219.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
colnames(GSE62219.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE62219", baseDir = bsdir)
# GSE62219 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE62219/", recursive = T, extended = T)

# "GSE118940" - age 0 - 4
Samp <- BLD$GSE118940$GSE118940_series_matrix.txt.gz
pd <- pData(Samp)
GSE118940_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE118940_appendix.xlsx") %>% as.data.frame()
pd$cohort <- "GSE118940"
pd$sampleid <- gsub(".* ", "", pd$title)
pd <- left_join(pd, GSE118940_appendix, by = c("sampleid" = "Child ID"))
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$Age_sample_collection)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE118940.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex", "disease_status", "Age_VABS_Assessment", "The reason of placement", "Age_enter_institution_care", "Sample_group")]
colnames(GSE118940.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Age_VABS_Assessment", "Reason_of_Placement", "Age_Enter_Institution_Care", "Sample_Group")
# getGEOSuppFiles("GSE118940", baseDir = bsdir)
# GSE118940 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE118940/", recursive = T, extended = T)

# "GSE152427,GSE152428" - age 0.5 - 47
Samp <- BLD$GSE152427$GSE152427_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE152427"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("ADNP proband", "ADNP-Related Disorder", .) %>% gsub("Control", "Healthy Control", .)
GSE152427.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "disease state subtype:ch1")]
colnames(GSE152427.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Disease_Subtype")
# getGEOSuppFiles("GSE152427", baseDir = bsdir)
# GSE152427 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE152427/", recursive = T, extended = T)

# "GSE57204,GSE57205" - age 4.33 - 12.65 - ask for exact age
Samp <- BLD$GSE57204$GSE57204_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE57204"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$sex <- NA
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "born small for gestational age"
GSE57204.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "rhgh treatment:ch1")]
colnames(GSE57204.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Treatment")
# getGEOSuppFiles("GSE57204", baseDir = bsdir)
# GSE57204 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE57204/", recursive = T, extended = T)

# "GSE44798" - age 2- 25 - ask for exact age
Samp <- BLD$GSE44798$GSE44798_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE44798"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub("_.*", "", pd$description)
pd$position <- gsub(".*_", "", pd$description)
pd$disease_status <- "Healthy Control"
pd$group <- gsub("AMS", "after maternal bariatric gastrointestinal bypass surgery", pd$`status:ch1`) %>% gsub("BMS", "before maternal bariatric gastrointestinal bypass surgery", .)
GSE44798.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "group", "chip", "position")]
colnames(GSE44798.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sample_Group", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE44798", baseDir = bsdir)
# GSE44798 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE44798/", recursive = T, extended = T)

# "GSE104812" - age 6.4 - 14.6
Samp <- BLD$GSE104812$GSE104812_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE104812"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age (y):ch1`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- NA
pd$position <- NA
pd$disease_status <- "Healthy Control"
GSE104812.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
colnames(GSE104812.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE104812", baseDir = bsdir)
# GSE104812 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE104812/", recursive = T, extended = T)

# "GSE131752" - age 9 - 59
Samp <- BLD$GSE131752$GSE131752_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE131752"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("control", "Healthy Control", .)
GSE131752.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
colnames(GSE131752.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE131752", baseDir = bsdir)
# GSE131752 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE131752/", recursive = T, extended = T)

# "GSE152084" - age 0 - 5.4
Samp <- BLD$GSE152084$GSE152084_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE152084"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("0", "F", pd$`Sex:ch1`) %>% gsub("1", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease:ch1` %>% gsub("Control", "Healthy Control", .)
GSE152084.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "ige:ch1")]
colnames(GSE152084.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "IgE")
# getGEOSuppFiles("GSE152084", baseDir = bsdir)
# GSE152084 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE152084/", recursive = T, extended = T)

# "GSE162554,GSE162560" - chemotherapy in pediatric patients - age???
Samp <- BLD$GSE162554$GSE162554_series_matrix.txt.gz
pd <- pData(Samp)
pd <- pd[pd$`tissue:ch1` == "peripheral blood", ]
pd$cohort <- "GSE162554"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Childhood Cancer"
pd$treatment <- gsub(".*patient ", "", pd$title) %>% gsub("_.*", "", .)
pd$group <- pd$`disease state:ch1`
GSE162554.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "treatment", "group", "chip", "position")]
colnames(GSE162554.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Treatment", "Sample_Group", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE162554", baseDir = bsdir)
# GSE162554 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE162554/", recursive = T, extended = T)

# "GSE165170" - pneumococcal conjugate vaccine cohort, age 12 - 24 months 
Samp <- BLD$GSE165170$GSE165170_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE165170"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- gsub(" months", "", pd$`age:ch1`) %>% as.numeric(.) / 12
pd$sex <- toupper(pd$`gender:ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
pd$timepoint <- gsub("12 months", "baseline", pd$`age:ch1`) %>% gsub("24 months", " post-vaccination", .)
GSE165170.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position", "timepoint")]
colnames(GSE165170.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Time_Point")
# getGEOSuppFiles("GSE165170", baseDir = bsdir)
# GSE165170 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE165170/", recursive = T, extended = T)

# "GSE175379" - age 9 - 21
Samp <- BLD$GSE175379$GSE175379_series_matrix.txt.gz
pd <- pData(Samp) 
pd$cohort <- "GSE175379"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- as.numeric(pd$`age (years):ch1`)
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease:ch1` %>% gsub("healthy control", "Healthy Control", .) %>% gsub("juvenile localized scleroderma", "Juvenile Localized Scleroderma", .)
GSE175379.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "ancestry:ch1", "disease_status", "chip", "position")]
colnames(GSE175379.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Ethnicity", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE175379", baseDir = bsdir)
# GSE175379 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE175379/", recursive = T, extended = T)

# "GSE32148,GSE32149" - age 3.5 - 76
Samp <- BLD$GSE32148$GSE32148_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE32148"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age (y):ch1`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("normal", "Healthy Control", .)
GSE32148.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "twin:ch1")]
colnames(GSE32148.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Twin")
# getGEOSuppFiles("GSE32148", baseDir = bsdir)
# GSE32148 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE32148/", recursive = T, extended = T)

# "GSE57107,GSE57205" - rhGH treatment cohort, age 6 - 16
Samp <- BLD$GSE57107$GSE57107_series_matrix.txt.gz
pd <- pData(Samp)
GSE57107_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE57107_appendix.xlsx") %>% as.data.frame()
pd$cohort <- "GSE57107"
pd$sampleid <- gsub(".*PBMC ", "", pd$title) %>% gsub(",.*", "", .)
pd <- left_join(pd, GSE57107_appendix, by = c("sampleid" = "Sample ID"))
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- as.numeric(pd$Age)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
GSE57107.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex", "Diagnosis", "patient:ch1", "treatment:ch1", "IGF1 baseline in ng/ml", "IGFBP3 baseline in microg/mL", "IGF1 baseline percentile", "IGFBP3 baseline percentile", "IGF1 stimulated in ng/mL", "IGFBP3 stimulated in microg/mL", "Knemometry rate in mm")]
colnames(GSE57107.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Individual", "Treatment", "IGF1_Baseline_in_ng/ml", "IGFBP3_Baseline_in_microg/mL", "IGF1_Baseline_Percentile", "IGFBP3_Baseline_Percentile", "IGF1_Stimulated_in_ng/mL", "IGFBP3_Stimulated_in_microg/mL", "Knemometry_Rate_in_mm")
# getGEOSuppFiles("GSE57107", baseDir = bsdir)
# GSE57107 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE57107/", recursive = T, extended = T)

# "GSE84624" - age 0.2 - 7
Samp <- BLD$GSE84624$GSE84624_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE84624"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("0", "F", pd$`Sex:ch1`) %>% gsub("1", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease:ch1` %>% gsub("Normal control", "Healthy Control", .) %>% gsub("disease", "Disease", .)
GSE84624.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
colnames(GSE84624.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE84624", baseDir = bsdir)
# GSE84624 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE84624/", recursive = T, extended = T)

# "GSE90117" - age 6 - 10 (no actual age)
Samp <- BLD$GSE90117$GSE90117_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE90117"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE90117.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "bmi z-score difference:ch1", "bmi z-scores:ch1", "bodyfat (%):ch1", "leptin (pg/ml):ch1", "pesticide exposure:ch1", "pon1 192 genotype:ch1", "pon1 activity:ch1")]
colnames(GSE90117.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "BMI_Zscore_Difference", "BMI_Zscore", "Body_Fat_Percentage", "Leptin_pg_per_mL", "Pesticide_Exposure", "PON1_192_Genotype", "PON1_Activity")
# getGEOSuppFiles("GSE90117", baseDir = bsdir)
# GSE90117 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE90117/", recursive = T, extended = T)

# "GSE51180" - birth and age 18
Samp <- BLD$GSE51180$GSE51180_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE51180"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spot"
pd$age <- gsub(".*infant.*", "0", pd$title) %>% gsub(".*18.*", "18", .) %>% as.numeric(.)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- gsub("Preterm.*", "Preterm Birth", pd$title) %>% gsub("Term.*", "Healthy Control", .)
GSE51180.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "onset of delivery:ch1", "pre-delivery:ch1", "type of delivery:ch1")]
colnames(GSE51180.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Onset_of_Delivery", "Pre_Delivery", "Delivery_Type")
# getGEOSuppFiles("GSE51180", baseDir = bsdir)
# GSE51180 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE51180/", recursive = T, extended = T)

# "GSE72777" - age 2 - 35
Samp <- BLD$GSE72777$GSE72777_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE72777"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("female", "F", pd$`Sex:ch1`) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE72777.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "ethnicity:ch1")]
colnames(GSE72777.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Ethnicity")
# getGEOSuppFiles("GSE72777", baseDir = bsdir)
# GSE72777 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE72777/", recursive = T, extended = T)

# "GSE119684" - age 6 - 20 (age of onset, not sample collection age)
Samp <- BLD$GSE119684$GSE119684_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE119684"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("Normal", "Healthy Control", .)
pd$family <- gsub("F.*|M.*|D.*|S.*", "", pd$title)
pd$relation <- gsub(".*from ", "", pd$title)
GSE119684.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "family", "relation", "chip", "position")]
colnames(GSE119684.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Family", "Sample_Relation", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE119684", baseDir = bsdir)
# GSE119684 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE119684/", recursive = T, extended = T)

# "GSE145233" - age 4 - 44 months
Samp <- BLD$GSE145233$GSE145233_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE145233"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age_months:ch1`) / 12
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`group:ch1` %>% gsub("Case", "Congenital Microcephaly", .) %>% gsub("Control", "Healthy Control", .)
GSE145233.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
colnames(GSE145233.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE145233", baseDir = bsdir)
# GSE145233 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE145233/", recursive = T, extended = T)

# "GSE66552" - age 2.4 - 10.7 (not matched)
Samp <- BLD$GSE66552$GSE66552_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE66552"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`group:ch1` %>% gsub("WS", "Williams Syndrome", .) %>% gsub("TD control", "Healthy Control", .) %>% gsub("Dup7", "Chromosome 7q11.23 Duplication Syndrome", .)
GSE66552.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "disease_status", "chip", "position")]
colnames(GSE66552.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE66552", baseDir = bsdir)
# GSE66552 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE66552/", recursive = T, extended = T)

# "GSE116300" - Kabuki syndrome cohort, age 1 - 11 (???)
Samp <- BLD$GSE116300$GSE116300_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE116300"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`case status:ch1` %>% gsub("control|case parent", "Healthy Control", .) %>% gsub("case", "Kabuki Syndrome", .)
GSE116300.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "chip", "position", "mutation:ch1", "variant classification:ch1")]
colnames(GSE116300.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Variant_Gene", "Variant_Classification")
# getGEOSuppFiles("GSE116300", baseDir = bsdir)
# GSE116300 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE116300/", recursive = T, extended = T)

# "GSE125367" - age 1 - 16
Samp <- BLD$GSE125367$GSE125367_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE125367"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age (years):ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$title %>% gsub("SMARCA2.*", "SMARCA2 Variant", .) %>% gsub("Control", "Healthy Control", .) %>% gsub("NCBRS.*", "Nicolaides-Baraitser Syndrome", .)
pd$variant <- pd$title %>% gsub("SMARCA.*", "VUS (variant of unknown significance)", .) %>% gsub("NCBRS.*", "pathogenic", .) %>% gsub("Control.*", NA, .)
GSE125367.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "chip", "position", "variant")]
colnames(GSE125367.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Variant_Classification")
# getGEOSuppFiles("GSE125367", baseDir = bsdir)
# GSE125367 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE125367/", recursive = T, extended = T)

# "GSE133801" - Pineoblastoma cohort - not blood
# Samp <- BLD$GSE133801$GSE133801_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE133801"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE133801.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE133801.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE133801", baseDir = bsdir)
# GSE133801 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE133801/", recursive = T, extended = T)

# "GSE60598" - age 3 - 60 months
Samp <- BLD$GSE60598$GSE60598_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE60598"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spot"
pd$age <- as.numeric(pd$`age (in months):ch1`) / 12
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
pd$ga <- as.numeric(pd$`gestational age (in months):ch1`) / 7
pd$maternal_age <- as.numeric(pd$`mother's age (in months):ch1`) / 12
GSE60598.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "ga", "smoking (0=no, 1=yes):ch1", "race:ch1", "maternal_age", "normalizedpbconc:ch1")]
colnames(GSE60598.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Gestational_Age", "Parental_Smoking", "Self_Reported_Race", "Maternal_Age", "Normalized_Lead_Concentration")
# getGEOSuppFiles("GSE60598", baseDir = bsdir)
# GSE60598 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE60598/", recursive = T, extended = T)

# "GSE154446" - DBS at birth
Samp <- BLD$GSE154446$GSE154446_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE154446"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Dried Blood Spot"
pd$age <- 2/365
pd$sex <- gsub("1", "M", pd$`sex (male= 1, female=2):ch1`) %>% gsub("2", "F", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE154446.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "gestational age:ch1", "race:ch1", "chip", "position")]
colnames(GSE154446.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Gestational_Age", "Self_Reported_Race", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE154446", baseDir = bsdir)
# GSE154446 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE154446/", recursive = T, extended = T)

# "GSE30870" - newborn samples are cord blood
# Samp <- BLD$GSE30870$GSE30870_series_matrix.txt.gz
# pd <- pData(Samp)
# pd <- pd[pd$`tissue:ch1` == "Whole blood" & pd$`cell type:ch1` == "PBMNC", ]
# pd$cohort <- "GSE30870"
# pd$tissue <- "Peripheral Blood"
# pd$celltype <- "Whole Blood"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE30870.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE30870.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE30870", baseDir = bsdir)
# GSE30870 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE30870/", recursive = T, extended = T)

# "GSE102177" - age 4 - 14
Samp <- BLD$GSE102177$GSE102177_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE102177"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`group:ch1` %>% gsub("Non-GDM", "Healthy Control", .) %>% gsub("GDM", "Maternal Gestational Diabetes Mellitus", .)
pd$relation <- "sibling"
GSE102177.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position", "sibling:ch1", "relation")]
colnames(GSE102177.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Family", "Sample_Relation")
# getGEOSuppFiles("GSE102177", baseDir = bsdir)
# GSE102177 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE102177/", recursive = T, extended = T)

# "GSE109430" - age 0 - 11
Samp <- BLD$GSE109430$GSE109430_series_matrix.txt.gz
pd <- pData(Samp)
GSE109430_appendix <- read_excel("bcchruser/Projects/Turvey/ASXL1/Reference_additional_info/GSE109430_appendix.xlsx") %>% as.data.frame()
test <- rbind(GSE109430_appendix[1:12, ], GSE109430_appendix[46:57, ], GSE109430_appendix[46:57, ])
pd$cohort <- "GSE109430"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
identical(test$Gender, pd$sex) # true
pd$age <- test$Age
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("control", "Healthy Control", .) %>% gsub(".*KD", "Kawasaki Disease", .)
pd$timepoint <- gsub("control", NA, pd$`disease state:ch1`) %>% gsub("acute.*", "acute phase 1 day before IVIG treatment (KD1)", .) %>% gsub("convalescent.*", "convalescent phase 3 weeks after IVIG treatment", .)
GSE109430.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "timepoint", "chip", "position")]
colnames(GSE109430.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Time_Point", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE109430", baseDir = bsdir)
# GSE109430 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE109430/", recursive = T, extended = T)

# "GSE113012,GSE113018" - age 3 - 10
Samp <- BLD$GSE113012$GSE113012_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE113012"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("control", "Healthy Control", .)
GSE113012.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "facial:ch1", "growth:ch1", "cns:ch1", "chip", "position")]
colnames(GSE113012.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "FASD_FASD_Score", "FASD_Growth_Score", "FASD_CNS_Score", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE113012", baseDir = bsdir)
# GSE113012 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE113012/", recursive = T, extended = T)

# "GSE111165" - age 5 - 61 (has blood, buccal, and saliva)
Samp <- BLD$GSE111165$`GSE111165-GPL21145_series_matrix.txt.gz`
pd <- pData(Samp)
pd <- pd[pd$`tissue:ch1` == "blood", ]
pd$cohort <- "GSE111165"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Intractable Epilepsy"
GSE111165_EPIC.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position", "subject:ch1")]
colnames(GSE111165_EPIC.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Individual")

Samp <- BLD$GSE111165$`GSE111165-GPL13534_series_matrix.txt.gz`
pd <- pData(Samp)
pd <- pd[pd$`tissue:ch1` == "blood", ]
pd$cohort <- "GSE111165"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- as.numeric(pd$`Sex:ch1`)
pd$sex <- gsub("Female", "F", pd$`age:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Intractable Epilepsy"
GSE111165_450k.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position", "subject:ch1")]
colnames(GSE111165_450k.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Individual")
# getGEOSuppFiles("GSE111165", baseDir = bsdir)
# GSE111165 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE111165/", recursive = T, extended = T)

# "GSE83334" - age 0 - 5 
Samp <- BLD$GSE83334$GSE83334_series_matrix.txt.gz
pd <- pData(Samp)
pd <- pd[pd$source_name_ch1 == "peripheral blood_5 year-old", ]
pd$cohort <- "GSE83334"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- 5
pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE83334.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "subject id:ch1")]
colnames(GSE83334.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Individual")
# getGEOSuppFiles("GSE83334", baseDir = bsdir)
# GSE83334 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE83334/", recursive = T, extended = T)

# "GSE116992" - age???
Samp <- BLD$GSE116992$`GSE116992-GPL13534_series_matrix.txt.gz`
pd <- pData(Samp)
pd$cohort <- "GSE116992"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("syndrome", "Syndrome", .)
pd$variant <- gsub(".*1", "ARID1B", pd$title) %>% gsub(".*3", "SMARCB1", .)
GSE116992_450.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "disease_status", "variant", "chip", "position", "source_name_ch1")]
colnames(GSE116992_450.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Disease_Status", "Variant_Gene", "Sentrix_ID", "Sentrix_Position", "Individual")

Samp <- BLD$GSE116992$`GSE116992-GPL21145_series_matrix.txt.gz`
pd <- pData(Samp)
pd$cohort <- "GSE116992"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "Whole Blood"
pd$age <- NA
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1` %>% gsub("syndrome", "Syndrome", .)
pd$variant <- gsub(".*1", "ARID1B", pd$title) %>% gsub(".*3", "SMARCB1", .) %>% gsub(".*4", "SMARCA4", .) %>% gsub(".*Nicolaides.*", "SMARCA2", .)
GSE116992_EPIC.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "disease_status", "variant", "chip", "position", "source_name_ch1")]
colnames(GSE116992_EPIC.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Disease_Status", "Variant_Gene", "Sentrix_ID", "Sentrix_Position", "Individual")
# getGEOSuppFiles("GSE116992", baseDir = bsdir)
# GSE116992 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE116992/", recursive = T, extended = T)

# "GSE48300" - infant receiving vaccine
Samp <- BLD$GSE48300$GSE48300_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE48300"
pd$tissue <- "Peripheral Blood"
pd$celltype <- "PBMC"
pd$age <- NA
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE48300.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "vaccine response:ch1")]
colnames(GSE48300.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Vaccine_Response")
# getGEOSuppFiles("GSE48300", baseDir = bsdir)
# GSE48300 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE48300/", recursive = T, extended = T)

paste(grep("\\.m", ls(), value = T), collapse = ", ")
save(GSE102177.m, GSE103657.m, GSE104812.m, GSE105018.m, GSE109430.m, GSE110128.m, GSE110828.m, GSE111165_450k.m, 
     GSE111165_EPIC.m, GSE112611.m, GSE112987.m, GSE113012.m, GSE113967.m, GSE116300.m, GSE116992_450.m, 
     GSE116992_EPIC.m, GSE118940.m, GSE119684.m, GSE124366.m, GSE125367.m, GSE131433.m, GSE131752.m, GSE132181.m, 
     GSE142512_450k.m, GSE142512_EPIC.m, GSE145233.m, GSE152027.m, GSE152084.m, GSE152204.m, 
     GSE152427.m, GSE154446.m, GSE159930.m, GSE162554.m, GSE165170.m, GSE166531.m, GSE175379.m, GSE32148.m, 
     GSE36054.m, GSE44798.m, GSE48300.m, GSE52588.m, GSE56105.m, GSE57107.m, GSE57204.m, GSE59592.m, GSE59999.m, 
     GSE60132.m, GSE60598.m, GSE62219.m, GSE64380.m, GSE64495.m, GSE66552.m, GSE67444.m, GSE72777.m, GSE73103.m, 
     GSE74432.m, GSE79257.m, GSE80283.m, GSE82273.m, GSE83334.m, GSE83424.m, GSE84624.m, GSE87571.m, GSE87648.m, 
     GSE89278.m, GSE89353.m, GSE90117.m, GSE97362.m, GSE99863.m, file = "RData/801-All_pd.RData")

allpd <- rbind.fill(GSE102177.m, GSE103657.m, GSE104812.m, GSE105018.m, GSE109430.m, GSE110128.m, GSE110828.m, GSE111165_450k.m,
                    GSE111165_EPIC.m, GSE112611.m, GSE112987.m, GSE113012.m, GSE113967.m, GSE116300.m, GSE116992_450.m, 
                    GSE116992_EPIC.m, GSE118940.m, GSE119684.m, GSE124366.m, GSE125367.m, GSE131433.m, GSE131752.m, GSE132181.m,
                    GSE142512_450k.m, GSE142512_EPIC.m, GSE145233.m, GSE152027.m, GSE152084.m, GSE152204.m, GSE152427.m, 
                    GSE154446.m, GSE159930.m, GSE162554.m, GSE165170.m, GSE166531.m, GSE175379.m, GSE32148.m, GSE36054.m, 
                    GSE44798.m, GSE48300.m, GSE52588.m, GSE56105.m, GSE57107.m, GSE57204.m, GSE59592.m, GSE59999.m, GSE60132.m, 
                    GSE60598.m, GSE62219.m, GSE64380.m, GSE64495.m, GSE66552.m, GSE67444.m, GSE72777.m, GSE73103.m, GSE74432.m, 
                    GSE79257.m, GSE80283.m, GSE82273.m, GSE83334.m, GSE83424.m, GSE84624.m, GSE87571.m, GSE87648.m, GSE89278.m, 
                    GSE89353.m, GSE90117.m, GSE97362.m, GSE99863.m)
```

```{r}
# 450k & EPIC
HM450 <- read.metharray.exp("GPL13534/", recursive = T, extended = T) # some files have problem - "GSM2363624_10003886027_R05C02"
length(list.files("GPL13534/")) / 200 %>% round(0)
test <- lapply(21:54, function(x){
    print(x)
    n <- (x*200):((x+1)*200-1)
    f <- list.files("GPL13534/")[n] %>% gsub("_Grn.*|_Red.*", "", .) %>% unique(.)
    dat <- read.metharray(paste0("GPL13534/", f), extended = T)
    return(dat)
})

HM450.n <- preprocessNoob(HM450)
HM450_pval <- detectionP(HM450)
HM450_out <- DetPBisCTest(HM450)
save(HM450_pval, HM450_out, file = "processed/HM450_pval.RData")
save(HM450.n, file = "processed/HM450_noob.RData")

EPIC <- read.metharray.exp("GPL21145/", recursive = T, extended = T)
EPIC.n <- preprocessNoob(EPIC)
EPIC_pval <- detectionP(EPIC)
EPIC_out <- DetPBisCTest(EPIC)
save(EPIC_pval, EPIC_out, file = "processed/EPIC_pval.RData")
save(EPIC.n, file = "processed/EPIC_noob.RData")

# GSE102177 - IDAT - 450k
GSE102177 <- read.metharray.exp("GSE102177/", recursive = T, extended = T)
GSE102177.n <- preprocessNoob(GSE102177)
GSE102177_pval <- detectionP(GSE102177)
GSE102177_out <- DetPBisCTest(GSE102177)
save(GSE102177_pval, GSE102177_out, file = "processed/GSE102177_pval.RData")
save(GSE102177.n, file = "processed/GSE102177_noob.RData")
dim(GSE102177)

# GSE103657 - intensity - 450k
GSE103657 <- read.metharray.exp("GSE103657/", recursive = T, extended = T)
GSE103657.n <- preprocessNoob(GSE103657)
GSE103657_pval <- detectionP(GSE103657)
GSE103657_out <- DetPBisCTest(GSE103657)
save(GSE103657_pval, GSE103657_out, file = "processed/GSE103657_pval.RData")
save(GSE103657.n, file = "processed/GSE103657_noob.RData")
dim(GSE103657)

# GSE104812 - IDAT - 450k
GSE104812 <- read.metharray.exp("GSE104812/", recursive = T, extended = T)
GSE104812.n <- preprocessNoob(GSE104812)
GSE104812_pval <- detectionP(GSE104812)
GSE104812_out <- DetPBisCTest(GSE104812)
save(GSE104812_pval, GSE104812_out, file = "processed/GSE104812_pval.RData")
save(GSE104812.n, file = "processed/GSE104812_noob.RData")
dim(GSE104812)

# GSE105018 - IDAT - 450k
GSE105018 <- read.metharray.exp("GSE105018/", recursive = T, extended = T)
GSE105018.n <- preprocessNoob(GSE105018)
GSE105018_pval <- detectionP(GSE105018)
GSE105018_out <- DetPBisCTest(GSE105018)
save(GSE105018_pval, GSE105018_out, file = "processed/GSE105018_pval.RData")
save(GSE105018.n, file = "processed/GSE105018_noob.RData")
dim(GSE105018)

# GSE109430 - IDAT - 450k
GSE109430 <- read.metharray.exp("GSE109430/", recursive = T, extended = T)
GSE109430.n <- preprocessNoob(GSE109430)
GSE109430_pval <- detectionP(GSE109430)
GSE109430_out <- DetPBisCTest(GSE109430)
save(GSE109430_pval, GSE109430_out, file = "processed/GSE109430_pval.RData")
save(GSE109430.n, file = "processed/GSE109430_noob.RData")
dim(GSE109430)

# GSE110128 - intensity - 450k
GSE110128 <- read.metharray.exp("GSE110128/", recursive = T, extended = T)
GSE110128.n <- preprocessNoob(GSE110128)
GSE110128_pval <- detectionP(GSE110128)
GSE110128_out <- DetPBisCTest(GSE110128)
save(GSE110128_pval, GSE110128_out, file = "processed/GSE110128_pval.RData")
save(GSE110128.n, file = "processed/GSE110128_noob.RData")
dim(GSE110128)

# GSE110828 - intensity - 450k
GSE110828 <- read.metharray.exp("GSE110828/", recursive = T, extended = T)
GSE110828.n <- preprocessNoob(GSE110828)
GSE110828_pval <- detectionP(GSE110828)
GSE110828_out <- DetPBisCTest(GSE110828)
save(GSE110828_pval, GSE110828_out, file = "processed/GSE110828_pval.RData")
save(GSE110828.n, file = "processed/GSE110828_noob.RData")
dim(GSE110828)

# GSE111165 - IDAT - 450k
GSE111165_450k <- read.metharray.exp("GSE111165_450k/", recursive = T, extended = T)
GSE111165_450k.n <- preprocessNoob(GSE111165_450k)
GSE111165_450k_pval <- detectionP(GSE111165_450k)
GSE111165_450k_out <- DetPBisCTest(GSE111165_450k)
save(GSE111165_450k_pval, GSE111165_450k_out, file = "processed/GSE111165_450k_pval.RData")
save(GSE111165_450k.n, file = "processed/GSE111165_450k_noob.RData")
dim(GSE111165_450k)

# GSE111165_EPIC - IDAT - EPIC
GSE111165_EPIC <- read.metharray.exp("GSE111165_EPIC/", recursive = T, extended = T)
GSE111165_EPIC.n <- preprocessNoob(GSE111165_EPIC)
GSE111165_EPIC_pval <- detectionP(GSE111165_EPIC)
GSE111165_EPIC_out <- DetPBisCTest(GSE111165_EPIC)
save(GSE111165_EPIC_pval, GSE111165_EPIC_out, file = "processed/GSE111165_EPIC_pval.RData")
save(GSE111165_EPIC.n, file = "processed/GSE111165_EPIC_noob.RData")
dim(GSE111165_EPIC)


# GSE112611 - intensity - EPIC
GSE112611 <- read.metharray.exp("GSE112611/", recursive = T, extended = T)
GSE112611.n <- preprocessNoob(GSE112611)
GSE112611_pval <- detectionP(GSE112611)
GSE112611_out <- DetPBisCTest(GSE112611)
save(GSE112611_pval, GSE112611_out, file = "processed/GSE112611_pval.RData")
save(GSE112611.n, file = "processed/GSE112611_noob.RData")
dim(GSE112611)

# GSE112987 - IDAT - 450k
GSE112987 <- read.metharray.exp("GSE112987/", recursive = T, extended = T)
GSE112987.n <- preprocessNoob(GSE112987)
GSE112987_pval <- detectionP(GSE112987)
GSE112987_out <- DetPBisCTest(GSE112987)
save(GSE112987_pval, GSE112987_out, file = "processed/GSE112987_pval.RData")
save(GSE112987.n, file = "processed/GSE112987_noob.RData")
dim(GSE112987)

# GSE113012 - IDAT - 450k
GSE113012 <- read.metharray.exp("GSE113012/", recursive = T, extended = T)
GSE113012.n <- preprocessNoob(GSE113012)
GSE113012_pval <- detectionP(GSE113012)
GSE113012_out <- DetPBisCTest(GSE113012)
save(GSE113012_pval, GSE113012_out, file = "processed/GSE113012_pval.RData")
save(GSE113012.n, file = "processed/GSE113012_noob.RData")
dim(GSE113012)

# GSE113967 - IDAT - 450k
GSE113967 <- read.metharray.exp("GSE113967/", recursive = T, extended = T)
GSE113967.n <- preprocessNoob(GSE113967)
GSE113967_pval <- detectionP(GSE113967)
GSE113967_out <- DetPBisCTest(GSE113967)
save(GSE113967_pval, GSE113967_out, file = "processed/GSE113967_pval.RData")
save(GSE113967.n, file = "processed/GSE113967_noob.RData")
dim(GSE113967)

# GSE116300 - IDAT - 450k
GSE116300 <- read.metharray.exp("GSE116300/", recursive = T, extended = T)
GSE116300.n <- preprocessNoob(GSE116300)
GSE116300_pval <- detectionP(GSE116300)
GSE116300_out <- DetPBisCTest(GSE116300)
save(GSE116300_pval, GSE116300_out, file = "processed/GSE116300_pval.RData")
save(GSE116300.n, file = "processed/GSE116300_noob.RData")
dim(GSE116300)

# GSE116992_450 - IDAT - 450k
GSE116992_450 <- read.metharray.exp("GSE116992/", recursive = T, extended = T)
GSE116992_450.n <- preprocessNoob(GSE116992)
GSE116992_450_pval <- detectionP(GSE116992)
GSE116992_450_out <- DetPBisCTest(GSE116992)
save(GSE116992_450_pval, GSE116992_450_out, file = "processed/GSE116992_450_pval.RData")
save(GSE116992_450.n, file = "processed/GSE116992_450_noob.RData")
dim(GSE116992_450)

# GSE116992_EPIC - IDAT - EPIC
GSE116992_EPIC <- read.metharray.exp("GSE116992/", recursive = T, extended = T)
GSE116992_EPIC.n <- preprocessNoob(GSE116992)
GSE116992_EPIC_pval <- detectionP(GSE116992)
GSE116992_EPIC_out <- DetPBisCTest(GSE116992)
save(GSE116992_EPIC_pval, GSE116992_EPIC_out, file = "processed/GSE116992_EPIC_pval.RData")
save(GSE116992_EPIC.n, file = "processed/GSE116992_EPIC_noob.RData")
dim(GSE116992_EPIC)

# GSE118940 - intensity - EPIC
GSE118940 <- read.metharray.exp("GSE118940/", recursive = T, extended = T)
GSE118940.n <- preprocessNoob(GSE118940)
GSE118940_pval <- detectionP(GSE118940)
GSE118940_out <- DetPBisCTest(GSE118940)
save(GSE118940_pval, GSE118940_out, file = "processed/GSE118940_pval.RData")
save(GSE118940.n, file = "processed/GSE118940_noob.RData")
dim(GSE118940)

# GSE119684 - IDAT - 450k
GSE119684 <- read.metharray.exp("GSE119684/", recursive = T, extended = T)
GSE119684.n <- preprocessNoob(GSE119684)
GSE119684_pval <- detectionP(GSE119684)
GSE119684_out <- DetPBisCTest(GSE119684)
save(GSE119684_pval, GSE119684_out, file = "processed/GSE119684_pval.RData")
save(GSE119684.n, file = "processed/GSE119684_noob.RData")
dim(GSE119684)

# GSE124366 - IDAT - 450k
GSE124366 <- read.metharray.exp("GSE124366/", recursive = T, extended = T)
GSE124366.n <- preprocessNoob(GSE124366)
GSE124366_pval <- detectionP(GSE124366)
GSE124366_out <- DetPBisCTest(GSE124366)
save(GSE124366_pval, GSE124366_out, file = "processed/GSE124366_pval.RData")
save(GSE124366.n, file = "processed/GSE124366_noob.RData")
dim(GSE124366)

# GSE125367 - IDAT - EPIC
GSE125367 <- read.metharray.exp("GSE125367/", recursive = T, extended = T)
GSE125367.n <- preprocessNoob(GSE125367)
GSE125367_pval <- detectionP(GSE125367)
GSE125367_out <- DetPBisCTest(GSE125367)
save(GSE125367_pval, GSE125367_out, file = "processed/GSE125367_pval.RData")
save(GSE125367.n, file = "processed/GSE125367_noob.RData")
dim(GSE125367)

# GSE131433 - IDAT - EPIC
GSE131433 <- read.metharray.exp("GSE131433/", recursive = T, extended = T)
GSE131433.n <- preprocessNoob(GSE131433)
GSE131433_pval <- detectionP(GSE131433)
GSE131433_out <- DetPBisCTest(GSE131433)
save(GSE131433_pval, GSE131433_out, file = "processed/GSE131433_pval.RData")
save(GSE131433.n, file = "processed/GSE131433_noob.RData")
dim(GSE131433)

# GSE131752 - intensity - EPIC
GSE131752 <- read.metharray.exp("GSE131752/", recursive = T, extended = T)
GSE131752.n <- preprocessNoob(GSE131752)
GSE131752_pval <- detectionP(GSE131752)
GSE131752_out <- DetPBisCTest(GSE131752)
save(GSE131752_pval, GSE131752_out, file = "processed/GSE131752_pval.RData")
save(GSE131752.n, file = "processed/GSE131752_noob.RData")
dim(GSE131752)

# GSE132181 - IDAT - EPIC
GSE132181 <- read.metharray.exp("GSE132181/", recursive = T, extended = T)
GSE132181.n <- preprocessNoob(GSE132181)
GSE132181_pval <- detectionP(GSE132181)
GSE132181_out <- DetPBisCTest(GSE132181)
save(GSE132181_pval, GSE132181_out, file = "processed/GSE132181_pval.RData")
save(GSE132181.n, file = "processed/GSE132181_noob.RData")
dim(GSE132181)

# GSE142512_450k - IDAT - 450k
GSE142512_450k <- read.metharray.exp("GSE142512_450k/", recursive = T, extended = T)
GSE142512_450k.n <- preprocessNoob(GSE142512_450k)
GSE142512_450k_pval <- detectionP(GSE142512_450k)
GSE142512_450k_out <- DetPBisCTest(GSE142512_450k)
save(GSE142512_450k_pval, GSE142512_450k_out, file = "processed/GSE142512_450k_pval.RData")
save(GSE142512_450k.n, file = "processed/GSE142512_450k_noob.RData")
dim(GSE142512_450k)

# GSE142512_EPIC - IDAT - EPIC
GSE142512_EPIC <- read.metharray.exp("GSE142512_EPIC/", recursive = T, extended = T)
GSE142512_EPIC.n <- preprocessNoob(GSE142512_EPIC)
GSE142512_EPIC_pval <- detectionP(GSE142512_EPIC)
GSE142512_EPIC_out <- DetPBisCTest(GSE142512_EPIC)
save(GSE142512_EPIC_pval, GSE142512_EPIC_out, file = "processed/GSE142512_EPIC_pval.RData")
save(GSE142512_EPIC.n, file = "processed/GSE142512_EPIC_noob.RData")
dim(GSE142512_EPIC)

# GSE145233 - IDAT - EPIC
GSE145233 <- read.metharray.exp("GSE145233/", recursive = T, extended = T)
GSE145233.n <- preprocessNoob(GSE145233)
GSE145233_pval <- detectionP(GSE145233)
GSE145233_out <- DetPBisCTest(GSE145233)
save(GSE145233_pval, GSE145233_out, file = "processed/GSE145233_pval.RData")
save(GSE145233.n, file = "processed/GSE145233_noob.RData")
dim(GSE145233)

# GSE152027 - intensity - 450k
GSE152027 <- read.metharray.exp("GSE152027/", recursive = T, extended = T)
GSE152027.n <- preprocessNoob(GSE152027)
GSE152027_pval <- detectionP(GSE152027)
GSE152027_out <- DetPBisCTest(GSE152027)
save(GSE152027_pval, GSE152027_out, file = "processed/GSE152027_pval.RData")
save(GSE152027.n, file = "processed/GSE152027_noob.RData")
dim(GSE152027)

# GSE152084 - intensity - 450k
GSE152084 <- read.metharray.exp("GSE152084/", recursive = T, extended = T)
GSE152084.n <- preprocessNoob(GSE152084)
GSE152084_pval <- detectionP(GSE152084)
GSE152084_out <- DetPBisCTest(GSE152084)
save(GSE152084_pval, GSE152084_out, file = "processed/GSE152084_pval.RData")
save(GSE152084.n, file = "processed/GSE152084_noob.RData")
dim(GSE152084)

# GSE152204 - IDAT - 450k
GSE152204 <- read.metharray.exp("GSE152204/", recursive = T, extended = T)
GSE152204.n <- preprocessNoob(GSE152204)
GSE152204_pval <- detectionP(GSE152204)
GSE152204_out <- DetPBisCTest(GSE152204)
save(GSE152204_pval, GSE152204_out, file = "processed/GSE152204_pval.RData")
save(GSE152204.n, file = "processed/GSE152204_noob.RData")
dim(GSE152204)

# GSE152427 - intensity - EPIC
GSE152427 <- read.metharray.exp("GSE152427/", recursive = T, extended = T)
GSE152427.n <- preprocessNoob(GSE152427)
GSE152427_pval <- detectionP(GSE152427)
GSE152427_out <- DetPBisCTest(GSE152427)
save(GSE152427_pval, GSE152427_out, file = "processed/GSE152427_pval.RData")
save(GSE152427.n, file = "processed/GSE152427_noob.RData")
dim(GSE152427)

# GSE154446 - IDAT - EPIC
GSE154446 <- read.metharray.exp("GSE154446/", recursive = T, extended = T)
GSE154446.n <- preprocessNoob(GSE154446)
GSE154446_pval <- detectionP(GSE154446)
GSE154446_out <- DetPBisCTest(GSE154446)
save(GSE154446_pval, GSE154446_out, file = "processed/GSE154446_pval.RData")
save(GSE154446.n, file = "processed/GSE154446_noob.RData")
dim(GSE154446)

# GSE159930 - IDAT - EPIC
GSE159930 <- read.metharray.exp("GSE159930/", recursive = T, extended = T)
GSE159930.n <- preprocessNoob(GSE159930)
GSE159930_pval <- detectionP(GSE159930)
GSE159930_out <- DetPBisCTest(GSE159930)
save(GSE159930_pval, GSE159930_out, file = "processed/GSE159930_pval.RData")
save(GSE159930.n, file = "processed/GSE159930_noob.RData")
dim(GSE159930)

# GSE162554 - IDAT - EPIC
GSE162554 <- read.metharray.exp("GSE162554/", recursive = T, extended = T)
GSE162554.n <- preprocessNoob(GSE162554)
GSE162554_pval <- detectionP(GSE162554)
GSE162554_out <- DetPBisCTest(GSE162554)
save(GSE162554_pval, GSE162554_out, file = "processed/GSE162554_pval.RData")
save(GSE162554.n, file = "processed/GSE162554_noob.RData")
dim(GSE162554)

# GSE165170 - IDAT - 450k
GSE165170 <- read.metharray.exp("GSE165170/", recursive = T, extended = T)
GSE165170.n <- preprocessNoob(GSE165170)
GSE165170_pval <- detectionP(GSE165170)
GSE165170_out <- DetPBisCTest(GSE165170)
save(GSE165170_pval, GSE165170_out, file = "processed/GSE165170_pval.RData")
save(GSE165170.n, file = "processed/GSE165170_noob.RData")
dim(GSE165170)

# GSE166531 - IDAT - 450k
GSE166531 <- read.metharray.exp("GSE166531/", recursive = T, extended = T)
GSE166531.n <- preprocessNoob(GSE166531)
GSE166531_pval <- detectionP(GSE166531)
GSE166531_out <- DetPBisCTest(GSE166531)
save(GSE166531_pval, GSE166531_out, file = "processed/GSE166531_pval.RData")
save(GSE166531.n, file = "processed/GSE166531_noob.RData")
dim(GSE166531)

# GSE175379 - IDAT - EPIC
GSE175379 <- read.metharray.exp("GSE175379/", recursive = T, extended = T)
GSE175379.n <- preprocessNoob(GSE175379)
GSE175379_pval <- detectionP(GSE175379)
GSE175379_out <- DetPBisCTest(GSE175379)
save(GSE175379_pval, GSE175379_out, file = "processed/GSE175379_pval.RData")
save(GSE175379.n, file = "processed/GSE175379_noob.RData")
dim(GSE175379)

# GSE32148 - intensity - 450k
GSE32148 <- read.metharray.exp("GSE32148/", recursive = T, extended = T)
GSE32148.n <- preprocessNoob(GSE32148)
GSE32148_pval <- detectionP(GSE32148)
GSE32148_out <- DetPBisCTest(GSE32148)
save(GSE32148_pval, GSE32148_out, file = "processed/GSE32148_pval.RData")
save(GSE32148.n, file = "processed/GSE32148_noob.RData")
dim(GSE32148)

# GSE36054 - intensity - 450k
GSE36054 <- read.metharray.exp("GSE36054/", recursive = T, extended = T)
GSE36054.n <- preprocessNoob(GSE36054)
GSE36054_pval <- detectionP(GSE36054)
GSE36054_out <- DetPBisCTest(GSE36054)
save(GSE36054_pval, GSE36054_out, file = "processed/GSE36054_pval.RData")
save(GSE36054.n, file = "processed/GSE36054_noob.RData")
dim(GSE36054)

# GSE44798 - intensity - 450k
GSE44798 <- read.metharray.exp("GSE44798/", recursive = T, extended = T)
GSE44798.n <- preprocessNoob(GSE44798)
GSE44798_pval <- detectionP(GSE44798)
GSE44798_out <- DetPBisCTest(GSE44798)
save(GSE44798_pval, GSE44798_out, file = "processed/GSE44798_pval.RData")
save(GSE44798.n, file = "processed/GSE44798_noob.RData")
dim(GSE44798)

# GSE48300 - intensity - 450k
GSE48300 <- read.metharray.exp("GSE48300/", recursive = T, extended = T)
GSE48300.n <- preprocessNoob(GSE48300)
GSE48300_pval <- detectionP(GSE48300)
GSE48300_out <- DetPBisCTest(GSE48300)
save(GSE48300_pval, GSE48300_out, file = "processed/GSE48300_pval.RData")
save(GSE48300.n, file = "processed/GSE48300_noob.RData")
dim(GSE48300)

# GSE52588 - intensity - 450k
GSE52588 <- read.metharray.exp("GSE52588/", recursive = T, extended = T)
GSE52588.n <- preprocessNoob(GSE52588)
GSE52588_pval <- detectionP(GSE52588)
GSE52588_out <- DetPBisCTest(GSE52588)
save(GSE52588_pval, GSE52588_out, file = "processed/GSE52588_pval.RData")
save(GSE52588.n, file = "processed/GSE52588_noob.RData")
dim(GSE52588)

# GSE56105 - intensity - 450k
GSE56105 <- read.metharray.exp("GSE56105/", recursive = T, extended = T)
GSE56105.n <- preprocessNoob(GSE56105)
GSE56105_pval <- detectionP(GSE56105)
GSE56105_out <- DetPBisCTest(GSE56105)
save(GSE56105_pval, GSE56105_out, file = "processed/GSE56105_pval.RData")
save(GSE56105.n, file = "processed/GSE56105_noob.RData")
dim(GSE56105)

# GSE57107 - intensity - 450k
GSE57107 <- read.metharray.exp("GSE57107/", recursive = T, extended = T)
GSE57107.n <- preprocessNoob(GSE57107)
GSE57107_pval <- detectionP(GSE57107)
GSE57107_out <- DetPBisCTest(GSE57107)
save(GSE57107_pval, GSE57107_out, file = "processed/GSE57107_pval.RData")
save(GSE57107.n, file = "processed/GSE57107_noob.RData")
dim(GSE57107)

# GSE57204 - intensity - 450k
GSE57204 <- read.metharray.exp("GSE57204/", recursive = T, extended = T)
GSE57204.n <- preprocessNoob(GSE57204)
GSE57204_pval <- detectionP(GSE57204)
GSE57204_out <- DetPBisCTest(GSE57204)
save(GSE57204_pval, GSE57204_out, file = "processed/GSE57204_pval.RData")
save(GSE57204.n, file = "processed/GSE57204_noob.RData")
dim(GSE57204)

# GSE59592 - intensity - 450k
GSE59592 <- read.metharray.exp("GSE59592/", recursive = T, extended = T)
GSE59592.n <- preprocessNoob(GSE59592)
GSE59592_pval <- detectionP(GSE59592)
GSE59592_out <- DetPBisCTest(GSE59592)
save(GSE59592_pval, GSE59592_out, file = "processed/GSE59592_pval.RData")
save(GSE59592.n, file = "processed/GSE59592_noob.RData")
dim(GSE59592)

# GSE59999 - intensity - 450k
GSE59999 <- read.metharray.exp("GSE59999/", recursive = T, extended = T)
GSE59999.n <- preprocessNoob(GSE59999)
GSE59999_pval <- detectionP(GSE59999)
GSE59999_out <- DetPBisCTest(GSE59999)
save(GSE59999_pval, GSE59999_out, file = "processed/GSE59999_pval.RData")
save(GSE59999.n, file = "processed/GSE59999_noob.RData")
dim(GSE59999)

# GSE60132 - intensity - 450k
GSE60132 <- read.metharray.exp("GSE60132/", recursive = T, extended = T)
GSE60132.n <- preprocessNoob(GSE60132)
GSE60132_pval <- detectionP(GSE60132)
GSE60132_out <- DetPBisCTest(GSE60132)
save(GSE60132_pval, GSE60132_out, file = "processed/GSE60132_pval.RData")
save(GSE60132.n, file = "processed/GSE60132_noob.RData")
dim(GSE60132)

# GSE60598 - intensity - 450k
GSE60598 <- read.metharray.exp("GSE60598/", recursive = T, extended = T)
GSE60598.n <- preprocessNoob(GSE60598)
GSE60598_pval <- detectionP(GSE60598)
GSE60598_out <- DetPBisCTest(GSE60598)
save(GSE60598_pval, GSE60598_out, file = "processed/GSE60598_pval.RData")
save(GSE60598.n, file = "processed/GSE60598_noob.RData")
dim(GSE60598)

# GSE62219 - IDAT - 450k
GSE62219 <- read.metharray.exp("GSE62219/", recursive = T, extended = T)
GSE62219.n <- preprocessNoob(GSE62219)
GSE62219_pval <- detectionP(GSE62219)
GSE62219_out <- DetPBisCTest(GSE62219)
save(GSE62219_pval, GSE62219_out, file = "processed/GSE62219_pval.RData")
save(GSE62219.n, file = "processed/GSE62219_noob.RData")
dim(GSE62219)

# GSE64380 - intensity - 450k
GSE64380 <- read.metharray.exp("GSE64380/", recursive = T, extended = T)
GSE64380.n <- preprocessNoob(GSE64380)
GSE64380_pval <- detectionP(GSE64380)
GSE64380_out <- DetPBisCTest(GSE64380)
save(GSE64380_pval, GSE64380_out, file = "processed/GSE64380_pval.RData")
save(GSE64380.n, file = "processed/GSE64380_noob.RData")
dim(GSE64380)

# GSE64495 - intensity - 450k
GSE64495 <- read.metharray.exp("GSE64495/", recursive = T, extended = T)
GSE64495.n <- preprocessNoob(GSE64495)
GSE64495_pval <- detectionP(GSE64495)
GSE64495_out <- DetPBisCTest(GSE64495)
save(GSE64495_pval, GSE64495_out, file = "processed/GSE64495_pval.RData")
save(GSE64495.n, file = "processed/GSE64495_noob.RData")
dim(GSE64495)

# GSE66552 - IDAT - 450k
GSE66552 <- read.metharray.exp("GSE66552/", recursive = T, extended = T)
GSE66552.n <- preprocessNoob(GSE66552)
GSE66552_pval <- detectionP(GSE66552)
GSE66552_out <- DetPBisCTest(GSE66552)
save(GSE66552_pval, GSE66552_out, file = "processed/GSE66552_pval.RData")
save(GSE66552.n, file = "processed/GSE66552_noob.RData")
dim(GSE66552)

# GSE67444 - IDAT - 450k
GSE67444 <- read.metharray.exp("GSE67444/", recursive = T, extended = T)
GSE67444.n <- preprocessNoob(GSE67444)
GSE67444_pval <- detectionP(GSE67444)
GSE67444_out <- DetPBisCTest(GSE67444)
save(GSE67444_pval, GSE67444_out, file = "processed/GSE67444_pval.RData")
save(GSE67444.n, file = "processed/GSE67444_noob.RData")
dim(GSE67444)

# GSE72777 - intensity - 450k
GSE72777 <- read.metharray.exp("GSE72777/", recursive = T, extended = T)
GSE72777.n <- preprocessNoob(GSE72777)
GSE72777_pval <- detectionP(GSE72777)
GSE72777_out <- DetPBisCTest(GSE72777)
save(GSE72777_pval, GSE72777_out, file = "processed/GSE72777_pval.RData")
save(GSE72777.n, file = "processed/GSE72777_noob.RData")
dim(GSE72777)

# GSE73103 - IDAT - 450k
GSE73103 <- read.metharray.exp("GSE73103/", recursive = T, extended = T)
GSE73103.n <- preprocessNoob(GSE73103)
GSE73103_pval <- detectionP(GSE73103)
GSE73103_out <- DetPBisCTest(GSE73103)
save(GSE73103_pval, GSE73103_out, file = "processed/GSE73103_pval.RData")
save(GSE73103.n, file = "processed/GSE73103_noob.RData")
dim(GSE73103)

# GSE74432 - IDAT - 450k
GSE74432 <- read.metharray.exp("GSE74432/", recursive = T, extended = T)
GSE74432.n <- preprocessNoob(GSE74432)
GSE74432_pval <- detectionP(GSE74432)
GSE74432_out <- DetPBisCTest(GSE74432)
save(GSE74432_pval, GSE74432_out, file = "processed/GSE74432_pval.RData")
save(GSE74432.n, file = "processed/GSE74432_noob.RData")
dim(GSE74432)

# GSE79257 - IDAT - 450k
GSE79257 <- read.metharray.exp("GSE79257/", recursive = T, extended = T)
GSE79257.n <- preprocessNoob(GSE79257)
GSE79257_pval <- detectionP(GSE79257)
GSE79257_out <- DetPBisCTest(GSE79257)
save(GSE79257_pval, GSE79257_out, file = "processed/GSE79257_pval.RData")
save(GSE79257.n, file = "processed/GSE79257_noob.RData")
dim(GSE79257)

# GSE80283 - intensity - 450k
GSE80283 <- read.metharray.exp("GSE80283/", recursive = T, extended = T)
GSE80283.n <- preprocessNoob(GSE80283)
GSE80283_pval <- detectionP(GSE80283)
GSE80283_out <- DetPBisCTest(GSE80283)
save(GSE80283_pval, GSE80283_out, file = "processed/GSE80283_pval.RData")
save(GSE80283.n, file = "processed/GSE80283_noob.RData")
dim(GSE80283)

# GSE82273 - intensity - 450k
GSE82273 <- read.metharray.exp("GSE82273/", recursive = T, extended = T)
GSE82273.n <- preprocessNoob(GSE82273)
GSE82273_pval <- detectionP(GSE82273)
GSE82273_out <- DetPBisCTest(GSE82273)
save(GSE82273_pval, GSE82273_out, file = "processed/GSE82273_pval.RData")
save(GSE82273.n, file = "processed/GSE82273_noob.RData")
dim(GSE82273)

# GSE83334 - intensity - 450k
GSE83334 <- read.metharray.exp("GSE83334/", recursive = T, extended = T)
GSE83334.n <- preprocessNoob(GSE83334)
GSE83334_pval <- detectionP(GSE83334)
GSE83334_out <- DetPBisCTest(GSE83334)
save(GSE83334_pval, GSE83334_out, file = "processed/GSE83334_pval.RData")
save(GSE83334.n, file = "processed/GSE83334_noob.RData")
dim(GSE83334)

# GSE83424 - IDAT - 450k
GSE83424 <- read.metharray.exp("GSE83424/", recursive = T, extended = T)
GSE83424.n <- preprocessNoob(GSE83424)
GSE83424_pval <- detectionP(GSE83424)
GSE83424_out <- DetPBisCTest(GSE83424)
save(GSE83424_pval, GSE83424_out, file = "processed/GSE83424_pval.RData")
save(GSE83424.n, file = "processed/GSE83424_noob.RData")
dim(GSE83424)

# GSE84624 - intensity - 450k
GSE84624 <- read.metharray.exp("GSE84624/", recursive = T, extended = T)
GSE84624.n <- preprocessNoob(GSE84624)
GSE84624_pval <- detectionP(GSE84624)
GSE84624_out <- DetPBisCTest(GSE84624)
save(GSE84624_pval, GSE84624_out, file = "processed/GSE84624_pval.RData")
save(GSE84624.n, file = "processed/GSE84624_noob.RData")
dim(GSE84624)

# GSE87571 - IDAT - 450k
GSE87571 <- read.metharray.exp("GSE87571/", recursive = T, extended = T)
GSE87571.n <- preprocessNoob(GSE87571)
GSE87571_pval <- detectionP(GSE87571)
GSE87571_out <- DetPBisCTest(GSE87571)
save(GSE87571_pval, GSE87571_out, file = "processed/GSE87571_pval.RData")
save(GSE87571.n, file = "processed/GSE87571_noob.RData")
dim(GSE87571)

# GSE87648 - IDAT - 450k
GSE87648 <- read.metharray.exp("GSE87648/", recursive = T, extended = T)
GSE87648.n <- preprocessNoob(GSE87648)
GSE87648_pval <- detectionP(GSE87648)
GSE87648_out <- DetPBisCTest(GSE87648)
save(GSE87648_pval, GSE87648_out, file = "processed/GSE87648_pval.RData")
save(GSE87648.n, file = "processed/GSE87648_noob.RData")
dim(GSE87648)

# GSE89278 - IDAT - 450k
GSE89278 <- read.metharray.exp("GSE89278/", recursive = T, extended = T)
GSE89278.n <- preprocessNoob(GSE89278)
GSE89278_pval <- detectionP(GSE89278)
GSE89278_out <- DetPBisCTest(GSE89278)
save(GSE89278_pval, GSE89278_out, file = "processed/GSE89278_pval.RData")
save(GSE89278.n, file = "processed/GSE89278_noob.RData")
dim(GSE89278)

# GSE89353 - intensity - 450k
GSE89353 <- read.metharray.exp("GSE89353/", recursive = T, extended = T)
GSE89353.n <- preprocessNoob(GSE89353)
GSE89353_pval <- detectionP(GSE89353)
GSE89353_out <- DetPBisCTest(GSE89353)
save(GSE89353_pval, GSE89353_out, file = "processed/GSE89353_pval.RData")
save(GSE89353.n, file = "processed/GSE89353_noob.RData")
dim(GSE89353)

# GSE90117 - intensity - 450k
GSE90117 <- read.metharray.exp("GSE90117/", recursive = T, extended = T)
GSE90117.n <- preprocessNoob(GSE90117)
GSE90117_pval <- detectionP(GSE90117)
GSE90117_out <- DetPBisCTest(GSE90117)
save(GSE90117_pval, GSE90117_out, file = "processed/GSE90117_pval.RData")
save(GSE90117.n, file = "processed/GSE90117_noob.RData")
dim(GSE90117)

# GSE97362 - IDAT - 450k
GSE97362 <- read.metharray.exp("GSE97362/", recursive = T, extended = T)
GSE97362.n <- preprocessNoob(GSE97362)
GSE97362_pval <- detectionP(GSE97362)
GSE97362_out <- DetPBisCTest(GSE97362)
save(GSE97362_pval, GSE97362_out, file = "processed/GSE97362_pval.RData")
save(GSE97362.n, file = "processed/GSE97362_noob.RData")
dim(GSE97362)

# GSE99863 - IDAT - 450k
GSE99863 <- read.metharray.exp("GSE99863/", recursive = T, extended = T)
GSE99863.n <- preprocessNoob(GSE99863)
GSE99863_pval <- detectionP(GSE99863)
GSE99863_out <- DetPBisCTest(GSE99863)
save(GSE99863_pval, GSE99863_out, file = "processed/GSE99863_pval.RData")
save(GSE99863.n, file = "processed/GSE99863_noob.RData")
dim(GSE99863)



```

### Cell Type Prediction 
```{r}
HM450.minfi <- estimateCellCounts2(HM450, processMethod = "preprocessNoob",
                                   probeSelect = "both",
                                   referencePlatform = "IlluminaHumanMethylation450k",
                                   IDOLOptimizedCpGs = NULL)
# HM450.idol <- estimateCellCounts2(HM450, processMethod = "preprocessNoob",
#                                   probeSelect = "IDOL",
#                                   referencePlatform = "IlluminaHumanMethylation450k",
#                                   IDOLOptimizedCpGs = NULL)
EPIC.ct <- estimateCellCounts2(EPIC, processMethod = "preprocessNoob",
                               probeSelect = "both",
                               referencePlatform = "IlluminaHumanMethylationEPIC",
                               referenceset = "FlowSorted.Blood.EPIC",
                               IDOLOptimizedCpGs = NULL)
```

## Preprocessing

### Next step

```{r}
meta <- pData(MLset.CC)
dim(meta)
target <- paste0(meta$GEO_Accession, "_", meta$IDAT_ID)
target[meta$Cohort == "ASXL1"] <- meta$IDAT_ID[meta$Cohort == "ASXL1"]
target_450 <- target[meta$Platform_ID != "GPL21145"] %>% as.data.frame()
target_EPIC <- target[meta$Platform_ID == "GPL21145"] %>% as.data.frame()
colnames(target_450) <- "Basename"
colnames(target_EPIC) <- "Basename"
save(target_450, target_EPIC, file = "Target_files.RData")
```


```{r}
load("RData/801-All_pd.RData")
load("RData/EPIC_FN.RData")
load("RData/HM450_FN.RData")
load("RData/EPIC_pval.RData")
load("RData/HM450_pval.RData")
load("~/Projects/ASXL1/RData/10-Biggs_patient_raw.RData")

ASXL1_bld_samp <- c("203077630141_R01C01", "203077630141_R06C01", "203077630141_R07C01")

pd1 <- read.csv("Turvey_Samplesheet.csv", stringsAsFactors = F, skip = 7)
pd1 <- pd1[pd1$Project == "Turvey", ]
pd1$Sample_ID <- pd1$Sample_Name
pd1$Individual <- gsub("_.*", "", pd1$Sample_Name)
pd1$Sex <- rep("F", nrow(pd1))
pd1$Ethnicity <- rep("Taiwanese_Canadian", nrow(pd1))
pd1$Cell_Type <- pd1$Sample_Name %>% gsub("Pat_|Ctrl_|_[12].*", "", .) %>% gsub("WB", "Whole Blood", .)
pd1$Tissue <- gsub("PBMC|Whole Blood", "Peripheral Blood", pd1$Cell_Type)
pd1$Age <- rep(11, nrow(pd1))
pd1[pd1$Individual == "Ctrl", "Age"] <- 14
pd1$Cohort <- "ASXL1"
pd1$Disease_Status <- gsub("Ctrl", "Healthy Control", pd1$Individual) %>% gsub("Pat", "Cutaneous granulomatous", .)
# pd1$Female <- pd1$Sex %>% 
#                 gsub("M", "0", .) %>%
#                 gsub("F", "1", .) # horvath anno needs age, female, and tissue type
pd1$Sentrix_ID <- as.character(pd1$Sentrix_ID)
pd1$Sentrix_Position <- as.character(pd1$Sentrix_Position)
ASXL1.m <- pd1[, c("Cohort", "Sample_ID", "Individual", "Sex", "Ethnicity", "Tissue", "Cell_Type", "Age", "Disease_Status", "Sentrix_ID", "Sentrix_Position")]

allpd <- rbind.fill(GSE102177.m, GSE103657.m, GSE104812.m, GSE105018.m, GSE109430.m, GSE110128.m, 
                    GSE110828.m, GSE111165_450k.m, GSE111165_EPIC.m, GSE112611.m, GSE112987.m, GSE113012.m, 
                    GSE113967.m, GSE116300.m, GSE116992_450.m, GSE116992_EPIC.m, GSE118940.m, GSE119684.m, 
                    GSE124366.m, GSE125367.m, GSE131433.m, GSE131752.m, GSE132181.m, GSE142512_450k.m, GSE142512_EPIC.m, 
                    GSE145233.m, GSE152027.m, GSE152084.m, GSE152204.m, GSE152427.m, GSE154446.m, GSE159930.m, 
                    GSE162554.m, GSE165170.m, GSE166531.m, GSE175379.m, GSE32148.m, GSE36054.m, GSE44798.m, 
                    GSE48300.m, GSE52588.m, GSE56105.m, GSE57107.m, GSE57204.m, GSE59592.m, GSE59999.m, GSE60132.m, 
                    GSE60598.m, GSE62219.m, GSE64380.m, GSE64495.m, GSE66552.m, GSE67444.m, GSE72777.m, GSE73103.m, 
                    GSE74432.m, GSE79257.m, GSE80283.m, GSE82273.m, GSE83334.m, GSE83424.m, GSE84624.m, GSE87571.m, 
                    GSE87648.m, GSE89278.m, GSE89353.m, GSE90117.m, GSE97362.m, GSE99863.m, ASXL1.m)
allpd <- allpd[!is.na(allpd$Cohort), ]
allpd$IDAT_ID <- paste0(allpd$Sentrix_ID, "_", allpd$Sentrix_Position)
# allpd$Twin <- allpd$Twin %>% gsub("0|NA", NA, .)
# allpd$Disease_Stage_Diagnosis <- allpd$Disease_Stage_Diagnosis %>% gsub("NA", NA, .)
# allpd$Disease_Stage_Followup <- allpd$Disease_Stage_Followup %>% gsub("NA", NA, .)

CpG <- Reduce(intersect, list(rownames(HM450.n), 
                              rownames(EPIC.n)))
allbt <- cbind(getBeta(HM450.n)[CpG, ], getBeta(EPIC.n)[CpG, ])
allsex <- rbind(getSex(HM450.n), getSex(EPIC.n)) %>% as.data.frame()

geo_accession <- gsub("_.*", "", colnames(allbt))
idat_id <- gsub("GSM[0-9]{7}_", "", colnames(allbt))
dim(allbt)

keep <- c(which(allpd$GEO_Accession %in% geo_accession), which(allpd$IDAT_ID %in% idat_id))
allpd <- allpd[unique(keep), ] # have duplicated rows???
test <- allpd[duplicated(allpd), ]
allpd <- distinct(allpd)
rownames(allpd) <- paste0(allpd$GEO_Accession, "_", allpd$IDAT_ID)
#allpd <- allpd[sort(rownames(allpd)), ]
allpd <- allpd[allpd$Cohort != "GSE104812", ] # does not have sentrix info

nm <- gsub("_.*", "", colnames(allbt))
test <- allbt[, nm %in% allpd$GEO_Accession[allpd$Cohort %in% c("GSE159930")]]
identical(nm[nm %in% allpd$GEO_Accession[allpd$Cohort %in% c("GSE159930")]], allpd$GEO_Accession[allpd$Cohort == "GSE159930"])
allpd$IDAT_ID[allpd$Cohort == "GSE159930"] <- gsub("GSM[0-9]{7}_", "", colnames(test))
allpd$Sentrix_ID[allpd$Cohort == "GSE159930"] <- gsub("GSM[0-9]{7}_", "", colnames(test)) %>% gsub("_.*", "", .)
allpd$Sentrix_Position[allpd$Cohort == "GSE159930"] <- gsub("GSM[0-9]{7}_", "", colnames(test)) %>% gsub(".*_", "", .)
rownames(allpd) <- allpd$IDAT_ID # GSE79257/GSE66552/GSE67444 & GSE113967/GSE97362/GSE66552 have duplicated samples!!!! Identical samples under separate annotations
test <- allpd[duplicated(allpd$IDAT_ID), ]
test <- allpd[allpd$IDAT_ID %in% allpd$IDAT_ID[duplicated(allpd$IDAT_ID)], ]

# Remove duplicated samples and make sure to keep all info
out <- c("GSM2089923_9266441185_R04C01", "GSM2089924_9266441185_R01C01", "GSM2089925_9266441185_R02C02", "GSM2089926_9266441186_R04C02", "GSM2089927_9285451032_R01C02", "GSM2089928_9297949059_R03C02", 
         "GSM2089929_9297949059_R05C02", "GSM2089932_9297953013_R06C01", "GSM2089933_9297953014_R06C02", "GSM2089935_9297953046_R05C01", "GSM2089936_9297953047_R01C01", "GSM2089938_9297953048_R04C02", 
         "GSM2089939_9266441185_R02C01", "GSM2089940_9266441186_R06C02", "GSM2089943_9285451032_R05C02", "GSM2089947_9285451033_R02C01", "GSM2089949_9297949058_R01C01", "GSM2089950_9297949058_R03C01", 
         "GSM2089952_9297949059_R04C02", "GSM2089954_9297953014_R03C02", "GSM2089955_9297953014_R04C02", "GSM2089959_9297953046_R01C01", "GSM2089961_9297953046_R02C02", "GSM2089962_9297953046_R06C01", 
         "GSM2089963_9297953048_R02C02", "GSM2089964_9297953048_R05C02", "GSM1624830_9376525009_R03C02", "GSM1624831_9376525033_R01C01", "GSM3124911_200069280110_R04C01", "GSM3124912_200069280110_R06C02", 
         "GSM3124913_200069280056_R04C01", "GSM3124914_200069280056_R03C02", "GSM3124915_200069280056_R06C02", "GSM3124916_200053730212_R03C02", "GSM2562859_200053730212_R03C02", "GSM2562864_200069280102_R06C02", 
         "GSM2562865_200053730213_R01C01", "GSM2562866_200053730213_R04C01", "GSM2562867_200053730213_R03C02", "GSM2562868_200053730213_R06C02", "GSM2562869_200069280068_R01C01", "GSM2562870_200069280068_R04C01",
         "GSM2562871_200069280068_R03C02", "GSM2562872_200069280068_R06C02", "GSM2562876_200053730211_R03C02", "GSM2562877_200053730211_R06C02", "GSM2562878_200053730149_R01C01", "GSM2562879_200053730149_R03C02", 
         "GSM2562880_200053730149_R06C02", "GSM3124931_200069280111_R06C02", "GSM2562886_200053730181_R04C01", "GSM2562887_200053730181_R06C02", "GSM2562890_200069280005_R03C02", "GSM3124935_200069280100_R01C01", 
         "GSM3124936_200069280070_R01C01", "GSM3124937_200069280070_R05C02", "GSM2562898_200069280030_R01C01", "GSM3124939_200053730139_R03C02", "GSM3124940_200053730139_R06C02", "GSM3124941_200069280013_R06C02", 
         "GSM3124942_200069280057_R01C01", "GSM3124943_200069280019_R04C01", "GSM3124944_200053730148_R04C01", "GSM3124945_200069280109_R01C01") 
# not sure whether GSM3124916_200053730212_R03C02 is age 6 or 19 - horvath age is 7, assume it's 6
allpd <- allpd[!rownames(allpd) %in% out, ]
allsex <- allsex[!rownames(allsex) %in% out, ]

allpd["GSM1646913_9266441185_R04C01", "Age"] <- 2/365
allpd["GSM1646915_9266441185_R01C01", "Age"] <- 2/365
allpd["GSM1646884_9266441185_R02C02", "Age"] <- 2/365
allpd["GSM1646910_9266441186_R04C02", "Age"] <- 2/365
allpd["GSM1646886_9285451032_R01C02", "Age"] <- 2/365
allpd["GSM1646894_9297949059_R03C02", "Age"] <- 2/365
allpd["GSM1646893_9297949059_R05C02", "Age"] <- 2/365
allpd["GSM1646899_9297953013_R06C01", "Age"] <- 2/365
allpd["GSM1646907_9297953014_R06C02", "Age"] <- 2/365
allpd["GSM1646918_9297953046_R05C01", "Age"] <- 2/365
allpd["GSM1646906_9297953047_R01C01", "Age"] <- 2/365
allpd["GSM1646887_9297953048_R04C02", "Age"] <- 2/365
allpd["GSM1646914_9266441185_R02C01", "Age"] <- 2/365
allpd["GSM1646892_9266441186_R06C02", "Age"] <- 2/365
allpd["GSM1646896_9285451032_R05C02", "Age"] <- 2/365
allpd["GSM1646890_9285451033_R02C01", "Age"] <- 2/365
allpd["GSM1646916_9297949058_R01C01", "Age"] <- 2/365
allpd["GSM1646889_9297949058_R03C01", "Age"] <- 2/365
allpd["GSM1646895_9297949059_R04C02", "Age"] <- 2/365
allpd["GSM1646904_9297953014_R03C02", "Age"] <- 2/365
allpd["GSM1646903_9297953014_R04C02", "Age"] <- 2/365
allpd["GSM1646891_9297953046_R01C01", "Age"] <- 2/365
allpd["GSM1646902_9297953046_R02C02", "Age"] <- 2/365
allpd["GSM1646917_9297953046_R06C01", "Age"] <- 2/365
allpd["GSM1646901_9297953048_R02C02", "Age"] <- 2/365
allpd["GSM1646888_9297953048_R05C02", "Age"] <- 2/365
rownames(allpd) <- allpd$IDAT_ID 

allbt <- allbt[, !colnames(allbt) %in% out]
colnames(allbt) <- gsub("GSM[0-9]{7}_", "", colnames(allbt))
rownames(allsex) <- gsub("GSM[0-9]{7}_", "", rownames(allsex))

allbt <- allbt[, colnames(allbt) %in% rownames(allpd)]
allpd <- allpd[colnames(allbt), ]
allsex <- allsex[colnames(allbt), ]
allpd$Estimated_Sex <- allsex$predictedSex

identical(rownames(allpd), colnames(allbt))
table(allpd$Cohort)

PedRef <- new("MethyLumiSet", 
              betas = as.matrix(allbt), 
              phenoData = as(allpd, "AnnotatedDataFrame"), 
              featureData = TB@featureData[rownames(allbt)], 
              experimentData = TB@experimentData, 
              .__classVersion__ = TB@.__classVersion__)

save(allbt, allpd, PedRef, file = "RData/02-All_GEO_reference.RData")
```


### Sample / Probe QC

```{r}
load("RData/02-All_GEO_reference.RData")

# Sample QC
MLset <- PedRef

# Remove Buccal samples
beta <- betas(MLset)
meta <- pData(MLset)
table(meta$Cohort)
meta$Cell_Type[meta$Cell_Type == "Dried Blood Spot"] <- "Dried Blood Spots"
pData(MLset) <- meta
MLset <- MLset[, meta$Tissue == "Peripheral Blood"]

#out1 <- DetPBisCTest(MLset)
# Sample(s) GSM3852526_200673760053_R01C01 have average detection p-values greater than 0.01!
# Sample(s) GSM3267119_202226320152_R02C01, GSM3780216_202818860022_R03C01, GSM3780268_202818860100_R07C01, GSM3852623_200673610008_R03C01 have bisulfite conversion rate lower than 75!
# Sample(s) GSM1522958_8221932108_R01C01, GSM1522959_8221932123_R02C02, GSM1522960_8221932108_R03C01, GSM1522962_8221932128_R05C02, GSM1522963_8221932118_R05C02, GSM1522968_8221932108_R06C02, GSM1522980_8221932132_R02C02, GSM1522982_8221932099_R02C01, GSM1522983_8221932132_R01C01, GSM1522984_8221932118_R06C02, GSM1522989_8221932099_R06C02, GSM1522996_8221932128_R04C02, GSM1523004_8221932099_R04C01, GSM1523010_8221932123_R05C02, GSM1523011_8221932108_R04C02, GSM1523013_8221932128_R02C01, GSM1523014_8221932122_R02C01, GSM1523016_8221932128_R01C02, GSM2089945_9285451032_R06C02, GSM2089953_9297949059_R06C02, GSM2090004_9611519098_R06C01, GSM2090026_9611519076_R04C02, GSM2090056_9611519080_R03C02, GSM2337461_9979553072_R05C01, GSM2337570_9878820094_R06C01, GSM2656366_9970497017_R06C01, GSM3380800_6929726053_R05C01, GSM3380802_6929726054_R01C02, GSM3380804_6929726054_R03C02, GSM3530006_9274651149_R05C02 have average detection p-values greater than 0.01!

out1 <- c("GSM3852526_200673760053_R01C01", "GSM3267119_202226320152_R02C01", "GSM3780216_202818860022_R03C01", "GSM3780268_202818860100_R07C01", "GSM3852623_200673610008_R03C01", "GSM1522958_8221932108_R01C01", "GSM1522959_8221932123_R02C02", "GSM1522960_8221932108_R03C01", "GSM1522962_8221932128_R05C02", "GSM1522963_8221932118_R05C02", "GSM1522968_8221932108_R06C02", "GSM1522980_8221932132_R02C02", "GSM1522982_8221932099_R02C01", "GSM1522983_8221932132_R01C01", "GSM1522984_8221932118_R06C02", "GSM1522989_8221932099_R06C02", "GSM1522996_8221932128_R04C02", "GSM1523004_8221932099_R04C01", "GSM1523010_8221932123_R05C02", "GSM1523011_8221932108_R04C02", "GSM1523013_8221932128_R02C01", "GSM1523014_8221932122_R02C01", "GSM1523016_8221932128_R01C02", "GSM2089945_9285451032_R06C02", "GSM2089953_9297949059_R06C02", "GSM2090004_9611519098_R06C01", "GSM2090026_9611519076_R04C02", "GSM2090056_9611519080_R03C02", "GSM2337461_9979553072_R05C01", "GSM2337570_9878820094_R06C01", "GSM2656366_9970497017_R06C01", "GSM3380800_6929726053_R05C01", "GSM3380802_6929726054_R01C02", "GSM3380804_6929726054_R03C02", "GSM3530006_9274651149_R05C02")
out1 <- gsub("GSM[0-9]{7}_", "", out1)
# out2 <- detectOutlier(as.matrix(betas(MLset))) %>% which(.) # Take too long to run
# save(out2, file = "RData/04-outliers.RData")
# out3 <- outlyx(MLset) 
# out3 <- rownames(out3)[out3$outliers == T]
# save(out2, out3, file = "RData/04-outliers.RData")
# out4 <- locfdr_LMM(betas(MLset), pData(MLset))
#save(out2, out3, out4, file = "RData/04-outliers.RData")
#load("RData/04-outliers.RData")

#pData(MLset)$Sample_ID <- sampleNames(MLset)
pData(MLset)$Out_or_not <- 0
pData(MLset)[sampleNames(MLset) %in% out1, ]$Out_or_not <- 1
pData(MLset)$Out_or_not <- as.factor(pData(MLset)$Out_or_not)
c(list(ClustOutlier = out2), list(IQRnPCOutlier = out3), list(LocfdrOutlier = out4))
out <- table(c(names(out4), out3, names(out2))) %>% .[.==3] %>% names() # remove samples that are identified as outliers in all methods (except for brain)
BetaPlot(MLset[, sampleNames(MLset) %in% out1], "Sample_ID") # examine what they look like
BetaPlot(MLset, "Cell_Type")
# all bad
MLset <- MLset[, !sampleNames(MLset) %in% out1]
BetaPlot(MLset[, 1:2500], "Cohort")
BetaPlot(MLset[, 2501:5941], "Cohort")

out <- table(c(names(out4), out3, names(out2))) %>% .[.==2] %>% names()
BetaPlot(MLset[, sampleNames(MLset) %in% out], "Sample_ID") 
```


### Subset for downstream analysis

```{r}
# reformating sentrix id info
beta <- betas(MLset)
meta <- pData(MLset)

sum(is.na(meta$Sentrix_ID)) # some missing info on sentrix id 
test <- meta[!is.na(meta$Sentrix_ID), ]
test$Sentrix_ID2 <- gsub("GSM[0-9]{7}_", "", rownames(test)) %>% gsub("_.*", "", .)
identical(test$Sentrix_ID, test$Sentrix_ID2) # true
test$Sentrix_Position2 <- gsub(".*_", "", rownames(test))
identical(test$Sentrix_Position, test$Sentrix_Position2) # true 
meta$Sentrix_ID <- gsub("GSM[0-9]{7}_", "", rownames(meta)) %>% gsub("_.*", "", .)
meta$Sentrix_Position <- gsub(".*_", "", rownames(meta))

meta$Sentrix_Position <- as.character(meta$Sentrix_Position)
meta$Sentrix_Row <- strsplit2(as.character(meta$Sentrix_Position), "C")[, 1] %>% gsub("R", "", .) # GSE104812 does not have Sentrix info
#meta <- meta[meta$Cohort != "GSE104812", ]
meta$Sentrix_Row <- as.factor(meta$Sentrix_Row)
levels(meta$Sentrix_Row) <- c(1,2,3,4,5,6,7,8)
table(meta$Sentrix_Row)
#colnames(meta)
meta <- meta[, !colnames(meta) %in% c("Sentrix_row", "Female", "cohort", "geo_accession", "title", "data_processing", "platform_id", 
                                      "chip", "position", "tissue", "celltype", "age", "sex", "disease_status", "art status:ch1", "art subtype:ch1")]

#Recode some diseases stuff
meta$Disease_Status <- gsub(".*Control.*", "Healthy Control", meta$Disease_Status) %>% 
    gsub("juvenile systemic sclerosis", "Juvenile Systemic Sclerosis", .) %>% 
    gsub("KDM6A.*|KMT2D.*|Kabuki", "Kabuki Syndrome", .) %>% 
    gsub("FASD", "Fetal Alcohol Spectrum Disorder", .) %>% 
    gsub("SMARCA2.*", "Nicolaides-Baraitser Syndrome", .) %>% 
    gsub("ASD", "Autism Spectrum Disorders", .) %>% 
    gsub("Cutaneous granulomatous", "ASXL1 variant", .)
table(meta$Disease_Status)

MLset <- MLset[, rownames(meta)]
pData(MLset) <- meta

# Subset to age group 6 - 18
meta <- pData(MLset)
meta <- meta[!is.na(meta$Age), ]
meta <-  meta[meta$Age < 18 & meta$Age >= 6, ]
MLset <- MLset[, rownames(meta)]
```

### Imputation

```{r, eval = F}
# Imputation of missing data
BetaPlot(MLset, "Cell_Type")

meta <- pData(MLset)
beta <- betas(MLset)
# beta1 <- beta[, 1:3000]
# beta2 <- beta[, 3001:ncol(beta)]
# bt_imputed1 <- impute.knn(beta1)
# bt_imputed2 <- impute.knn(beta2)
# bt_imputed <- cbind(bt_imputed1$data, bt_imputed2$data)
bt_imputed <- impute.knn(beta)
betas(MLset) <- bt_imputed$data
save(MLset, file = "RData/04-Imputed.RData")
```

### Cell type prediction

```{r}
load("RData/04-Imputed.RData")

source(paste0(home, "Functions/MRP.R"))
source(paste0(home, "Functions/BloodPlot.R"))
source(paste0(home, "/Functions/MRP_accessory.R"))
source(paste0(home, "/Functions/pickCompProbesCaret.R"))

# MRP function
load("/mnt/scratch/KoborLab/Personal_Folders/mfu/Projects/MRP/reference/FlowSorted.BloodExtended.EPIC_FN.RData") # load rfeference dataset

ct <- MRP(dataset = betas(MLset),
          sampType = rep("PB", ncol(MLset)), # c("PB", "CB"),
          class = "betas", # c("rgset", "betas")
          reference = FlowSorted.BloodExtended.EPIC.FN, # c(FlowSorted.CordBlood.450k.FN, FlowSorted.Blood.450k.FN, FlowSorted.Blood.EPIC.FN, FlowSorted.BloodExtended.EPIC.FN)
          cellTypes = c("Bas", "Bmem", "Bnv", "CD4mem", "CD4nv",
                        "CD8mem", "CD8nv", "Eos", "Mono", "Neu", "NK", "Treg"),
          norm = "QN.b", # c("Noob", "FN", "QN", "QN.b", "none")
          normComb = TRUE,
          probeList = "Ttest", #c("Ttest", "Caret", "IDOL", "DHS")
          method = "CP", #c("CP", "RPC", "SVR")
          probeSelect = "both", #c("both", "any", "pval")
          nProbes = 100,
          caretMods = c("EL", "BLR", "CART", "RF", "GBM", "RFEnNB", "RFEnSVM"),
          verbose = TRUE,
          EPIConly = F)

meta <- pData(MLset)
identical(rownames(ct$predictedProp), rownames(meta))
meta <- cbind(meta, ct$predictedProp)
pData(MLset) <- meta

plt.wb <- meta[meta$Disease_Status %in% c("Healthy Control", "ASXL1 variant") & meta$Cell_Type == "Whole Blood", ]
BloodPlot(mt = plt.wb, 
          vartype = "categorical", 
          varname = "Disease_Status", 
          tissue = "bloodextended") #+ scale_fill_manual(values = group.colors) 
# Remove any whole blood samples that have neutrophil proportions below 30% or above 70% (consider as outliers)
keep.wb <- plt.wb[plt.wb$Neu < 0.7 & plt.wb$Neu > 0.3, ]

plt.pbmc <- meta[meta$Disease_Status %in% c("Healthy Control", "ASXL1 variant") & meta$Cell_Type == "PBMC", ]
BloodPlot(mt = plt.pbmc, 
          vartype = "categorical", 
          varname = "Disease_Status", 
          tissue = "bloodextended") #+ scale_fill_manual(values = group.colors) 
# Remove any whole blood samples that have neutrophil proportions above 10% - should be 0 (consider as outliers)
keep.pbmc <- plt.pbmc[plt.pbmc$Neu < 0.1, ]


MLset_pbmc <- MLset[, rownames(plt.pbmc)]
MLset_wb <- MLset[, rownames(plt.wb)]

save(MLset_pbmc, MLset_wb, file = "RData/04-Imputed_ctPredicted.RData")

# estimatecellcount2
# load("RData/HM450_CT.RData")
# load("RData/EPIC_CT.RData")
# ct <- rbind(HM450.ct$counts, EPIC.ct$counts)
# rownames(ct) <- gsub("GSM[0-9]{7}_", "", rownames(ct))
# Plt <- MLset.F[, rownames(ct)]
# ct <- ct[sampleNames(MLset.F), ]
# meta <- pData(Plt)
# identical(rownames(ct), rownames(meta))
# Plt1 <- Plt[, meta$Disease_Status %in% c("Healthy Control", "ASXL1 variant") & meta$Cell_Type == "Whole Blood"]
# BloodPlot(Plt1, "Disease_Status", "categorical") + 
#     theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank())
# Plt2 <- Plt[, meta$Disease_Status %in% c("Healthy Control", "ASXL1 variant") & meta$Cell_Type == "PBMC"]
# BloodPlot(Plt2, "Disease_Status", "categorical") + 
#     theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank())
# 
# BloodPlot(Plt2, "Disease_Status", "categorical") + 
#     theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank())
```

#### Make cell type PCs

```{r}
load("RData/04-Imputed_ctPredicted.RData")
meta <- pData(MLset_wb)
meta <- pData(MLset_pbmc)

source("../../Functions/pcaCoDa_NG.R")

ct <- meta[, c("Bas", "Bmem", "Bnv", "CD4mem", "CD4nv",
               "CD8mem", "CD8nv", "Eos", "Mono", "Neu", "NK", "Treg")] %>% 
    as.data.frame()
sum(ct==0) #Are there any proportions predicted to be 0?
ct <- (ct + 0.0001) #If necessary, add the offset of 0.001
min(ct) # Cannot be = or < 0

pca_object <- pcaCoDa(ct)
CTP <- as.data.frame(pca_object$scores)
colnames(CTP) <- paste0("CTP_PC", 1:ncol(CTP))
rownames(CTP) <- rownames(ct)
identical(rownames(meta), rownames(CTP))
meta <- cbind(meta, CTP)

pData(MLset_wb) <- meta
pData(MLset_pbmc) <- meta

save(MLset_pbmc, MLset_wb, file = "RData/04-Imputed_ctPredicted.RData")
```

### Try outlier detection again

```{r}
load("RData/04-Imputed_ctPredicted.RData")

#separate PBMC and WB
# MLset_wb
out2 <- detectOutlier(as.matrix(betas(MLset_wb))) %>% which(.) # This calculates potential outliers
out3 <- outlyx(MLset_wb) 
out3 <- rownames(out3)[out3$outliers == T] 
out4 <- locfdr_LMM(betas(MLset_wb), pData(MLset_wb))
out <- c(names(out2), out3) %>% unique(.)
meta <- pData(MLset_wb)
meta$out <- 0
meta[out, "out"] <- 1
BetaPlot(betas(MLset_wb), meta, "out") #remove all outliers
MLset_wb.F <- MLset_wb[, !sampleNames(MLset_wb) %in% out]

# MLset_pbmc
out2 <- detectOutlier(as.matrix(betas(MLset_pbmc))) %>% which(.) # This calculates potential outliers
out3 <- outlyx(MLset_pbmc) 
out4 <- locfdr_LMM(betas(MLset_pbmc), pData(MLset_pbmc))
# only detected outlier is 200673610041_R03C01
MLset_pbmc.F <- MLset_pbmc[, sampleNames(MLset_pbmc) != "200673610041_R03C01"]

save(MLset_pbmc.F, MLset_wb.F, file = "RData/05-Outlier_filtered.RData")
```

### Check DNAm clustering - Hierarchical and PCA

```{r}
# Try the multiple test corrected plot
load("RData/05-Outlier_filtered.RData")
meta <- pData(MLset_wb.F)
beta <- betas(MLset_wb.F) 
meta <- pData(MLset_pbmc.F)
beta <- betas(MLset_pbmc.F) 

# source("../../Functions/PCA_plot.R")
# MLset.F <- MLset.F[, meta$Disease_Status]
# PCA.N <- prcomp(betas(MLset.F))
# meta_categorical <- meta[, c("Cohort", "Platform_ID", "Sentrix_ID", "Sentrix_row", "Tissue", "Sex", "Disease_Status")] 
# colnames(meta_categorical) <- c("Cohort", "Platform", "Barcode", "Barcode Position", "Tissue", "Sex", "Disease Status")
# meta_continuous <- meta[, c("Age"), drop = F] 
# colnames(meta_continuous) <- "Age"
# heat_scree_plot(MLset.N)
# PCA_Plot(PCA.N)

# save(PCA.N, file = "RData/05-PCA_normalized.RData")

# Heatmap
bt <- beta[sample(1:nrow(beta),50000), ]
pheatmap(bt, show_rownames = F)

# PCA
source("../../Functions/heatscreesimple.R")
PCA.C <- prcomp(beta)
meta_cat <- meta[,  c("Cohort", "Platform_ID", "Sentrix_ID", "Sex", "Disease_Status")] 
# colnames(meta_cat) <- c("Predicted_Sex", "Sample_Group")
meta_con <- meta[, c("Sentrix_Row", "Age", 
                     "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5",
                     "CTP_PC6", "CTP_PC7", "CTP_PC8", "CTP_PC9", "CTP_PC10", "CTP_PC11")] 
heatscreesimple(PCA.C$rotation, PCA.C$sdev^2, meta_cat, meta_con, 5) 

pcaObj <- PCA.C$rotation %>% as.data.frame() %>% cbind(., meta_cat) %>% cbind(., meta_con)
group.colors <- c("family control" = "#EBCB8BFF", "unrelated healthy control" = "#A3BE8CFF", "patient empty vector" ="#BF616AFF", "patient wild type" = "#B48EADFF")
ggplot(pcaObj, aes(PC1, PC2, color = Disease_Status, shape = Sex)) + 
    geom_point(size = 4) + 
    theme_bw() 
    # scale_color_manual(values = group.colors)
# In whole blood, PC3 seems to be sex and PC5 seems to separate disease status
```

```{r}
# Normalization (intra- and inter-sample) - skipped because we already did funnorm
# MLset.N <- dasen(MLset.P)
# NormPlot(MLset.P, MLset.N)

# OR BMIQ
# nfitmax <- sum(fData(MLset.P)$INFINIUM_DESIGN_TYPE == "I")
# MLset.N <- BMIQ(MLset.P, nfit = nfitmax)
# NormPlot(MLset, MLset)
# BetaPlot(MLset.P, "Tissue", "Individual") 
#save(MLset.N, file = "RData/05-normalized.RData")
```

### Add more meta info before adjusting value

#### Sex prediction

```{r}
load("RData/05-Outlier_filtered.RData")
source(paste0(home, "Functions/SexAgePrediction.R"))

MLset.F <- MLset_wb.F
MLset.F <- MLset_pbmc.F

if (is.null(fData(MLset.F)$CHR)) {
    if (dim(MLset)[[1]] > 600000) {
        load("~/Reference/EPIC_fdat.RData")
        fdat <- merge(fData(MLset.F), fData_EPIC, by = 0, sort = F)
    } else if (dim(MLset)[[1]] > 300000) {
        load("~/Reference/HM450_fdat.RData")
        fdat <- merge(fData(MLset.F), fData_450, by = 0, sort = F)
    }
    rownames(fdat) <- fdat$Row.names
    fData(MLset) <- fdat[rownames(MLset.F), -1]
}
(prediction <- SexPred(MLset.F)) 
pData(MLset.F)$Estimated_Sex_LMM <- prediction

meta <- pData(MLset.F)
beta <- betas(MLset.F)
test <- meta[meta$Sex!=meta$Estimated_Sex, ]
test2 <- meta[meta$Sex!=meta$Estimated_Sex_LMM, ]
out <- c(rownames(test), rownames(test2)) %>% .[duplicated(.)] # Only 1 is estimated to have the wrong sex using both methods
MLset.F <- MLset.F[, !sampleNames(MLset.F) %in% out]

BetaPlot(betas(MLset.F)[fData(MLset.F)$CHR == "Y", out], meta, "Sex") #remove all outliers
BetaPlot(betas(MLset.F)[fData(MLset.F)$CHR == "X", out], meta, "Sex") # intersex?

# remove outlier and save (for wb only, no intersex samples in pbmc)
MLset_wb.F <- MLset.F[, !sampleNames(MLset.F) %in% out]

save(MLset_pbmc.F, MLset_wb.F, file = "RData/05-Outlier_filtered.RData")
```


#### DNAm Age

```{r}
load("RData/05-Outlier_filtered.RData")

MLset.F <- MLset_wb.F
MLset.F <- MLset_pbmc.F

# DNAm Age
beta <- betas(MLset.F) %>% as.data.frame()
beta <- cbind(CpG = rownames(beta), beta)
cpgs.missing <- checkClocks(beta)
DNAmage <- DNAmAge(beta, clocks = c("Horvath", "Hannum", "skinHorvath", "Levine", "Wu", "TL"))
pData(MLset.F) <- cbind(pData(MLset.F), as.data.frame(DNAmage))

DNAmage_wb <- DNAmage
DNAmage_pbmc <- DNAmage
save(DNAmage_wb, DNAmage_pbmc, file = "RData/03-DNAm_Age.RData")

out <- c("9379082138_R03C02", "200069280019_R06C02") # predicted to be very very old
MLset_wb.F <- MLset.F[, !sampleNames(MLset.F) %in% out] 
MLset_pbmc.F <- MLset.F
save(MLset_pbmc.F, MLset_wb.F, file = "RData/05-Outlier_filtered.RData")

load("~/Projects/ASXL1/RData/03-DNAm_Age.RData")
MLset.F <- MLset_wb.F
MLset.F <- MLset_pbmc.F

meta <- pData(MLset.F)
# meta <- meta[meta$Disease_Status %in% c("Healthy Control", "ASXL1 variant"), ]
ggplot(meta, aes(Age, skinHorvath)) + geom_smooth(method = "lm") + geom_point(aes(color = Disease_Status, shape = Cell_Type)) + theme_classic()
ggplot(meta, aes(Age, Horvath)) + geom_smooth(method = "lm") + geom_point(aes(color = Disease_Status, shape = Cell_Type)) + theme_classic()
ggplot(meta, aes(Age, Hannum)) + geom_smooth(method = "lm") + geom_point(aes(color = Disease_Status, shape = Cell_Type)) + theme_classic()
ggplot(meta, aes(Age, TL)) + geom_smooth(method = "lm") + geom_point(aes(color = Disease_Status, shape = Cell_Type)) + theme_classic()
lm(Horvath ~ Age, data = meta) %>% residuals()
ggplot(meta, aes(Age, Horvath)) + geom_smooth(method = "lm") + geom_point(aes(color = Disease_Status)) + theme_classic()
lm(Horvath ~ Age, data = meta) %>% residuals()

# meta <- pData(MLset.P)
# ggplot(meta[meta$Disease_Status %in% c("Healthy", "ASXL1", "Control") & meta$Tissue %in% c("WB", "PBMC", "BEC"), ], aes(Age, DNAmAge, color = Tissue, shape = Disease_Status)) + geom_point(size = 2) + scale_shape_manual(values = c(18, 15, 1)) + geom_abline() + theme_bw()
# SamplePlots(MLset.P[, meta$Tissue %in% c("WB", "PBMC", "BEC")], hmvars = meta[, c("Age", "Cohort", "Platform_ID", "Tissue", "Disease_Status")], hmcor = T, mds = F, hclust = F, SNP = F, hmname = F)


# Nicole wrote the amazing code to get Lisa's clock running!
# source(paste0(home, "Functions/Age_Pred_McEwen_NG.R"))
# # MLset <- MLset[, !sampleNames(MLset) %in% c("EMVB089H ")]
# 
# MLset_BEC <- MLset[,pData(MLset)$Tissue == "BEC"]
# MLset_PA <- PedAge_McEwen_NG(MLset_BEC)
# MLset_PA$Sample_ID <- as.character(MLset_PA$SampleID) %>% gsub("X", "", .) %>% gsub("\\.", "-", .)
# #MLset_PA <- MLset_PA[MLset_PA$SampleID!= "EMVB089H.", ]
# pData(MLset)$PedBE <- NA
# sum(MLset_PA$Sample_ID %in% rownames(pData(MLset)))
# pData(MLset)[MLset_PA$Sample_ID, "PedBE"] <-  MLset_PA$DNAmAge
```

### Correct for cell type

```{r}
load("RData/05-Outlier_filtered.RData")

# WB
MLset.F <- MLset_wb.F
MLset.F <- MLset_pbmc.F

beta <- betas(MLset.F)
meta <- pData(MLset.F)
residuals <- t(apply(beta, 1, function(x){
    x <- as.numeric(x)
    residuals(summary(lm(x ~ CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4, data = meta))) # top 4 PCs account for 90% variance
}))
colnames(residuals) <- colnames(beta)
adj.residuals <- residuals + matrix(apply(beta, 1, mean), nrow = nrow(residuals), ncol = ncol(residuals))
adj.residuals[adj.residuals <= 0] <- 0.0001 # convert any values that are less than or equal to zero to 0.0001
adj.residuals[adj.residuals > 1] <- 0.9999

adj.residuals_wb <- adj.residuals
adj.residuals_pbmc <- adj.residuals

beta.CC <- cbind(adj.residuals_wb, adj.residuals_pbmc)
meta.CC <- rbind(pData(MLset_wb.F), pData(MLset_pbmc.F))
identical(rownames(meta.CC), colnames(beta.CC))
save(beta.CC, meta.CC, file = "RData/06-Celltype_corrected.RData")
```



### Batch correction

```{r}
load("RData/06-Celltype_corrected.RData")

beta <- beta.CC
meta <- meta.CC
# Subset out only blood & buccal and healthy + mut sample to make batch correction better
# MLset.C <- MLset.C[, sampleNames(MLset.C) %in% rownames(meta[meta$Tissue %in% c("BEC", "PBMC", "WB"), ])]
# MLset.C <- MLset.C[, sampleNames(MLset.C) %in% rownames(meta[meta$Disease_Status %in% c("Healthy", "Crohn's disease", "FASD", "ASXL1"), ])]
# meta <- pData(MLset.C)
# dim(MLset.C)
# Features  Samples 
#   405686      730 

#Combat
test <- apply(beta, 1, var) # Some value is exactly 0 - problem when changed to Mval
beta[beta == 0] <- 1e-5
dat <- as.matrix(beta2m(beta))
test2 <- apply(dat, 1, var)
test3 <- lapply(unique(meta$Platform_ID), function(x){
    out <- apply(dat[, meta$Sentrix_Row == x], 1, var)
    return(out)
})

#meta <- meta[meta$Age <=  ]
#sum(apply(beta, 1, var) == 0) # 0
#mod <- model.matrix(~ as.factor(Cell_Type), data = meta)
table(meta$Sentrix_Row)
M_c1 <- ComBat(dat, batch = meta$Sentrix_Row, mod = NULL, par.prior = T)
table(meta$Sentrix_ID)
M_c2 <- M_c2[meta$Sentrix_ID %in% names(table(meta$Sentrix_ID) > 1)]
M_c2 <- ComBat(M_c1, batch = meta$Sentrix_ID, mod = NULL, par.prior = T)
table(meta$Platform_ID)
meta$Platform_ID[meta$Cohort == "ASXL1"] <- "GPL21145"
M_c3 <- ComBat(M_c2, batch = meta$Platform_ID, mod = NULL, par.prior = T)
beta.BC <- m2beta(M_c3)
save(beta.BC, meta, file = "RData/07-Batch_corrected.RData")

# Use SVA to remove other batch effect (SentrixID and row missing from some cohorts)
# mod <- model.matrix(~ as.factor(Disease_Status), data = meta)
# mod0 <- model.matrix(~1, data = meta)
# (n.sv <- num.sv(dat, mod, method = "leek"))
# #n.sv <- 2
# svobj <- sva(beta2m(beta), mod, mod0, n.sv = n.sv)
# save(svobj, file = "RData/06-SVA.RData")

# PCA
source("../../Functions/heatscreesimple.R")
PCA.C <- prcomp(beta.BC)
meta_cat <- meta[,  c("Cohort", "Platform_ID", "Sentrix_ID", "Sex", "Disease_Status")] 
# colnames(meta_cat) <- c("Predicted_Sex", "Sample_Group")
meta_con <- meta[, c("Sentrix_Row", "Age", 
                     "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5",
                     "CTP_PC6", "CTP_PC7", "CTP_PC8", "CTP_PC9", "CTP_PC10", "CTP_PC11")] 
heatscreesimple(PCA.C$rotation, PCA.C$sdev^2, meta_cat, meta_con, 5) 
```



```{r}
load("RData/05-PCA_normalized.RData")
load("RData/06-PCA_batch_corrected.RData")

# Sanity Check
meta <- pData(MLset.C)
meta$Platform_ID[meta$Cohort == "ASXL1"] <- "GPL21145"
meta_categorical <- meta[, c("Cohort", "Platform_ID", "Sentrix_ID", "Sentrix_Row", "Sex", "Cell_Type", "Disease_Status")]
meta_continuous <- meta[, c("Age", "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5")]
heat_scree_plot(PCA.N, meta.cat = meta_categorical, meta.con = meta_continuous, PCs_to_view = 20) 
heat_scree_plot(PCA.C, meta.cat = meta_categorical, meta.con = meta_continuous, PCs_to_view = 20) 

Loadings <- PCA.N %>% .$rotation %>% unclass() %>% as.data.frame()
PC <- cbind(Loadings[, 1:10], meta[, c("Cohort", "Platform_ID", "Sentrix_ID", "Sentrix_Row", "Age", "Sex", "Cell_Type", "Disease_Status", "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5")])
test <- which(PC$PC3>0.01 & meta$Sex == "M")
plt <- PC[-test, ]
meta <- meta[-test, ]
plt <- plt[which(meta$Disease_Status %in% c("Healthy Control", "ASXL1 variant")), ]
ggplot(plt, aes(PC2, PC3, color = Cohort)) + geom_point() + theme_bw()
ggplot(plt, aes(PC3, PC4, color = Platform_ID)) + geom_point() + theme_bw() + labs(x = "PC2", y = "PC3")
ggplot(plt, aes(PC1, PC2, color = Sentrix_ID)) + geom_point() + theme_bw()
ggplot(plt, aes(PC4, PC5, color = Sentrix_Row)) + geom_point() + theme_bw()
ggplot(plt, aes(PC2, PC3, color = Sex)) + geom_point() + theme_bw() + labs(x = "PC1", y = "PC2")
ggplot(plt, aes(PC3, PC4, color = Cell_Type)) + geom_point() + theme_bw() + labs(x = "PC3", y = "PC4")
ggplot(plt, aes(PC3, PC4, color = Disease_Status)) + geom_point() + theme_bw() + labs(x = "PC2", y = "PC3")

MLset.C <- MLset.C[, sampleNames(MLset.C) %in% rownames(meta)]
pData(MLset.C) <- meta
save(MLset.C, file = "RData/06-Batch_corrected.RData")
```

```{r}
# See if Sentrix ID / row are correlated with either SV
colnames(svobj$sv) <- c("sv1", "sv2")
meta <- pData(MLset.C)
meta <- cbind(meta, svobj$sv) %>% as.data.frame()
aov(meta$sv1 ~ meta$Sentrix_ID + meta$Sentrix_row) %>% summary()
#                   Df  Sum Sq   Mean Sq F value Pr(>F)
# meta$Sentrix_ID  133 0.31132 0.0023408   3.640 <2e-16 ***
# meta$Sentrix_row   7 0.00598 0.0008538   1.328  0.236
aov(meta$sv2 ~ meta$Sentrix_ID + meta$Sentrix_row) %>% summary()
#                   Df Sum Sq   Mean Sq F value Pr(>F)
# meta$Sentrix_ID  133 0.2685 0.0020188   3.181 <2e-16 ***
# meta$Sentrix_row   7 0.0030 0.0004292   0.676  0.692

pData(MLset.C) <- meta
ggplot(meta, aes(sv1, sv2, color = Sentrix_ID)) + geom_point()
ggplot(meta, aes(sv1, sv2, color = Sentrix_row)) + geom_point()
ggplot(meta, aes(sv1, sv2, color = Tissue)) + geom_point()

with(meta, aov(sv1 ~ Epi + Fib + B + NK + CD4T + CD8T + Mono + Neutro + Eosino)) %>% summary()
with(meta, aov(sv2 ~ Epi + Fib + B + NK + CD4T + CD8T + Mono + Neutro + Eosino)) %>% summary()

# Given that sv1 and sv2 are significantly associated with Sentrix ID and Sentrix row, we will use them as proxy for those measures, and adjust them with combat
mod <- model.matrix(~ as.factor(Tissue) + as.factor(Disease_Status), data = meta)
# M_c3 <- ComBat(M_c2, batch = meta$Sentrix_ID, mod = mod)
# M_c4 <- ComBat(M_c3, batch = meta$Platform_ID, mod = mod)
# Cannot do it because svs are continuous variables -> will regress out with cell type

save(MLset.C, svobj, file = "RData/06-batch_corrected_tissue_protected.RData")
```

### Check for rep correlation

```{r}
source(paste0(home, "Functions/RepCorTest.R"))
load("RData/05-Outlier_filtered.RData")
load("RData/06-Celltype_corrected.RData")
load("RData/07-Batch_corrected.RData")

reps <- c("203077630141_R01C01", "203077630141_R06C01")
Rnorm <- betas(MLset_pbmc.F)[, sampleNames(MLset_pbmc.F) %in% reps] # normalized, outlier removed
Rcc <- beta.CC[, reps] # celltype corrected
Rbc <- beta.BC[, reps] # batch corrected

# Calculate correlation between tech reps after each pre-processing step
Cnorm <- cor(na.omit(Rnorm))[1,2]
Ccc <- cor(na.omit(Rcc))[1,2]
Cbc <- cor(na.omit(Rbc))[1,2]
Cval <- c(Cnorm, Ccc, Cbc)
Cname <- c(1:length(Cval)) %>% as.character(.)
Cdf <- data.frame(Cname, Cval)
    
p1 <- ggplot(Cdf, aes(Cname, Cval)) +
    geom_point(color = "#005C66", alpha = 0.75, size = 3.5) +
    xlab("Pre-Processing and Normalization Step") +
    ylab("Spearman's Correlation (rho)") +
    scale_x_discrete(breaks = c(1:length(Cval)), 
                     labels = c("Normalized", "Celltype Corrected", "Batch Corrected")) +
    theme_bw(base_size = 10)

# Calculate root mean square error between tech reps after each pre-processing step    
Enorm <- rmse(na.omit(Rnorm)[,1], na.omit(Rnorm)[,2])
Ecc <- rmse(na.omit(Rcc)[,1], na.omit(Rcc)[,2])
Ebc <- rmse(na.omit(Rbc)[,1], na.omit(Rbc)[,2])
Eval <- c(Enorm, Ecc, Ebc)
Ename <- c(1:length(Eval)) %>% as.character(.)
Edf <- data.frame(Ename, Eval)
    
p2 <- ggplot(Edf, aes(Ename, Eval)) +
        geom_point(color = "#733556", alpha = 0.75, size = 3.5) +
        xlab("Pre-Processing and Normalization Step") +
        ylab("Root Mean Square Error") +
        scale_x_discrete(breaks = c(1:length(Eval)), 
                         labels = c("Normalized", "Celltype Corrected", "Batch Corrected")) +
        theme_bw(base_size = 10)

grid.arrange(p1, p2)
```



## Test

### Outlier detection test

#### WB

```{r}
load("RData/07-Batch_corrected.RData")

# Whole Blood
beta <- beta.BC
betas.pat <- beta[, "203077630141_R07C01"]
betas.ref <- beta[, meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "Whole Blood"]

betas.refIQR <- apply(betas.ref, 1, function(x) {quantile(x, c(0.025, 0.25, 0.5, 0.75, 0.975))}) %>% t() %>% as.data.frame()
betas.refIQR$iqr <- betas.refIQR[, 4] - betas.refIQR[, 2]
betas.refIQR$betas.refIQRl <- betas.refIQR[, 2] - 1.5*betas.refIQR$iqr
betas.refIQR$betas.refIQRh <- betas.refIQR[, 4] + 1.5*betas.refIQR$iqr
betas.refSD <- apply(betas.ref, 1, sd)
betas.refM <- apply(betas.ref, 1, mean)

ASXL1.test <- cbind(betas.refIQR, betas.refSD, betas.refM, betas.pat) %>% as.data.frame()
ASXL1.test$betas.refSD2l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD2h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD3l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD3h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD4l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*4)
ASXL1.test$betas.refSD4h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*4)

out <- apply(as.matrix(ASXL1.test), 1, function(x){
    IQR.test.wb <- x["betas.pat"] < x["2.5%"] | 
        x["betas.pat"] > x["97.5%"]
    IQR2.test.wb <- x["betas.pat"] < x["betas.refIQRl"] |
        x["betas.pat"] > x["betas.refIQRh"]
    SD2.test.wb <- x["betas.pat"] < x["betas.refSD2l"] |
        x["betas.pat"] > x["betas.refSD2h"]
    SD3.test.wb <- x["betas.pat"] < x["betas.refSD3l"] |
        x["betas.pat"] > x["betas.refSD3h"]
    SD4.test.wb <- x["betas.pat"] < x["betas.refSD4l"] |
        x["betas.pat"] > x["betas.refSD4h"]
    return(cbind(IQR.test.wb, IQR2.test.wb, SD2.test.wb, SD3.test.wb, SD4.test.wb))
})

out <- t(out)
colnames(out) <- c("IQR.test.wb", "IQR2.test.wb", "SD2.test.wb", "SD3.test.wb", "SD4.test.wb")
ASXL1.test.wb <- cbind(ASXL1.test, as.data.frame(out))
ASXL1.test.wb$meandiff <- ASXL1.test.wb$betas.pat - ASXL1.test.wb$betas.refM

save(ASXL1.test.wb, file = "RData/10-global_test_wb.RData")

sum(ASXL1.test.wb$IQR.test.wb) # 136301
sum(ASXL1.test.wb$IQR2.test.wb) # 106073
sum(ASXL1.test.wb$SD2.test.wb) # 116455
sum(ASXL1.test.wb$SD3.test.wb) # 38662
sum(ASXL1.test.wb$SD4.test.wb) # 13138
```

#### WB - But with shuffled labels to check test sensitivity / precision

```{r}
load("RData/07-Batch_corrected.RData")

# Whole Blood
beta <- beta.BC
test <- sample(rownames(meta)[meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "Whole Blood"], 1)
betas.pat <- beta[, test]
betas.ref <- beta[, meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "Whole Blood" & colnames(beta) != test]

betas.refIQR <- apply(betas.ref, 1, function(x) {quantile(x, c(0.025, 0.25, 0.5, 0.75, 0.975))}) %>% t() %>% as.data.frame()
betas.refIQR$iqr <- betas.refIQR[, 4] - betas.refIQR[, 2]
betas.refIQR$betas.refIQRl <- betas.refIQR[, 2] - 1.5*betas.refIQR$iqr
betas.refIQR$betas.refIQRh <- betas.refIQR[, 4] + 1.5*betas.refIQR$iqr
betas.refSD <- apply(betas.ref, 1, sd)
betas.refM <- apply(betas.ref, 1, mean)

ASXL1.test <- cbind(betas.refIQR, betas.refSD, betas.refM, betas.pat) %>% as.data.frame()
ASXL1.test$betas.refSD2l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD2h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD3l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD3h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD4l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*4)
ASXL1.test$betas.refSD4h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*4)

out <- apply(as.matrix(ASXL1.test), 1, function(x){
    IQR.test.wb <- x["betas.pat"] < x["2.5%"] | 
        x["betas.pat"] > x["97.5%"]
    IQR2.test.wb <- x["betas.pat"] < x["betas.refIQRl"] |
        x["betas.pat"] > x["betas.refIQRh"]
    SD2.test.wb <- x["betas.pat"] < x["betas.refSD2l"] |
        x["betas.pat"] > x["betas.refSD2h"]
    SD3.test.wb <- x["betas.pat"] < x["betas.refSD3l"] |
        x["betas.pat"] > x["betas.refSD3h"]
    SD4.test.wb <- x["betas.pat"] < x["betas.refSD4l"] |
        x["betas.pat"] > x["betas.refSD4h"]
    return(cbind(IQR.test.wb, IQR2.test.wb, SD2.test.wb, SD3.test.wb, SD4.test.wb))
})

out <- t(out)
colnames(out) <- c("IQR.test.wb", "IQR2.test.wb", "SD2.test.wb", "SD3.test.wb", "SD4.test.wb")
ASXL1.test.wb_ctr <- cbind(ASXL1.test, as.data.frame(out)) #control sample
ASXL1.test.wb_ctr$meandiff <- ASXL1.test.wb_ctr$betas.pat - ASXL1.test.wb_ctr$betas.refM

save(ASXL1.test.wb_ctr, file = "RData/10-global_test_wb_ctr.RData")

sum(ASXL1.test.wb_ctr$IQR.test.wb & ASXL1.test.wb_ctr$meandiff > 0.05) # 44531 & 5759 (w/o and w/ effect size cutoff)
sum(ASXL1.test.wb_ctr$IQR2.test.wb & ASXL1.test.wb_ctr$meandiff > 0.05) # 47622 & 4767
sum(ASXL1.test.wb_ctr$SD2.test.wb & ASXL1.test.wb_ctr$meandiff > 0.05) # 48887 & 6127 (1.4%)
sum(ASXL1.test.wb_ctr$SD3.test.wb & ASXL1.test.wb_ctr$meandiff > 0.05) # 7942 & 1352 (0.3%)
sum(ASXL1.test.wb_ctr$SD4.test.wb & ASXL1.test.wb_ctr$meandiff > 0.05) # 1890 & 391
```


#### PBMC

```{r}
load("RData/07-Batch_corrected.RData")

# PBMC
beta <- beta.BC
betas.pat <- cbind(betas.pat_pbmc = beta[, "203077630141_R01C01"], 
                   betas.pat_pbmc.r = beta[, "203077630141_R06C01"])
betas.ref <- beta[, meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "PBMC"]

betas.refIQR <- apply(betas.ref, 1, function(x) {quantile(x, c(0.025, 0.25, 0.5, 0.75, 0.975))}) %>% t() %>% as.data.frame()
betas.refIQR$iqr <- betas.refIQR[, 4] - betas.refIQR[, 2]
betas.refIQR$betas.refIQRl <- betas.refIQR[, 2] - 1.5*betas.refIQR$iqr
betas.refIQR$betas.refIQRh <- betas.refIQR[, 4] + 1.5*betas.refIQR$iqr
betas.refSD <- apply(betas.ref, 1, sd)
betas.refM <- apply(betas.ref, 1, mean)

ASXL1.test <- cbind(betas.refIQR, betas.refSD, betas.refM, betas.pat) %>% as.data.frame()
ASXL1.test$betas.refSD2l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD2h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD3l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD3h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD4l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*4)
ASXL1.test$betas.refSD4h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*4)

test <- as.matrix(ASXL1.test)

out <- apply(as.matrix(ASXL1.test), 1, function(x){
    IQR.test.pbmc <- x["betas.pat_pbmc"] < x["2.5%"] | 
        x["betas.pat_pbmc"] > x["97.5%"]
    IQR.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["2.5%"] | 
        x["betas.pat_pbmc"] > x["97.5%"]
    IQR.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["2.5%"] | 
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["97.5%"]
    IQR2.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refIQRl"] |
        x["betas.pat_pbmc"] > x["betas.refIQRh"]
    IQR2.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refIQRl"] |
        x["betas.pat_pbmc"] > x["betas.refIQRh"]
    IQR2.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refIQRl"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refIQRh"]
    SD2.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refSD2l"] |
        x["betas.pat_pbmc"] > x["betas.refSD2h"]
    SD2.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refSD2l"] |
        x["betas.pat_pbmc"] > x["betas.refSD2h"]
    SD2.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refSD2l"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refSD2h"]
    SD3.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refSD3l"] |
        x["betas.pat_pbmc"] > x["betas.refSD3h"]
    SD3.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refSD3l"] |
        x["betas.pat_pbmc"] > x["betas.refSD3h"]
    SD3.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refSD3l"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refSD3h"]
    SD4.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refSD4l"] |
        x["betas.pat_pbmc"] > x["betas.refSD4h"]
    SD4.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refSD4l"] |
        x["betas.pat_pbmc"] > x["betas.refSD4h"]
    SD4.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refSD4l"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refSD4h"]
    return(cbind(IQR.test.pbmc, IQR.test.pbmc.r, IQR.test.avg,
                 IQR2.test.pbmc, IQR2.test.pbmc.r, IQR2.test.avg,
                 SD2.test.pbmc, SD2.test.pbmc.r, SD2.test.avg,
                 SD3.test.pbmc, SD3.test.pbmc.r, SD3.test.avg,
                 SD4.test.pbmc, SD4.test.pbmc.r, SD4.test.avg))
})

out <- t(out)
colnames(out) <- c("IQR.test.pbmc", "IQR.test.pbmc.r", "IQR.test.avg",
                 "IQR2.test.pbmc", "IQR2.test.pbmc.r", "IQR2.test.avg",
                 "SD2.test.pbmc", "SD2.test.pbmc.r", "SD2.test.avg",
                 "SD3.test.pbmc", "SD3.test.pbmc.r", "SD3.test.avg",
                 "SD4.test.pbmc", "SD4.test.pbmc.r", "SD4.test.avg")
ASXL1.test.pbmc <- cbind(ASXL1.test, as.data.frame(out))
ASXL1.test.pbmc$meandiff.pbmc <- ASXL1.test.pbmc$betas.pat_pbmc - ASXL1.test.pbmc$betas.refM
ASXL1.test.pbmc$meandiff.pbmc.r <- ASXL1.test.pbmc$betas.pat_pbmc.r - ASXL1.test.pbmc$betas.refM
ASXL1.test.pbmc$meandiff.pbmc.avg <- rowMeans(ASXL1.test.pbmc[, c("betas.pat_pbmc", "betas.pat_pbmc.r")]) - ASXL1.test.pbmc$betas.refM
save(ASXL1.test.pbmc, file = "RData/10-global_test_pbmc.RData")

sum(ASXL1.test.pbmc$IQR.test.pbmc & ASXL1.test.pbmc$meandiff.pbmc > 0.05) # 39734 & 11691
sum(ASXL1.test.pbmc$IQR.test.pbmc.r & ASXL1.test.pbmc$meandiff.pbmc.r > 0.05) # 39136 & 7214
sum(ASXL1.test.pbmc$IQR.test.avg & ASXL1.test.pbmc$meandiff.pbmc.avg > 0.05) # 39734 & 9253
sum(ASXL1.test.pbmc$IQR2.test.pbmc & ASXL1.test.pbmc$meandiff.pbmc > 0.05) # 30530 & 7184
sum(ASXL1.test.pbmc$IQR2.test.pbmc.r & ASXL1.test.pbmc$meandiff.pbmc.r > 0.05) # 31097 & 4918
sum(ASXL1.test.pbmc$IQR2.test.avg & ASXL1.test.pbmc$meandiff.pbmc.avg > 0.05) # 30530 & 5959

sum(ASXL1.test.pbmc$SD2.test.pbmc & ASXL1.test.pbmc$meandiff.pbmc > 0.05) # 38998 & 10294
sum(ASXL1.test.pbmc$SD2.test.pbmc.r & ASXL1.test.pbmc$meandiff.pbmc.r > 0.05) # 39515 & 6531
sum(ASXL1.test.pbmc$SD2.test.avg & ASXL1.test.pbmc$meandiff.pbmc.avg > 0.05) # 38998 & 8243
sum(ASXL1.test.pbmc$SD3.test.pbmc & ASXL1.test.pbmc$meandiff.pbmc > 0.05) # 12864 & 4758
sum(ASXL1.test.pbmc$SD3.test.pbmc.r & ASXL1.test.pbmc$meandiff.pbmc.r > 0.05) # 13208 & 3349
sum(ASXL1.test.pbmc$SD3.test.avg & ASXL1.test.pbmc$meandiff.pbmc.avg > 0.05) # 12864 & 4067
sum(ASXL1.test.pbmc$SD4.test.pbmc & ASXL1.test.pbmc$meandiff.pbmc > 0.05) # 5316 & 2613
sum(ASXL1.test.pbmc$SD4.test.pbmc.r & ASXL1.test.pbmc$meandiff.pbmc.r > 0.05) # 5410 & 1983
sum(ASXL1.test.pbmc$SD4.test.avg & ASXL1.test.pbmc$meandiff.pbmc.avg > 0.05) # 5316 & 2318
```

#### PBMC - But with shuffled labels to check test sensitivity / precision

```{r}
load("RData/07-Batch_corrected.RData")

# PBMC
beta <- beta.BC
test <- sample(rownames(meta)[meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "PBMC"], 2)
betas.pat <- cbind(betas.pat_pbmc = beta[, test[1]], 
                   betas.pat_pbmc.r = beta[, test[2]])
betas.ref <- beta[, meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "PBMC" & !colnames(beta) %in% test]

betas.refIQR <- apply(betas.ref, 1, function(x) {quantile(x, c(0.025, 0.25, 0.5, 0.75, 0.975))}) %>% t() %>% as.data.frame()
betas.refIQR$iqr <- betas.refIQR[, 4] - betas.refIQR[, 2]
betas.refIQR$betas.refIQRl <- betas.refIQR[, 2] - 1.5*betas.refIQR$iqr
betas.refIQR$betas.refIQRh <- betas.refIQR[, 4] + 1.5*betas.refIQR$iqr
betas.refSD <- apply(betas.ref, 1, sd)
betas.refM <- apply(betas.ref, 1, mean)

ASXL1.test <- cbind(betas.refIQR, betas.refSD, betas.refM, betas.pat) %>% as.data.frame()
ASXL1.test$betas.refSD2l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD2h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD3l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD3h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD4l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*4)
ASXL1.test$betas.refSD4h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*4)

test <- as.matrix(ASXL1.test)

out <- apply(as.matrix(ASXL1.test), 1, function(x){
    IQR.test.pbmc <- x["betas.pat_pbmc"] < x["2.5%"] | 
        x["betas.pat_pbmc"] > x["97.5%"]
    IQR.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["2.5%"] | 
        x["betas.pat_pbmc"] > x["97.5%"]
    IQR.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["2.5%"] | 
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["97.5%"]
    IQR2.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refIQRl"] |
        x["betas.pat_pbmc"] > x["betas.refIQRh"]
    IQR2.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refIQRl"] |
        x["betas.pat_pbmc"] > x["betas.refIQRh"]
    IQR2.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refIQRl"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refIQRh"]
    SD2.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refSD2l"] |
        x["betas.pat_pbmc"] > x["betas.refSD2h"]
    SD2.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refSD2l"] |
        x["betas.pat_pbmc"] > x["betas.refSD2h"]
    SD2.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refSD2l"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refSD2h"]
    SD3.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refSD3l"] |
        x["betas.pat_pbmc"] > x["betas.refSD3h"]
    SD3.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refSD3l"] |
        x["betas.pat_pbmc"] > x["betas.refSD3h"]
    SD3.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refSD3l"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refSD3h"]
    SD4.test.pbmc <- x["betas.pat_pbmc"] < x["betas.refSD4l"] |
        x["betas.pat_pbmc"] > x["betas.refSD4h"]
    SD4.test.pbmc.r <- x["betas.pat_pbmc.r"] < x["betas.refSD4l"] |
        x["betas.pat_pbmc"] > x["betas.refSD4h"]
    SD4.test.avg <- mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) < x["betas.refSD4l"] |
        mean(x["betas.pat_pbmc"], x["betas.pat_pbmc.r"]) > x["betas.refSD4h"]
    return(cbind(IQR.test.pbmc, IQR.test.pbmc.r, IQR.test.avg,
                 IQR2.test.pbmc, IQR2.test.pbmc.r, IQR2.test.avg,
                 SD2.test.pbmc, SD2.test.pbmc.r, SD2.test.avg,
                 SD3.test.pbmc, SD3.test.pbmc.r, SD3.test.avg,
                 SD4.test.pbmc, SD4.test.pbmc.r, SD4.test.avg))
})

out <- t(out)
colnames(out) <- c("IQR.test.pbmc", "IQR.test.pbmc.r", "IQR.test.avg",
                 "IQR2.test.pbmc", "IQR2.test.pbmc.r", "IQR2.test.avg",
                 "SD2.test.pbmc", "SD2.test.pbmc.r", "SD2.test.avg",
                 "SD3.test.pbmc", "SD3.test.pbmc.r", "SD3.test.avg",
                 "SD4.test.pbmc", "SD4.test.pbmc.r", "SD4.test.avg")
ASXL1.test.pbmc_ctr <- cbind(ASXL1.test, as.data.frame(out))
ASXL1.test.pbmc_ctr$meandiff.pbmc <- ASXL1.test.pbmc_ctr$betas.pat_pbmc - ASXL1.test.pbmc_ctr$betas.refM
ASXL1.test.pbmc_ctr$meandiff.pbmc.r <- ASXL1.test.pbmc_ctr$betas.pat_pbmc.r - ASXL1.test.pbmc_ctr$betas.refM
ASXL1.test.pbmc_ctr$meandiff.pbmc.avg <- rowMeans(ASXL1.test.pbmc_ctr[, c("betas.pat_pbmc", "betas.pat_pbmc.r")]) - ASXL1.test.pbmc_ctr$betas.refM
save(ASXL1.test.pbmc_ctr, file = "RData/10-global_test_pbmc_ctr.RData")

sum(ASXL1.test.pbmc_ctr$IQR.test.pbmc & ASXL1.test.pbmc_ctr$meandiff.pbmc > 0.05) # 67362 & 6317 (without and with effect size cutoff)
sum(ASXL1.test.pbmc_ctr$IQR.test.pbmc.r & ASXL1.test.pbmc_ctr$meandiff.pbmc.r > 0.05) # 35249 & 1877
sum(ASXL1.test.pbmc_ctr$IQR.test.avg & ASXL1.test.pbmc_ctr$meandiff.pbmc.avg > 0.05) # 67362 & 3677
sum(ASXL1.test.pbmc_ctr$IQR2.test.pbmc & ASXL1.test.pbmc_ctr$meandiff.pbmc > 0.05) # 20258 & 1508
sum(ASXL1.test.pbmc_ctr$IQR2.test.pbmc.r & ASXL1.test.pbmc_ctr$meandiff.pbmc.r > 0.05) # 12928 & 400
sum(ASXL1.test.pbmc_ctr$IQR2.test.avg & ASXL1.test.pbmc_ctr$meandiff.pbmc.avg > 0.05) # 20258 & 890

sum(ASXL1.test.pbmc_ctr$SD2.test.pbmc & ASXL1.test.pbmc_ctr$meandiff.pbmc > 0.05) # 46527 & 3449
sum(ASXL1.test.pbmc_ctr$SD2.test.pbmc.r & ASXL1.test.pbmc_ctr$meandiff.pbmc.r > 0.05) # 25239 & 1111
sum(ASXL1.test.pbmc_ctr$SD2.test.avg & ASXL1.test.pbmc_ctr$meandiff.pbmc.avg > 0.05) # 46527 & 2169
sum(ASXL1.test.pbmc_ctr$SD3.test.pbmc & ASXL1.test.pbmc_ctr$meandiff.pbmc > 0.05) # 4122 & 263
sum(ASXL1.test.pbmc_ctr$SD3.test.pbmc.r & ASXL1.test.pbmc_ctr$meandiff.pbmc.r > 0.05) # 2636 & 39
sum(ASXL1.test.pbmc_ctr$SD3.test.avg & ASXL1.test.pbmc_ctr$meandiff.pbmc.avg > 0.05) # 4122 & 156
sum(ASXL1.test.pbmc_ctr$SD4.test.pbmc & ASXL1.test.pbmc_ctr$meandiff.pbmc > 0.05) # 549 & 76
sum(ASXL1.test.pbmc_ctr$SD4.test.pbmc.r & ASXL1.test.pbmc_ctr$meandiff.pbmc.r > 0.05) # 653 & 10
sum(ASXL1.test.pbmc_ctr$SD4.test.avg & ASXL1.test.pbmc_ctr$meandiff.pbmc.avg > 0.05) # 549 & 45
```


### Permutation Test

```{r}
load("RData/07-Batch_corrected.RData")
beta <- beta.BC

# Set progress bar
set.seed(1234)

# Sign test
permTest <- function(cpg, group, group_var, n) {
     g <- unique(group)
     m0 <- mean(cpg[group == group_var]) # find the average DNAm level for group_var (in this case patient sample)
     m1 <- sapply(n, function(x) {
         ngroup <- sample(group, length(group)) # random shuffling of labels
        mean(cpg[ngroup == group_var]) # find the average DNAm level for randomly labelled samples
    })
     out <- c(sum(m0>=m1), sum(m0<=m1))
     return(out)
}

N <- 2000
cl <- 10

# Whole blood - sign test
beta.t <- beta[, meta$Cell_Type == "Whole Blood"]
meta.t <- meta[colnames(beta.t), ]
permOut.wb <- pbapply(beta.t, 1, function(x) permTest(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
permOut.wb <- permOut.wb/N
colnames(permOut.wb) <- c("Hypo_pval", "Hyper_pval")
permOut.wb$Hypo_qval <- p.adjust(permOut.wb$Hypo_pval)
permOut.wb$Hyper_qval <- p.adjust(permOut.wb$Hyper_pval)
permOut.wb$DeltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])

# PBMC - sign test
beta.t <- beta[, meta$Cell_Type =="PBMC"]
meta.t <- meta[colnames(beta.t), ]
permOut.pbmc <- pbapply(beta.t, 1, function(x) permTest(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
permOut.pbmc <- permOut.pbmc/N
colnames(permOut.pbmc) <- c("Hypo_pval", "Hyper_pval")
permOut.pbmc$Hypo_qval <- p.adjust(permOut.pbmc$Hypo_pval)
permOut.pbmc$Hyper_qval <- p.adjust(permOut.pbmc$Hyper_pval)

# PBMC (REP1 only) - sign test
beta.t <- beta[, meta$Cell_Type == "PBMC" & colnames(beta) != "203077630141_R06C01"]
meta.t <- meta[colnames(beta.t), ]
permOut.pbmc_r1 <- pbapply(beta.t, 1, function(x) permTest(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
permOut.pbmc_r1 <- permOut.pbmc_r1/N
colnames(permOut.pbmc_r1) <- c("Hypo_pval", "Hyper_pval")
permOut.pbmc_r1$Hypo_qval <- p.adjust(permOut.pbmc_r1$Hypo_pval)
permOut.pbmc_r1$Hyper_qval <- p.adjust(permOut.pbmc_r1$Hyper_pval)

# PBMC (REP2 only) - sign test
beta.t <- beta[, meta$Cell_Type == "PBMC" & colnames(beta) != "203077630141_R01C01"]
meta.t <- meta[colnames(beta.t), ]
permOut.pbmc_r2 <- pbapply(beta.t, 1, function(x) permTest(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
permOut.pbmc_r2 <- permOut.pbmc_r2/N
colnames(permOut.pbmc_r2) <- c("Hypo_pval", "Hyper_pval")
permOut.pbmc_r2$Hypo_qval <- p.adjust(permOut.pbmc_r2$Hypo_pval)
permOut.pbmc_r2$Hyper_qval <- p.adjust(permOut.pbmc_r2$Hyper_pval)

# Check overlap
cutoff <- 0.05 # Likelihood for patient sample to get picked
test.hyper <- intersect(rownames(permOut.pbmc_r1)[permOut.pbmc_r1$Hyper_qval < cutoff], rownames(permOut.pbmc_r2)[permOut.pbmc_r2$Hyper_qval < cutoff]) # 77 vs 0
test.hypo <- intersect(rownames(permOut.pbmc_r1)[permOut.pbmc_r1$Hypo_qval < cutoff], rownames(permOut.pbmc_r2)[permOut.pbmc_r2$Hypo_qval < cutoff]) # 207 vs 0
# test2.hyper <- intersect(rownames(permOut2.pat)[permOut2.pat$Hyper_pval < cutoff], rownames(permOut2.res)[permOut2.res$Hyper_pval < cutoff]) # 28 / 185
# test2.hypo <- intersect(rownames(permOut2.pat)[permOut2.pat$Hypo_pval < cutoff], rownames(permOut2.res)[permOut2.res$Hypo_pval < cutoff]) # 28 / 185
length(rownames(permOut.pbmc)[permOut.pbmc$Hyper_pval < cutoff]) # 48515
length(rownames(permOut.pbmc)[permOut.pbmc$Hypo_pval < cutoff]) # 137860
length(rownames(permOut.WT)[permOut.WT$Hyper_pval < cutoff]) # 94067
length(rownames(permOut.WT)[permOut.WT$Hypo_pval < cutoff]) # 76163
length(intersect(rownames(permOut2.pat)[permOut2.pat$pval < cutoff], rownames(permOut2.res)[permOut2.res$pval < cutoff]))

# Make heatmap
plt.hyper <- beta[rownames(permOut.pbmc)[permOut.pbmc$Hyper_pval < cutoff], ] %>% 
    .[.]
plt.hypo <- beta[rownames(permOut.pbmc)[permOut.pbmc$Hypo_pval < cutoff], ]
pheatmap(plt.hyper, show_rownames = F, scale = "row", treeheight_row = 0)
pheatmap(plt.hypo[sample(1:nrow(plt.hypo), 50000), ], show_rownames = F, scale = "row", treeheight_row = 0)

plt.hyper <- beta[rownames(permOut.WT)[permOut.WT$Hyper_pval < cutoff], ]
plt.hypo <- beta[rownames(permOut.WT)[permOut.WT$Hypo_pval < cutoff], ]
pheatmap(plt.hyper[sample(1:nrow(plt.hypo), 50000), ], show_rownames = F, scale = "row", treeheight_row = 0)
pheatmap(plt.hypo[sample(1:nrow(plt.hypo), 50000), ], show_rownames = F, scale = "row", treeheight_row = 0)

pheatmap(rbind(plt.hyper, plt.hypo), show_rownames = F, scale = "row")
```

```{r}
load("RData/07-Batch_corrected.RData")
beta <- beta.BC

# Set progress bar`
set.seed(1234)

# Modified rank sum test
permTest2 <- function(cpg, group, group_var, n) {
    g <- unique(group)
    ord0 <- mean(order(cpg)[group == group_var]) # calculate the rank of patient sample's beta value for a given CpG (compared to HCs)
    ord0_d <- (length(group) / 2 - ord0) %>% abs() # calculate the distance from the median rank (the larger the more extreme the value is)
    ord1_d <- sapply(n, function(x) {
        ngroup <- sample(group, length(group)) # random shuffling of labels
        ord1 <- mean(order(cpg)[ngroup %in% group_var])
        ord1_d <- (length(group) / 2 - ord1) %>% abs()
        return(ord1_d)
    })
    out <- sum(ord1_d >= ord0_d) # if the shuffled (permuted) rank is rarely more extreme than the actual rank (ord0), than the patient is very different from HC
    return(out)
} 

N <- 2000
cl <- 10

# Whole blood - modified rank sum test
beta.t <- beta[, meta$Cell_Type == "Whole Blood"]
meta.t <- meta[colnames(beta.t), ]
permOut.wb <- pbapply(beta.t, 1, function(x) permTest2(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) 
deltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])
permOut.wb <- cbind(deltaB = deltaB, pval = permOut.wb/N) %>% as.data.frame()
permOut.wb$qval <- p.adjust(permOut.wb$pval)

# PBMC - modified rank sum test
beta.t <- beta[, meta$Cell_Type =="PBMC"]
meta.t <- meta[colnames(beta.t), ]
permOut.pbmc <- pbapply(beta.t, 1, function(x) permTest2(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) 
deltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])
permOut.pbmc <- cbind(deltaB = deltaB, pval = permOut.pbmc/N) %>% as.data.frame()
permOut.pbmc$qval <- p.adjust(permOut.pbmc$pval)

# PBMC (REP1 only) - modified rank sum test
beta.t <- beta[, meta$Cell_Type == "PBMC" & colnames(beta) != "203077630141_R06C01"]
meta.t <- meta[colnames(beta.t), ]
permOut.pbmc_r1 <- pbapply(beta.t, 1, function(x) permTest2(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) 
deltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])
permOut.pbmc_r1 <- cbind(deltaB = deltaB, pval = permOut.pbmc_r1/N) %>% as.data.frame()
permOut.pbmc_r1$qval <- p.adjust(permOut.pbmc_r1$pval)

# PBMC (REP2 only) - modified rank sum test
beta.t <- beta[, meta$Cell_Type =="PBMC" & colnames(beta) != "203077630141_R01C01"]
meta.t <- meta[colnames(beta.t), ]
permOut.pbmc_r2 <- pbapply(beta.t, 1, function(x) permTest2(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) 
deltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])
permOut.pbmc_r2 <- cbind(deltaB = deltaB, pval = permOut.pbmc_r2/N) %>% as.data.frame()
permOut.pbmc_r2$qval <- p.adjust(permOut.pbmc_r2$pval)

save(permOut.wb, permOut.pbmc, 
     permOut.pbmc_r1, permOut.pbmc_r2, 
     file = "RData/11-permutation_rank_test.RData")
```

#### Precision test

```{r}
load("RData/07-Batch_corrected.RData")
beta <- beta.BC

# Set progress bar
set.seed(1234)

# Sign test
permTest <- function(cpg, group, group_var, n) {
     g <- unique(group)
     m0 <- mean(cpg[group == group_var]) # find the average DNAm level for group_var (in this case patient sample)
     m1 <- sapply(n, function(x) {
         ngroup <- sample(group, length(group)) # random shuffling of labels
        mean(cpg[ngroup == group_var]) # find the average DNAm level for randomly labelled samples
    })
     out <- c(sum(m0>=m1), sum(m0<=m1))
     return(out)
}

N <- 2000
cl <- 10

# Whole blood - sign test
beta.t <- beta[, meta$Cell_Type == "Whole Blood" & meta$Disease_Status == "Healthy Control"]
meta.t <- meta[colnames(beta.t), ]
set.seed(1234)
meta.t$Disease_Status_P1 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1)) %>% sample(., nrow(meta.t))
meta.t$Disease_Status_P2 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1)) %>% sample(., nrow(meta.t))
meta.t$Disease_Status_P3 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1)) %>% sample(., nrow(meta.t))
permOut.wb_ctr <- lapply(c("Disease_Status_P1", "Disease_Status_P2", "Disease_Status_P3"), function(p){
    test <- pbapply(beta.t, 1, function(x) permTest(x, meta.t[, p], "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
    test <- test/N
    colnames(test) <- c("Hypo_pval", "Hyper_pval")
    test$Hypo_qval <- p.adjust(test$Hypo_pval)
    test$Hyper_qval <- p.adjust(test$Hyper_pval)
    test$DeltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])
    return(test)
})
permOut.wb_ctr <- do.call(cbind, permOut.wb_ctr)
colnames(permOut.wb_ctr) <- paste0(colnames(permOut.wb_ctr), c(rep("_1", 5), rep("_2", 5), rep("_3", 5)))

# PBMC - sign test
beta.t <- beta[, meta$Cell_Type == "PBMC" & meta$Disease_Status == "Healthy Control"]
meta.t <- meta[colnames(beta.t), ]
set.seed(1234)
meta.t$Disease_Status_P1 <- c(rep("ASXL1 variant", 2), rep("Healthy Control", nrow(meta.t)-2)) %>% sample(., nrow(meta.t))
meta.t$Disease_Status_P2 <- c(rep("ASXL1 variant", 2), rep("Healthy Control", nrow(meta.t)-2)) %>% sample(., nrow(meta.t))
meta.t$Disease_Status_P3 <- c(rep("ASXL1 variant", 2), rep("Healthy Control", nrow(meta.t)-2)) %>% sample(., nrow(meta.t))
permOut.pbmc_ctr <- lapply(c("Disease_Status_P1", "Disease_Status_P2", "Disease_Status_P3"), function(p){
    test <- pbapply(beta.t, 1, function(x) permTest(x, meta.t[, p], "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
    test <- test/N
    colnames(test) <- c("Hypo_pval", "Hyper_pval")
    test$Hypo_qval <- p.adjust(test$Hypo_pval)
    test$Hyper_qval <- p.adjust(test$Hyper_pval)
    test$DeltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])
    return(test)
})
permOut.pbmc_ctr <- do.call(cbind, permOut.pbmc_ctr)
colnames(permOut.pbmc_ctr) <- paste0(colnames(permOut.pbmc_ctr), c(rep("_1", 5), rep("_2", 5), rep("_3", 5)))

# PBMC (REP1 only) - sign test
# beta.t <- beta[, meta$Cell_Type == "PBMC" & colnames(beta) != "203077630141_R06C01"]
# meta.t <- meta[colnames(beta.t), ]
# permOut.pbmc_r1 <- pbapply(beta.t, 1, function(x) permTest(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
# permOut.pbmc_r1 <- permOut.pbmc_r1/N
# colnames(permOut.pbmc_r1) <- c("Hypo_pval", "Hyper_pval")
# permOut.pbmc_r1$Hypo_qval <- p.adjust(permOut.pbmc_r1$Hypo_pval)
# permOut.pbmc_r1$Hyper_qval <- p.adjust(permOut.pbmc_r1$Hyper_pval)

# PBMC (REP2 only) - sign test
# beta.t <- beta[, meta$Cell_Type == "PBMC" & colnames(beta) != "203077630141_R01C01"]
# meta.t <- meta[colnames(beta.t), ]
# permOut.pbmc_r2 <- pbapply(beta.t, 1, function(x) permTest(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
# permOut.pbmc_r2 <- permOut.pbmc_r2/N
# colnames(permOut.pbmc_r2) <- c("Hypo_pval", "Hyper_pval")
# permOut.pbmc_r2$Hypo_qval <- p.adjust(permOut.pbmc_r2$Hypo_pval)
# permOut.pbmc_r2$Hyper_qval <- p.adjust(permOut.pbmc_r2$Hyper_pval)

# Check overlap
cutoff <- 0.05 # Likelihood for patient sample to get picked
apply(permOut.wb_ctr[, grep("qval", colnames(permOut.wb_ctr))], 2, function(x) sum(x >= cutoff)) # 48515

save(permOut.wb_ctr, permOut.pbmc_ctr, file = "RData/11-permutation sign_test_ctr.RData")

# Make heatmap
plt.hyper <- beta[rownames(permOut.pbmc)[permOut.pbmc$Hyper_pval < cutoff], ] %>% 
    .[.]
plt.hypo <- beta[rownames(permOut.pbmc)[permOut.pbmc$Hypo_pval < cutoff], ]
pheatmap(plt.hyper, show_rownames = F, scale = "row", treeheight_row = 0)
pheatmap(plt.hypo[sample(1:nrow(plt.hypo), 50000), ], show_rownames = F, scale = "row", treeheight_row = 0)

plt.hyper <- beta[rownames(permOut.WT)[permOut.WT$Hyper_pval < cutoff], ]
plt.hypo <- beta[rownames(permOut.WT)[permOut.WT$Hypo_pval < cutoff], ]
pheatmap(plt.hyper[sample(1:nrow(plt.hypo), 50000), ], show_rownames = F, scale = "row", treeheight_row = 0)
pheatmap(plt.hypo[sample(1:nrow(plt.hypo), 50000), ], show_rownames = F, scale = "row", treeheight_row = 0)

pheatmap(rbind(plt.hyper, plt.hypo), show_rownames = F, scale = "row")
```

#### Enrichment Analysis - preranked GSEA

```{r}
# Genes involved in the hits

# Enrichment analysis
```

### Epimutacions package - Require GRset input - not practical

```{r}
# library(epimutacions)
# library(minfi)
# library(ExperimentHub)
# hub <- ExperimentHub()
# query(hub, c("epimutacionsData"))
# reference_panel <- hub[["EH6691"]]
# methy <- hub[["EH6690"]]
# 
# # Regions 450K
# library("IlluminaHumanMethylation450kanno.ilmn12.hg19")
# 
# data(Locations)
# 
# ### Select CpGs (names starting by cg) in autosomic chromosomes
# locs.450 <- subset(Locations, grepl("^cg", rownames(Locations)) & chr %in% paste0("chr", 1:22))
# locs.450GR <- makeGRangesFromDataFrame(locs.450, 
#                                        start.field = "pos", 
#                                        end.field = "pos", 
#                                        strand = "*")
# locs.450GR <- sort(locs.450GR)
# mat <- matrix(0, nrow = length(locs.450GR), ncol = 2, 
#               dimnames = list(names(locs.450GR), c("A", "B")))
# 
# ## Set sample B to all 1
# mat[, 2] <- 1
# 
# ## Define model matrix
# pheno <- data.frame(var = c(0, 1))
# model <- model.matrix(~ var, pheno)
# 
# ## Run bumphunter
# bumps <- bumphunter(mat, design = model, pos = start(locs.450GR), 
#                     chr = as.character(seqnames(locs.450GR)),
#                     cutoff = 0.05)$table
# bumps.fil <- subset(bumps, L >= 3)
```



```{r}
load("RData/10-global_test.RData")
load("~/Reference/EPIC_fdat_Mreg.RData")
load("~/Reference/EPIC_Annotation_Complete.RData")
source(paste0(home, "Functions/plotmultigeneregion.R"))
source(paste0(home, "Functions/plotGeneRegion.R"))

fdat <- merge(ASXL1.test, fdat_EPIC_Mreg[, c("CHR", "MAPINFO", "Mreg", "UCSC_REFGENE_NAME", "GENCODECOMPV12_NAME")], by = 0)
rownames(fdat) <- fdat$Row.names

# Take the biggest DMRs
ASXL1.test.dmr <- table(fdat[fdat$IQR.test.wb == "s", "Mreg"]) %>% sort(decreasing = T) %>% .[.>=3] %>% names()
length(ASXL1.test.dmr)
ASXL1.test.dmr <- table(fdat[test, "Mreg"]) %>% sort(decreasing = T) %>% .[.>=3] %>% names()
ASXL1.test.dmr.gene <- sapply(ASXL1.test.dmr, function(x) {
    return(fdat[fdat$Mreg == x, "UCSC_REFGENE_NAME"] %>% 
               strsplit2(., ";") %>%
               as.vector() %>%
               unique())
}) 
dmr.genes <- do.call(c, ASXL1.test.dmr.gene)
toString(shQuote(dmr.genes))
title <- c('MICALCL', '', 'CLEC9A', 'BAIAP2', 'MAST3', 'LOC401127')
ASXL1.test.dmr.genefdat <- fdat[fdat$Mreg %in% ASXL1.test.dmr, ]

mcolor <- c("#fc8d62", "light gray")
malpha <- c(1, 0.3)
plotmultigeneregion(bt = beta, # [, meta$Disease_Status == "Healthy Control" | rownames(meta) == "203077630141_R07C01"] 
                    fdat = ASXL1.test.dmr.genefdat,
                    plist = ASXL1.test.dmr,
                    colorvar = meta$Disease_Status, # [meta$Disease_Status == "Healthy Control" | rownames(meta) == "203077630141_R07C01"]
                    cscale = mcolor, ascale = malpha, 
                    hide.legend = T, byvar = "Mreg", 
                    title = title, nCol = 4)
```


```{r}
plotmultigeneregion <- function(bt, fdat, plist, colorvar, cscale, ascale, hide.legend = F, byvar = c("Mreg", "UCSC_REFGENE_NAME"), title, nCol = NULL) {
    library(gridExtra)
    source(paste0(home, "Functions/plotGeneRegion.R"))
    
    diffregion <- lapply(1:length(plist), function(p) {
        genefdat <- fdat[grep(plist[p], fdat[, byvar]), ]
        if (length(unique(genefdat$CHR)) > 1) {
            chr <- names(sort(table(genefdat$CHR), decreasing = T))[1]
            genefdat <- genefdat[genefdat$CHR == chr, ]
        } 
        return(plotGeneRegion(x = bt, y = colorvar, z = genefdat, tss = 0, title = title[p], cscale, ascale, HL = hide.legend))
    })
    return(grid.arrange(grobs = diffregion, ncol = nCol))
}

```


### Limma

```{r}
load("RData/07-Batch_corrected.RData")
load("../../Reference/EPIC_fdat_Mreg.RData")

beta <- beta.BC

# Check WB patient vs HC - Limma
beta.t <- beta[, meta$Cell_Type == "Whole Blood"]
meta.t <- meta[colnames(beta.t), ]
meta.t$Disease_Status <- factor(meta.t$Disease_Status, levels = c("Healthy Control", "ASXL1 variant"))

design <- model.matrix(~Disease_Status + Age + Sex, data = meta.t)
fit <- lmFit(beta.t, design)
fit <- eBayes(fit)
limmaOut <- topTable(fit, coef = 2, adjust = "BH", number = nrow(beta.t))
limmaOut.wb <- merge(limmaOut, 
                     fdat_EPIC_Mreg[, c("CHR", "MAPINFO", "Mreg", "Mreg_plt", 
                                        "UCSC_REFGENE_NAME", "GENCODECOMPV12_NAME")], 
                     by = 0) %>%
    .[order(.$P.Value, decreasing = F), ]
rownames(limmaOut.wb) <- as.character(limmaOut.wb$Row.names)
dim(limmaOut.wb[limmaOut.wb$adj.P.Val < 0.05, ])
dim(limmaOut.wb_down <- limmaOut.wb[limmaOut.wb$logFC < 0 & limmaOut.wb$adj.P.Val < 0.05, ]) #31480    
dim(limmaOut.wb_up <- limmaOut.wb[limmaOut.wb$logFC > 0 & limmaOut.wb$adj.P.Val < 0.05, ]) #19405            


# Check PBMC patient vs HC - Limma
beta.t <- beta[, meta$Cell_Type == "PBMC"]
meta.t <- meta[colnames(beta.t), ]
meta.t$Disease_Status <- factor(meta.t$Disease_Status, levels = c("Healthy Control", "ASXL1 variant"))

design <- model.matrix(~Disease_Status + Age + Sex, data = meta.t)
fit <- lmFit(beta.t, design)
fit <- eBayes(fit)
limmaOut <- topTable(fit, coef = 2, adjust = "BH", number = nrow(beta.t))
limmaOut.pbmc <<- merge(limmaOut, 
                     fdat_EPIC_Mreg[, c("CHR", "MAPINFO", "Mreg", "Mreg_plt", 
                                        "UCSC_REFGENE_NAME", "GENCODECOMPV12_NAME")], 
                     by = 0) %>%
    .[order(.$P.Value, decreasing = F), ]
rownames(limmaOut.pbmc) <- as.character(limmaOut.pbmc$Row.names)
dim(limmaOut.pbmc[limmaOut.pbmc$adj.P.Val < 0.05, ])
dim(limmaOut.pbmc_down <- limmaOut.pbmc[limmaOut.pbmc$logFC < 0 & limmaOut.pbmc$adj.P.Val < 0.05, ]) #7910   
dim(limmaOut.pbmc_up <- limmaOut.pbmc[limmaOut.pbmc$logFC > 0 & limmaOut.pbmc$adj.P.Val < 0.05, ]) #10671    

# Check overlap between EV and WT
length(intersect(limmaOut.wb_up$Row.names, limmaOut.pbmc_up$Row.names)) #2
length(intersect(limmaOut.wb_down$Row.names, limmaOut.pbmc_down$Row.names)) #3
limmaOut.shared <- c(intersect(limmaOut.EV_up$Row.names, limmaOut.WT_up$Row.names), 
                     intersect(limmaOut.EV_down$Row.names, limmaOut.WT_down$Row.names))

# Save results
save(limmaOut.wb, limmaOut.pbmc, file = "RData/12-Limma_test.RData")
```

#### Precision test - use HC

```{r}
load("RData/07-Batch_corrected.RData")
load("../../Reference/EPIC_fdat_Mreg.RData")

beta <- beta.BC

# Check WB patient vs HC - Limma
beta.t <- beta[, meta$Cell_Type == "Whole Blood" & meta$Disease_Status == "Healthy Control"]
meta.t <- meta[colnames(beta.t), ]
set.seed(1234)
meta.t$Disease_Status_P1 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1)) %>% 
    sample(., nrow(meta.t)) %>% 
    factor(., levels = c("Healthy Control", "ASXL1 variant"))
meta.t$Disease_Status_P2 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1)) %>% 
    sample(., nrow(meta.t)) %>% 
    factor(., levels = c("Healthy Control", "ASXL1 variant"))
meta.t$Disease_Status_P3 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1))%>% 
    sample(., nrow(meta.t)) %>% 
    factor(., levels = c("Healthy Control", "ASXL1 variant"))
limmaOut.wb_ctr <- lapply(c("Disease_Status_P1", "Disease_Status_P2", "Disease_Status_P3"), function(p){
    design <- model.matrix(~get(p) + Age + Sex, data = meta.t)
    fit <- lmFit(beta.t, design)
    fit <- eBayes(fit)
    limmaOut <- topTable(fit, coef = 2, adjust = "BH", number = nrow(beta.t))
    limmaOut <- merge(limmaOut,
                      fdat_EPIC_Mreg[, c("CHR", "MAPINFO", "Mreg", "Mreg_plt",
                                         "UCSC_REFGENE_NAME", "GENCODECOMPV12_NAME")],
                      by = 0) %>%
        .[order(.$P.Value, decreasing = F), ]
    rownames(limmaOut) <- as.character(limmaOut$Row.names)
    return(limmaOut)
})
cutoff.pv <- 0.05
cutoff.db <- 0.05
sum(limmaOut.wb_ctr$adj.P.Val < cutoff.pv & abs(limmaOut.wb_ctr$logFC) >= cutoff.db)
sum(limmaOut.wb_ctr$logFC < 0 & limmaOut.wb_ctr$adj.P.Val < 0.05) #
sum(limmaOut.wb_ctr$logFC > 0 & limmaOut.wb_ctr$adj.P.Val < 0.05) # 

# Check PBMC patient vs HC - Limma
beta.t <- beta[, meta$Cell_Type == "PBMC" & meta$Disease_Status == "Healthy Control"]
meta.t <- meta[colnames(beta.t), ]
set.seed(1234)
meta.t$Disease_Status_P1 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1)) %>% 
    sample(., nrow(meta.t)) %>% 
    factor(., levels = c("Healthy Control", "ASXL1 variant"))
meta.t$Disease_Status_P2 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1)) %>% 
    sample(., nrow(meta.t)) %>% 
    factor(., levels = c("Healthy Control", "ASXL1 variant"))
meta.t$Disease_Status_P3 <- c("ASXL1 variant", rep("Healthy Control", nrow(meta.t)-1))%>% 
    sample(., nrow(meta.t)) %>% 
    factor(., levels = c("Healthy Control", "ASXL1 variant"))
limmaOut.pbmc_ctr <- lapply(c("Disease_Status_P1", "Disease_Status_P2", "Disease_Status_P3"), function(p){
    design <- model.matrix(~get(p) + Age + Sex, data = meta.t)
    fit <- lmFit(beta.t, design)
    fit <- eBayes(fit)
    limmaOut <- topTable(fit, coef = 2, adjust = "BH", number = nrow(beta.t))
    limmaOut <- merge(limmaOut,
                      fdat_EPIC_Mreg[, c("CHR", "MAPINFO", "Mreg", "Mreg_plt",
                                         "UCSC_REFGENE_NAME", "GENCODECOMPV12_NAME")],
                      by = 0) %>%
        .[order(.$P.Value, decreasing = F), ]
    rownames(limmaOut) <- as.character(limmaOut$Row.names)
    return(limmaOut)
})
cutoff.pv <- 0.05
cutoff.db <- 0.05
sum(limmaOut.pbmc_ctr$adj.P.Val < cutoff.pv & abs(limmaOut.pbmc_ctr$logFC) >= cutoff.db)
sum(limmaOut.pbmc_ctr$logFC < 0 & limmaOut.pbmc_ctr$adj.P.Val < 0.05) #
sum(limmaOut.pbmc_ctr$logFC > 0 & limmaOut.pbmc_ctr$adj.P.Val < 0.05) # 

save(limmaOut.pbmc_ctr, limmaOut.wb_ctr, file = "RData/12-Limma_test_ctr.RData")
```


#### Plotting Limma's hits

##### Heatmap

```{r}
beta <- beta.BC

hc_wb <- sample(rownames(meta)[meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "Whole Blood"], 20)
p_wb <- rownames(meta)[meta$Disease_Status == "ASXL1 variant" & meta$Cell_Type == "Whole Blood"]
plt <- beta[rownames(limmaOut.wb)[limmaOut.wb$adj.P.Val < 0.05], c(hc_wb, p_wb)]
anno <- meta[colnames(plt), c("Disease_Status", "Sex", "Cohort")]

pheatmap(plt[sample(1:nrow(plt), 50000),], # cannot plot all 
         show_rownames = F, 
         show_colnames = F,
         scale = "row", 
         treeheight_row = 0, 
         cluster_cols = F,
         cluster_rows = T, 
         annotation_col = anno)
pheatmap(plt[sample(1:nrow(plt), 50000),], # cannot plot all 
         show_rownames = F, 
         show_colnames = F,
         scale = "none", 
         treeheight_row = 0, 
         cluster_cols = F,
         cluster_rows = T, 
         annotation_col = anno) # do not show clustering

hc_pbmc <- sample(rownames(meta)[meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "PBMC"], 20)
p_pbmc <- rownames(meta)[meta$Disease_Status == "ASXL1 variant" & meta$Cell_Type == "PBMC"]plt <- beta[, c(hc_pbmc, p_pbmc)]
plt <- beta[rownames(limmaOut.pbmc)[limmaOut.pbmc$adj.P.Val < 0.05], c(hc_pbmc, p_pbmc)]
anno <- meta[colnames(plt), c("Disease_Status", "Sex", "Cohort")]

pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         show_colnames = T, 
         scale = "row", 
         treeheight_row = 0, 
         cluster_cols = T,
         cluster_rows = T) # do not show clustering
pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         show_colnames = F, 
         scale = "none", 
         treeheight_row = 0, 
         cluster_cols = T,
         cluster_rows = T) # do not show clustering
```

##### Volcano Plot

```{r}
### Volcano plot
source("../../Functions/makeVolcano2.R")
makeVolcano2(name = rownames(limmaOut.wb), 
             pvalue = limmaOut.wb$adj.P.Val, 
             deltabeta = limmaOut.wb$logFC, 
             ymax = 70, 
             alpha = 0.8, 
             palette = 3)
makeVolcano2(name = rownames(limmaOut.pbmc), 
             pvalue = limmaOut.pbmc$adj.P.Val, 
             deltabeta = limmaOut.pbmc$logFC, 
             ymax = 70, 
             alpha = 0.8, 
             palette = 3)
```


```{r}
hc_wb <- sample(rownames(meta)[meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "Whole Blood"], 20)
hc_pbmc <- sample(rownames(meta)[meta$Disease_Status == "Healthy Control" & meta$Cell_Type == "PBMC"], 20)
p_wb <- rownames(meta)[meta$Disease_Status == "ASXL1 variant" & meta$Cell_Type == "Whole Blood"]
p_pbmc <- rownames(meta)[meta$Disease_Status == "ASXL1 variant" & meta$Cell_Type == "PBMC"]

plt <- beta[, c(hc_wb, p_wb)]
plt <- plt[rownames(limmaOut.WB), ] #%>% 
    # .[order(.[, "P_EV"]), ] 
    # using the SD5 definition
pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         show_colnames = F, 
         scale = "row", 
         treeheight_row = 0, 
         cluster_cols = T,
         cluster_rows = T) # do not show clustering
pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         show_colnames = F, 
         scale = "none", 
         treeheight_row = 0, 
         cluster_cols = T,
         cluster_rows = T) # do not show clustering



plt <- beta[limmaOut.shared, ] #%>% 
    # .[order(.[, "P_EV"]), ] 
    # using the SD5 definition
pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         scale = "row", 
         treeheight_row = 0, 
         cluster_cols = F,
         cluster_rows = T) # do not show clustering
pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         scale = "none", 
         treeheight_row = 0, 
         cluster_cols = F,
         cluster_rows = T) # do not show clustering
```





#### Enrichment analysis on Limma's result

```{r}
load("RData/12-Limma_test.RData")
load("../../Reference/EPIC_fdat_Mreg.RData")

genes_wb <- limmaOut.wb$UCSC_REFGENE_NAME %>% 
    strsplit2(., ";") %>%
    as.vector() %>%
    unique() %>%
    .[. != ""]
genes_wb_up <- limmaOut.wb$UCSC_REFGENE_NAME[limmaOut.wb$logFC > 0] %>% 
    strsplit2(., ";") %>%
    as.vector() %>%
    unique() %>%
    .[. != ""]
genes_wb_down <- limmaOut.wb$UCSC_REFGENE_NAME[limmaOut.wb$logFC < 0] %>% 
    strsplit2(., ";") %>%
    as.vector() %>%
    unique() %>%
    .[. != ""]

genes_pbmc <- limmaOut.pbmc$UCSC_REFGENE_NAME %>% 
    strsplit2(., ";") %>%
    as.vector() %>%
    unique() %>%
    .[. != ""]
genes_pbmc_up <- limmaOut.pbmc$UCSC_REFGENE_NAME[limmaOut.pbmc$logFC > 0] %>% 
    strsplit2(., ";") %>%
    as.vector() %>%
    unique() %>%
    .[. != ""]
genes_pbmc_down <- limmaOut.pbmc$UCSC_REFGENE_NAME[limmaOut.pbmc$logFC < 0] %>% 
    strsplit2(., ";") %>%
    as.vector() %>%
    unique() %>%
    .[. != ""]

# Background genes measured on the array
genes_universe <- fdat_EPIC_Mreg$UCSC_REFGENE_NAME %>% 
    strsplit2(., ";") %>%
    as.vector() %>%
    unique() %>%
    .[. != ""] 

# Hallmark enrichment - want to check IL2-STAT5 pathway
h_gene_sets <- msigdbr(species = "human", category = "H")
msigdbr_t2g <- h_gene_sets %>% 
    dplyr::distinct(gs_name, gene_symbol) %>% 
    as.data.frame 
(pathway_wb <- enricher(gene = genes_wb, TERM2GENE = msigdbr_t2g, universe = genes_universe)) #"HALLMARK_MYOGENESIS" "HALLMARK_UV_RESPONSE_DN" "HALLMARK_APICAL_JUNCTION" "HALLMARK_ESTROGEN_RESPONSE_EARLY" "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" "HALLMARK_KRAS_SIGNALING_DN" "HALLMARK_HEDGEHOG_SIGNALING" "HALLMARK_COAGULATION" "HALLMARK_NOTCH_SIGNALING" 
(pathway_wb_up <- enricher(gene = genes_wb_up, TERM2GENE = msigdbr_t2g, universe = genes_universe)) #"HALLMARK_ESTROGEN_RESPONSE_EARLY" "HALLMARK_KRAS_SIGNALING_DN" "HALLMARK_APICAL_JUNCTION"                   "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" "HALLMARK_WNT_BETA_CATENIN_SIGNALING" "HALLMARK_ANGIOGENESIS" "HALLMARK_UV_RESPONSE_DN" "HALLMARK_ESTROGEN_RESPONSE_LATE" "HALLMARK_NOTCH_SIGNALING"
(pathway_wb_down <- enricher(gene = genes_wb_down, TERM2GENE = msigdbr_t2g, universe = genes_universe)) #"HALLMARK_MYOGENESIS" "HALLMARK_UV_RESPONSE_DN" "HALLMARK_APICAL_JUNCTION" "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" "HALLMARK_COAGULATION" "HALLMARK_APICAL_SURFACE"

(pathway_pbmc <- enricher(gene = genes_pbmc, TERM2GENE = msigdbr_t2g, universe = genes_universe)) #  "HALLMARK_UV_RESPONSE_DN" "HALLMARK_APICAL_JUNCTION"
(pathway_pbmc_up <- enricher(gene = genes_pbmc_up, TERM2GENE = msigdbr_t2g, universe = genes_universe)) #"HALLMARK_UV_RESPONSE_DN"
(pathway_pbmc_down <- enricher(gene = genes_wb_down, TERM2GENE = msigdbr_t2g, universe = genes_universe)) #"HALLMARK_MYOGENESIS"  "HALLMARK_UV_RESPONSE_DN" "HALLMARK_APICAL_JUNCTION" "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" "HALLMARK_COAGULATION" "HALLMARK_APICAL_SURFACE"

# Plot enrichment
pathway_wb <- pathway_wb@result
pathway_wb$log10Padj <- log10(pathway_wb$p.adjust)*-1
ggplot(pathway_wb[pathway_wb$p.adjust < 0.05, ], aes(reorder(ID, log10Padj), log10Padj)) + geom_bar(stat = "identity") + coord_flip() + theme_bw() + labs(y = "Adjusted P-value (-log10)", x = "")
pathway_pbmc <- pathway_pbmc@result
pathway_pbmc$log10Padj <- log10(pathway_pbmc$p.adjust)*-1
ggplot(pathway_pbmc[pathway_pbmc$p.adjust < 0.05, ], aes(reorder(ID, log10Padj), log10Padj)) + geom_bar(stat = "identity") + coord_flip() + theme_bw() + labs(y = "Adjusted P-value (-log10)", x = "")
```

#### ChromHMM enrichment on Limma hits

```{r}
library(missMethyl)
library(dplyr)

load("/mnt/scratch/KoborLab/Personal_Folders/mfu/Reference/EPIC_Anno_ChromeHMM.RData") # This is where I saved an annotation file
load("../../Reference/EPIC_fdat_Mreg.RData") # You can use this for v2 or the one Beryl created or the one in shared coding resource folder for v1
anno <- fdat_EPIC_Mreg
Chrom_df <- EPIC_Anno_ChromeHMM[rownames(limmaOut.pbmc), c("Name", "E062_Mononuclear_State")] # You want to look through the colnames and pick a cell type that fit with your samples (e.g. PBMC) for the second column!
Chrom_df <- EPIC_Anno_ChromeHMM[rownames(limmaOut.wb), c("Name", "E030_Neutrophils_15_State_State")]
colnames(Chrom_df) <- c("Probe_ID", "region")
 
# This part creates an annotation that can be put into missmethyl
ChromHMMset <- lapply(unique(Chrom_df$region), function(x){
    cpgs <- Chrom_df$Probe_ID[Chrom_df$region == x]
    id <- getMappedEntrezIDs(cpgs, array.type = "EPIC")
    return(id$sig.eg)
})
names(ChromHMMset) <- unique(Chrom_df$region)
pathChromHMM <- gsameth(sig.cpg = rownames(limmaOut.wb)[limmaOut.wb$adj.P.Val < 0.05], # Change this to a vector of names of significantly differentially methylated CpGs
                           all.cpg = rownames(limmaOut.wb), # Change this to all the CpGs you are testing
                           collection = ChromHMMset, 
                           array.type = "EPIC") %>% .[order(.$FDR), ]
pathChromHMM$Term <- rownames(pathChromHMM) %>% 
    gsub("1_TssA", "Active TSS", .) %>%
    gsub("2_TssFlnk", "Flanking TSS", .) %>% 
    gsub("3_TssFlnkU", "Flanking TSS Upstream", .) %>% 
    gsub("4_TssFlnkD", "Flanking TSS Downstream", .) %>% 
    gsub("5_Tx", "Strong Transcription", .) %>% 
    gsub("6_TxWk", "Weak Transcription", .) %>% 
    gsub("7_EnhG1", "Genic Enhancer 1", .) %>% 
    gsub("8_EnhG2", "Genic Enhancer 2", .) %>% 
    gsub("9_EnhA1", "Active Enhancer 1", .) %>% 
    gsub("10_EnhA2", "Active Enhancer 2", .) %>% 
    gsub("11_EnhWk", "Weak Enhancer", .) %>% 
    gsub("12_ZNF/Rpts", "ZNF genes & repeats", .) %>% 
    gsub("13_Het", "Heterochromatin", .) %>% 
    gsub("14_TssBiv", "Bivalent/Poised TSS", .) %>% 
    gsub("15_EnhBiv", "Bivalent Enhancer", .) %>% 
    gsub("16_ReprPC", "Repressed PolyComb", .) %>% 
    gsub("17_ReprPCWk", "Weak Repressed PolyComb", .) %>% 
    gsub("18_Quies", "	Quiescent/Low", .) 
pathChromHMM$log10Padj <- log10(pathChromHMM$FDR)*-1

# This is for plotting
plt <- pathChromHMM
ggplot(plt, aes(reorder(Term, log10Padj), log10Padj)) + 
    geom_bar(stat = "identity") + 
    labs(y = "Adjusted P-value (-log10)", x = "ChromHMM 18 States Annotation") + 
    geom_hline(yintercept = log10(0.01)*-1, linetype = 'dashed', color = "gray") +
    coord_flip() + 
    theme_bw() + 
    theme(text = element_text(size = 13)) 
```

## CT-Specific Differential DNAm

```{r}
library(TOAST)
beta <- beta.BC

beta.t <- beta.BC[, meta$Cell_Type == "Whole Blood"]
meta.t <- meta[colnames(beta.t), ]
meta.t$Disease_Status <- factor(meta.t$Disease_Status, levels = c("Healthy Control", "ASXL1 variant"))
ct <- meta.t[, c("Bas", "Bnv", "CD4mem", "Bmem","CD4nv", "CD8mem", "CD8nv", 
                 "Eos", "Mono", "Neu", "NK", "Treg")] #  proportions are too low and removed
pheno1 <- meta.t[, c("Age"), drop = F]
pheno2 <- meta.t[, c("Sex"), drop = F]
# test <- pheno1
# test$Disease_Status <- c(rep("Healthy", 30), rep("Sick", (nrow(test) - 30)))
library(caret)
test <- upSample(x = t(beta.t),
                 y = meta.t$Disease_Status)    
res_cedar2 <- cedar(Y_raw = beta.t,
                    prop = ct,
                    design.1 = pheno1,
                    design.2 = pheno2,
                    factor.to.test = 'Age',
                    cutoff.tree = c('pval', 0.01),
                    cutoff.prior.prob = c('pval', 0.01),
                    parallel.core = 10)

library(TCA)
ct[ct < 0] <- 0 #If necessary, add the offset of 0.001
ct <- apply(ct, 1, function(x) x / sum(x)) %>% 
    t() %>%
    as.data.frame()
pheno1 <- model.matrix(~Disease_Status + Cell_Type + Age, data = meta.t) %>% .[, -1, drop = F]
pheno2 <- model.matrix(~Sex, data = meta.t) %>% .[, -1, drop = F]
tca_output <- tca(X = beta.t[1:200, ],
                  W = ct,
                  C1 = pheno1,
                  C2 = pheno2)

library(EpiDISH)
mod <- model.matrix(~Cell_Type + Age + Sex, data = meta.t) %>% .[, -1, drop = F]	
dmc_output <- CellDMC(beta.m = beta.t[1:100, ], 
                      pheno.v = meta.t$Disease_Status, 
                      frac.m = as.matrix(ct), 
                      cov.mod = mod)

```

#### DNAm Age - redo plotting

```{r}
load("RData/07-Batch_corrected.RData")

plt <- meta[meta$Cell_Type == "Whole Blood", ]
plt <- meta[meta$Cell_Type == "PBMC", ]
group.colors <- c("Healthy Control" = "#A3BE8CFF", 
                  "ASXL1 variant" = "#BF616AFF")

ggplot(plt, aes(Age, skinHorvath)) + 
    geom_smooth(method = "lm") + 
    geom_point(aes(color = Disease_Status)) + 
    theme_classic() + 
    scale_color_manual(values = group.colors) + 
    theme(text = element_text(size = 13))
ggplot(plt, aes(Age, Horvath)) + geom_smooth(method = "lm") + 
    geom_smooth(method = "lm") + 
    geom_point(aes(color = Disease_Status)) + 
    theme_classic() + 
    scale_color_manual(values = group.colors) + 
    theme(text = element_text(size = 13))
ggplot(plt, aes(Age, TL)) + geom_smooth(method = "lm") + 
    geom_smooth(method = "lm") + 
    geom_point(aes(color = Disease_Status)) + 
    theme_classic() + 
    scale_color_manual(values = group.colors) + 
    theme(text = element_text(size = 13))

lm(Horvath ~ Disease_Status + Age + Sex, data = plt) %>% summary() # P = 0.00884 (WB); P < 2e-16 (PBMC)
lm(TL ~ Disease_Status + Age + Sex, data = plt) %>% summary() # P = 0.0101 (WB); P = 3.14e-10 (PBMC)
```

#### Check if ASXL1 is differentially methylated

```{r}
load("RData/07-Batch_corrected.RData")
beta <- beta.BC

# WB
limmaOut.wb.s <- limmaOut.wb[limmaOut.wb$adj.P.Val < 0.05, ]
test <- limmaOut.wb.s[grep("ASXL1", limmaOut.wb.s$UCSC_REFGENE_NAME), ]
plt <- beta[rownames(test), meta$Cell_Type == "Whole Blood"]
plt <- reshape2::melt(plt)
colnames(plt) <- c("CpG", "Sample_ID", "DNAm")
plt <- left_join(plt, meta, by = join_by(Sample_ID == IDAT_ID))
ggplot(plt, aes(Disease_Status, DNAm, color= Disease_Status)) +
    geom_violin() + 
    geom_boxplot(width = 0.3, alpha = 0.2) +
    # geom_jitter(alpha = 0.2, width = 0.25) +
    scale_color_manual(values = group.colors) + 
    theme_bw() +
    theme(text = element_text(size = 15)) +
    facet_wrap(~CpG)

# PBMC
limmaOut.pbmc.s <- limmaOut.pbmc[limmaOut.pbmc$adj.P.Val < 0.05, ]
test <- limmaOut.pbmc.s[grep("ASXL1", limmaOut.pbmc.s$UCSC_REFGENE_NAME), ]
```

### Overlap - different tests

```{r}
load("RData/10-global_test_pbmc.RData")
load("RData/10-global_test_wb.RData")
load("RData/11-permutation_sign_test.RData")
load("RData/12-Limma_test.RData")

cutoff.pv <- 0.05
cutoff.db <- 0.05

# Subset to hits
limmaHits.wb <- rownames(limmaOut.wb_ctr)[limmaOut.wb_ctr$adj.P.Val < cutoff.pv & abs(limmaOut.wb_ctr$logFC) >= cutoff.db]
limmaHits.pbmc <- rownames(limmaOut.pbmc_ctr)[limmaOut.pbmc_ctr$adj.P.Val < cutoff.pv & abs(limmaOut.pbmc_ctr$logFC) >= cutoff.db]
refHits.wb <- rownames(ASXL1.test.wb)[ASXL1.test.wb$SD3.test.wb & ASXL1.test.wb$meandiff > cutoff.db]
refHits.pbmc <- rownames(ASXL1.test.pbmc)[ASXL1.test.pbmc$SD3.test.pbmc & ASXL1.test.pbmc$meandiff.avg > cutoff.db]
permHits.wb <- rownames(permOut.wb)[permOut.wb$Hypo_qval < cutoff.pv | permOut.wb$Hyper_pval < cutoff.pv]
permHits.pbmc <- rownames(permOut.pbmc)[permOut.pbmc$Hypo_pval < cutoff.pv | permOut.pbmc$Hyper_pval < cutoff.pv]

# Make Venn Diagram
allHits.wb <- list(Outlier_test = refHits.wb,
                   Permutation_test = permHits.wb,
                   Limma_test = limmaHits.wb)
allHits.pbmc <- list(Outlier_test = refHits.pbmc,
                   Permutation_test = permHits.pbmc,
                   Limma_test = limmaHits.pbmc)

plot(euler(allHits.wb, shape = "ellipse"), quantities = TRUE)
plot(euler(allHits.pbmc, shape = "ellipse"), quantities = TRUE)
```

# Buccal Reference

## Get buccal samples

```{r}
getSQLiteFile(destdir = "~/KoborLab/kobor_space/mfu/GEO")
con <- dbConnect(SQLite(), "~/KoborLab/kobor_space/mfu/GEO/GEOmetadb.sqlite")
dbListFields(con, "gsm")

x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL13534' and  organism_ch1 = 'Homo sapiens'"))
sampl <- x
x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL21145' and  organism_ch1 = 'Homo sapiens'"))
sampl2 <- x
x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL16304' and  organism_ch1 = 'Homo sapiens'"))
sampl3 <- x
x <- dbGetQuery(con, paste0("select ", paste0(dbListFields(con, "gsm"), collapse = ", "), " from gsm where gpl = 'GPL23976' and  organism_ch1 = 'Homo sapiens'"))
sampl4 <- x

allsamp <- rbind(sampl, sampl2)
allsamp2 <- rbind(sampl3, sampl4)
allsamp <- rbind(allsamp, allsamp2)

bucID <- c(grep("[Bb]uccal|oral swab|BEC", allsamp$characteristics_ch1), 
           grep("[Bb]uccal|oral swab|BEC", allsamp$source_name_ch1))
bucID <- unique(bucID)
bucsamp <- allsamp[bucID, ]

bucGEO <- strsplit(paste0(unique(bucsamp$series_id), collapse = ","), ",")[[1]]
Sys.setenv("VROOM_CONNECTION_SIZE" = 13107200)
getOption("download.file.method.GEOquery")
options(download.file.method.GEOquery = "libcurl")
dest <- "~/KoborLab/kobor_space/mfu/GEO/Buccal/"

BUC <- sapply(bucGEO, function (x) getGEO(x, destdir = dest))
save(BUC, file = "~/KoborLab/kobor_space/mfu/GEO/Buccal/BUC.RData")
```

```{r}
# "GSE41114" - age ?   # squamous cell carcinoma tumor
# Samp <- BUC$GSE41114$GSE41114_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE41114"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE41114.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE41114.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE41114", baseDir = bsdir)
# GSE41114 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE41114/", recursive = T, extended = T)

# "GSE41117" - age ? # squamous cell carcinoma tumor
# Samp <- BUC$GSE41117$`GSE41117-GPL13534_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE41117"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE41117.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE41117.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE41117", baseDir = bsdir)
# GSE41117 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE41117/", recursive = T, extended = T)

# "GSE42700" - age ?
# Samp <- BUC$GSE42700$GSE42700_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE42700"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- gsub("18 months", "1.5", pd$`age:ch1`) %>% gsub("birth", "0", .) %>% as.numeric(.)
# pd$sex <- gsub("female", "F", pd$`gender:ch1`) %>% gsub("male", "M", .) 
# # pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# # pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- "Healthy Control"
# GSE42700.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "sibship:ch1", "twin:ch1", "monochorionic:ch1", "monozygotic:ch1")]
# colnames(GSE42700.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Twin_ID", "Twin_Individual", "Twin_Monochorionic", "Twin_Monozygotic")
# getGEOSuppFiles("GSE42700", baseDir = bsdir)
# GSE42700 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE42700/", recursive = T, extended = T)

# "GSE48472" - age ?
# Samp <- BUC$GSE48472$GSE48472_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE48472"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE48472.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE48472.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE48472", baseDir = bsdir)
# GSE48472 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE48472/", recursive = T, extended = T)

# "GSE60275" - XCI
# Samp <- BUC$GSE60275$GSE60275_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE60275"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE60275.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE60275.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE60275", baseDir = bsdir)
# GSE60275 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE60275/", recursive = T, extended = T)

# "GSE64491" - age 112
# Samp <- BUC$GSE64491$GSE64491_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE64491"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE64491.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE64491.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE64491", baseDir = bsdir)
# GSE64491 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE64491/", recursive = T, extended = T)

# "GSE64511" - age 15-114
# Samp <- BUC$GSE64511$GSE64511_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE64511"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE64511.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE64511.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE64511", baseDir = bsdir)
# GSE64511 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE64511/", recursive = T, extended = T)

# "GSE68379" - not buccal
# Samp <- BUC$GSE68379$GSE68379_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE68379"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE68379.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE68379.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE68379", baseDir = bsdir)
# GSE68379 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE68379/", recursive = T, extended = T)

# "GSE80261" - age 
Samp <- BUC$GSE80261$GSE80261_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE80261"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- pd$`Sex:ch1` %>% gsub("Female", "F", .) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease status:ch1`
GSE80261.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "disease subgroup:ch1", "self-reported ethnicity:ch1", "chip", "position")]
colnames(GSE80261.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Disease_Subtype", "Ethnicity_Self_Reported", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE80261", baseDir = bsdir)
# GSE80261 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE80261/", recursive = T, extended = T)

# "GSE92843" - cell line
# Samp <- BUC$GSE92843$GSE92843_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE92843"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE92843.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE92843.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE92843", baseDir = bsdir)
# GSE92843 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE92843/", recursive = T, extended = T)

# "GSE101673" - not buccal
# Samp <- BUC$GSE101673$GSE101673_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE101673"
# s$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE101673.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE101673.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE101673", baseDir = bsdir)
# GSE101673 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE101673/", recursive = T, extended = T)

# "GSE101864" - not buccal
# Samp <- BUC$GSE101864$`GSE101864-GPL13534_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE101864"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE101864.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE101864.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE101864", baseDir = bsdir)
# GSE101864 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE101864/", recursive = T, extended = T)

# "GSE108187" - cell line
# Samp <- BUC$GSE108187$GSE108187_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE108187"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$position <- pd$title %>% gsub(".*,.", "", .) %>% gsub(".*_", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE108187.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status")]
# colnames(GSE108187.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE108187", baseDir = bsdir)
# GSE108187 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE108187/", recursive = T, extended = T)

# "GSE109042" - age 22 - 93
Samp <- BUC$GSE109042$GSE109042_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE109042"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease state:ch1`
GSE109042.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "disease subgroup:ch1", "self-reported ethnicity:ch1", "chip", "position")]
colnames(GSE109042.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Disease_Subtype", "Ethnicity_Self_Reported", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE109042", baseDir = bsdir)
# GSE109042 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE109042/", recursive = T, extended = T)

# "GSE137503" - PedBE - The supplementary files do not actually match the sample sheet
# Samp <- BUC$GSE137503$`GSE137503-GPL13534_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE137503"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .)
# pd$disease_status <- "Healthy Control"
# GSE137503.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
# colnames(GSE137503.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status")
# getGEOSuppFiles("GSE137503", baseDir = bsdir)
# GSE137503 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137503/", recursive = T, extended = T)

# "GSE124366" - CARE - keep for the reps
Samp <- BUC$GSE124366$GSE124366_series_matrix.txt.gz
pd <- pData(Samp)
pd <- pd[pd$`tissue:ch1` == "Buccal", ]
pd$cohort <- "GSE124366"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age at sample collection (years):ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE124366.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "Sex:ch1", "disease_status", "chip", "position")]
colnames(GSE124366.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE124366", baseDir = bsdir)
# GSE124366 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE124366/", recursive = T, extended = T)

# "GSE137495" - Apron - in GSE137503
Samp <- BUC$GSE137495$GSE137495_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137495"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137495.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137495.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137495", baseDir = bsdir)
# GSE137495 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137495/", recursive = T, extended = T)

# "GSE137688" - MAVAN, maternal cohort
# Samp <- BUC$GSE137688$GSE137688_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE137688"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
# pd$disease_status <- "Healthy Control"
# GSE137688.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
# colnames(GSE137688.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137688", baseDir = bsdir)
# GSE137688 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137688/", recursive = T, extended = T)

# "GSE137682" - BIBO in PedBE
Samp <- BUC$GSE137682$GSE137682_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137682"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137682.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137682.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137682", baseDir = bsdir)
# GSE137682 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137682/", recursive = T, extended = T)

# "GSE137502" - WSFW
Samp <- BUC$GSE137502$GSE137502_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137502"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137502.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137502.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137502", baseDir = bsdir)
# GSE137502 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137502/", recursive = T, extended = T)

# "GSE137841" - UCI
Samp <- BUC$GSE137841$GSE137841_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137841"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`) / 365
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137841.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137841.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137841", baseDir = bsdir)
# GSE137841 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137841/", recursive = T, extended = T)

# "GSE137884" - CARE in PedBE
Samp <- BUC$GSE137884$GSE137884_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137884"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137884.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137884.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137884", baseDir = bsdir)
# GSE137884 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137884/", recursive = T, extended = T)

# "GSE137894" - MAVAN
Samp <- BUC$GSE137894$GSE137894_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137894"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137894.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137894.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137894", baseDir = bsdir)
# GSE137894 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137894/", recursive = T, extended = T)

# "GSE137898" - MAVAN II
Samp <- BUC$GSE137898$GSE137898_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137898"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137898.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137898.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137898", baseDir = bsdir)
# GSE137898 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137898/", recursive = T, extended = T)

# "GSE137903" - GECKO
Samp <- BUC$GSE137903$GSE137903_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137903"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137903.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137903.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137903", baseDir = bsdir)
# GSE137903 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137903/", recursive = T, extended = T)

# "GSE137904" - GUSTO
Samp <- BUC$GSE137904$GSE137904_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE137904"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".* ", "", pd$title) %>% gsub("_.*", "", .) 
pd$position <- gsub(".* ", "", pd$title) %>% gsub(".*_", "", .)
pd$disease_status <- "Healthy Control"
GSE137904.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position")]
colnames(GSE137904.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE137904", baseDir = bsdir)
# GSE137904 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE137904/", recursive = T, extended = T)

# "GSE112893" - age 0.250 - 49.583
Samp <- BUC$GSE112893$GSE112893_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE112893"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- pd$`disease group:ch1` %>% gsub("NESAM", "Non-Edematous Severe Acute Malnutrition", .) %>% gsub("ESAM", "Edematous Severe Acute Malnutrition", .)
GSE112893.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "gender:ch1", "disease_status", "chip", "position", "diagnosis:ch1", "nationality:ch1")]
colnames(GSE112893.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Sample_Group", "Ethnicity")
# getGEOSuppFiles("GSE112893", baseDir = bsdir)
# GSE112893 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE112893/", recursive = T, extended = T)

# "GSE156669" - age?
# Samp <- BUC$GSE156669$GSE156669_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE156669"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE156669.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE156669.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE156669", baseDir = bsdir)
# GSE156669 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE156669/", recursive = T, extended = T)

# "GSE111165" - not pediatric
# Samp <- BUC$GSE111165$`GSE111165-GPL13534_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE111165"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE111165.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE111165.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE111165", baseDir = bsdir)
# GSE111165 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE111165/", recursive = T, extended = T)

# "GSE128821" - age 0
Samp <- BUC$GSE128821$GSE128821_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE128821"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- 0 # collected near term equivalant age
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Born less than 30 weeks postmenstrual age (PMA)"
GSE128821.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position", "inf:ch1", "bga:ch1", "nnns profile:ch1", "outborn:ch1", "riskmdb34:ch1", "rop:ch1", "sbi:ch1", "site:ch1")]
colnames(GSE128821.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Infection", "Gestational_Age", "Neonatal_Intensive_Care_Unit_Network_Neurobehavioral_Scale", "Outborn_Delivery", "Metabolic_Bone_Disease_Risk_wk34", "Retinopathy_of_Prematurity", "Serious_Bacterial_Infections", "Sample_Collection_Site")
# getGEOSuppFiles("GSE128821", baseDir = bsdir)
# GSE128821 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE128821/", recursive = T, extended = T)

# "GSE133355" - not pediatric, but relatively young cohort 28.1 ± 5.0
# Samp <- BUC$GSE133355$GSE133355_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE133355"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE133355.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE133355.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE133355", baseDir = bsdir)
# GSE133355 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE133355/", recursive = T, extended = T)

# "GSE121433", "GSE121434" - cell line
# Samp <- BUC$GSE121433$GSE121433_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE121433"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE121433.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE121433.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE121433", baseDir = bsdir)
# GSE121433 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE121433/", recursive = T, extended = T)

# "GSE147058" - close to 8.5
Samp <- BUC$GSE147058$GSE147058_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE147058"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- 8.5
pd$sex <- gsub("Female", "F", pd$`Sex:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE147058.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position", "ses.narrow.new:ch1", "logcortnm5.1_mean:ch1", "logcortnm5.2_mean:ch1", "logcortnm5.3_mean:ch1", "cnmmean2.1_mean:ch1", "cnmmean2.2_mean:ch1", "cnmmean2.3_mean:ch1", "amslopenm_mean:ch1", "pmslopenm_mean:ch1", "dayslopenm_mean:ch1")]
colnames(GSE147058.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "SES_Narrow", "Cortisol_Log_D1_mean", "Cortisol_Log_D2_mean", "Cortisol_Log_D3_mean", "Cortisol_D1_mean", "Cortisol_D2_mean", "Cortisol_D3_mean", "Cortisol_Slope_AM_mean", "Cortisol_Slope_PM_mean", "Cortisol_Slope_Days_mean")
# getGEOSuppFiles("GSE147058", baseDir = bsdir)
# GSE147058 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE147058/", recursive = T, extended = T)

# "GSE154566" - Blood not buccal
# Samp <- BUC$GSE154566$`GSE154566-GPL13534_series_matrix.txt.gz`
# pd <- pData(Samp)
# pd$cohort <- "GSE154566"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- 18
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE154566.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE154566.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE154566", baseDir = bsdir)
# GSE154566 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE154566/", recursive = T, extended = T)

# "GSE45529", "GSE46573" - age ??, few buccal samples
# Samp <- BUC$GSE45529$GSE45529_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE45529"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE45529.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE45529.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE45529", baseDir = bsdir)
# GSE45529 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE45529/", recursive = T, extended = T)

# "GSE50586" - down syndrome, not pediatric
# Samp <- BUC$GSE50586$GSE50586_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE50586"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE50586.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE50586.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE50586", baseDir = bsdir)
# GSE50586 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE50586/", recursive = T, extended = T)

# "GSE53849" - age?? probably not pediatric
# Samp <- BUC$GSE53849$GSE53849_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE53849"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE53849.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE53849.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE53849", baseDir = bsdir)
# GSE53849 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE53849/", recursive = T, extended = T)

# "GSE94876" - not pediatric
# Samp <- BUC$GSE94876$GSE94876_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE94876"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE94876.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE94876.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE94876", baseDir = bsdir)
# GSE94876 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE94876/", recursive = T, extended = T)

# "GSE94734" - age 10-12
Samp <- BUC$GSE94734$GSE94734_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE94734"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- as.numeric(pd$`age:ch1`)
pd$sex <- gsub("0", "F", pd$`gender:ch1`) %>% gsub("1", "M", .)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- "Healthy Control"
GSE94734.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "sentrix_id:ch1", "sentrix_position:ch1", "genetic ancestry:ch1", "education:ch1", "adversity:ch1", "income:ch1", "minority:ch1")]
colnames(GSE94734.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Genetic_Ancestry_PC1", "Education_Level", "Family_Psychosocial_Adversity", "Income_Level", "Minority_Group")
# getGEOSuppFiles("GSE94734", baseDir = bsdir)
# GSE94734 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE94734/", recursive = T, extended = T)

# "GSE166844" - age ??
# Samp <- BUC$GSE166844$GSE166844_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE166844"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE166844.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE166844.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE166844", baseDir = bsdir)
# GSE166844 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE166844/", recursive = T, extended = T)

# "GSE42409" - 4 samples
# Samp <- BUC$GSE42409$GSE42409_series_matrix.txt.gz
# pd <- pData(Samp)
# pd$cohort <- "GSE42409"
# pd$tissue <- "Buccal"
# pd$celltype <- "Buccal Epithelial Cells"
# pd$age <- as.numeric(pd$`age:ch1`)
# pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
# pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
# pd$disease_status <- pd$`disease state:ch1`
# GSE42409.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position")]
# colnames(GSE42409.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position")
# getGEOSuppFiles("GSE42409", baseDir = bsdir)
# GSE42409 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE42409/", recursive = T, extended = T)

# "GSE150901" - age 8
Samp <- BUC$GSE150901$GSE150901_series_matrix.txt.gz
pd <- pData(Samp)
pd$cohort <- "GSE150901"
pd$tissue <- "Buccal"
pd$celltype <- "Buccal Epithelial Cells"
pd$age <- 8 # 7.7 +- 0.7
pd$sex <- gsub("Female", "F", pd$`gender:ch1`) %>% gsub("Male", "M", .)
pd$chip <- gsub(".*GSM[0-9]{7}_", "", pd$supplementary_file) %>% gsub("_.*", "", .) 
pd$position <- gsub(".*[0-9]{10}_", "", pd$supplementary_file) %>% gsub("_.*", "", .)
pd$disease_status <- gsub("Control", "Healthy Control", pd$`fiv_group:ch1`) %>% gsub("ICSI|FIV", "Potential malformation associated with ART", .)
GSE150901.m <- pd[, c("cohort", "geo_accession", "title", "data_processing", "platform_id", "tissue", "celltype", "age", "sex", "disease_status", "chip", "position", "culture_medium:ch1", "fiv_group:ch1")]
colnames(GSE150901.m) <- c("Cohort", "GEO_Accession", "Sample_ID", "Data_Processing", "Platform_ID", "Tissue", "Cell_Type", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Culture_Medium", "Assisted_Reproductive_Technology")
# getGEOSuppFiles("GSE150901", baseDir = bsdir)
# GSE150901 <- read.metharray.exp("~/KoborLab/kobor_space/mfu/GEO/Blood/GSE150901/", recursive = T, extended = T)

dest <- "~/KoborLab/kobor_space/mfu/GEO/Buccal/"
cohorts <- c("GSE112893", "GSE124366", "GSE128821", "GSE137495", "GSE137502", "GSE137841", "GSE137884", "GSE137894", "GSE137898", "GSE137903", "GSE137904", "GSE147058", "GSE150901", "GSE80261", "GSE94734")
lapply(cohorts, function(x) getGEOSuppFiles(x, baseDir = dest))

allpd <- rbind.fill(GSE112893.m, GSE124366.m, GSE128821.m, GSE137495.m, GSE137502.m, GSE137841.m, GSE137884.m, GSE137894.m, GSE137898.m, GSE137903.m, GSE137904.m, GSE147058.m, GSE150901.m, GSE80261.m, GSE94734.m)
save(GSE112893.m, GSE124366.m, GSE128821.m, GSE137495.m, GSE137502.m, GSE137841.m, GSE137884.m, GSE137894.m, GSE137898.m, GSE137903.m, GSE137904.m, GSE147058.m, GSE150901.m, GSE80261.m, GSE94734.m, allpd, file = "RData/802-All_pd_buccal.RData")
```


```{r}
library(plyr)
load("RData/802-All_pd_buccal.RData")

# GSE112893 <- read.metharray.exp("GEO/GSE112893/", extended = T, recursive = T)
GSE112893.m$Basename <- paste0(GSE112893.m$GEO_Accession, "_", GSE112893.m$Sentrix_ID, "_", GSE112893.m$Sentrix_Position) 
# GSE124366 <- read.metharray.exp("GEO/GSE124366/", extended = T, recursive = T)
GSE124366.m$Basename <- paste0(GSE124366.m$GEO_Accession, "_", GSE124366.m$Sentrix_ID, "_", GSE124366.m$Sentrix_Position)
GSE128821.m$Basename <- paste0(GSE128821.m$GEO_Accession, "_", GSE128821.m$Sentrix_ID, "_", GSE128821.m$Sentrix_Position)
GSE128821.m <- GSE128821.m[GSE128821.m$Basename != "GSM3686020_201557540048_R07C01", ] # idat file corrupted - remove
# GSE128821_2 <- sapply(GSE128821.m$Basename, function(x){
    #print(paste0(x, ":"))
    # test <- read.metharray(paste0("GEO/GSE128821/", x))
    #print(dim(test)[1])
    # return(c(x, dim(test)))
# }) %>% t() %>% as.data.frame()
# GSE128821 <- read.metharray.exp("GEO/GSE128821/", extended = T, recursive = T, force = T)
# GSE137503 <- read.metharray.exp("GEO/GSE137503/", extended = T, recursive = T) # Not all the
# GSE137503.m$Basename <- paste0(GSE137503.m$GEO_Accession, "_", GSE137503.m$Sentrix_ID, "_", GSE137503.m$Sentrix_Position)
GSE137495.m$Basename <- paste0(GSE137495.m$Sentrix_ID, "_", GSE137495.m$Sentrix_Position)
GSE137502.m$Basename <- paste0(GSE137502.m$Sentrix_ID, "_", GSE137502.m$Sentrix_Position)
GSE137841.m$Basename <- paste0(GSE137841.m$Sentrix_ID, "_", GSE137841.m$Sentrix_Position)
GSE137884.m$Basename <- paste0(GSE137884.m$Sentrix_ID, "_", GSE137884.m$Sentrix_Position)
GSE137894.m$Basename <- paste0(GSE137894.m$Sentrix_ID, "_", GSE137894.m$Sentrix_Position)
GSE137898.m$Basename <- paste0(GSE137898.m$Sentrix_ID, "_", GSE137898.m$Sentrix_Position)
GSE137903.m$Basename <- paste0(GSE137903.m$Sentrix_ID, "_", GSE137903.m$Sentrix_Position)
GSE137904.m$Basename <- paste0(GSE137904.m$Sentrix_ID, "_", GSE137904.m$Sentrix_Position)
# GSE147058 <- read.metharray.exp("GEO/GSE147058/", extended = T, recursive = T)
GSE147058.m$Basename <- paste0(GSE147058.m$GEO_Accession, "_", GSE147058.m$Sample_ID) %>% gsub("_G.*", "", .)
# GSE150901 <- read.metharray.exp("GEO/GSE150901/", extended = T, recursive = T)
GSE150901.m$Basename <- paste0(GSE150901.m$GEO_Accession, "_", GSE150901.m$Sentrix_ID, "_", GSE150901.m$Sentrix_Position)
# GSE80261 <- read.metharray.exp("GEO/GSE80261/", extended = T, recursive = T)
GSE80261.m$Basename <- paste0(GSE80261.m$GEO_Accession, "_", GSE80261.m$Sentrix_ID, "_", GSE80261.m$Sentrix_Position)
# GSE94734 <- read.metharray.exp("GEO/GSE94734/", target = GSE94734.m, extended = T, recursive = T)
GSE94734.m$Basename <- paste0(GSE94734.m$Sentrix_ID, "_", GSE94734.m$Sentrix_Position)

load("RData/10-Biggs_patient_raw.RData")
ASXL1.m <- pData(TB)
ASXL1.m$Basename <- paste0(ASXL1.m$Sentrix_ID, "_", ASXL1.m$Sentrix_Position)
ASXL1.m$Cohort <- "ASXL1"
ASXL1.m$Platform_ID <- "GPL21145"
ASXL1.m$Disease_Status <- gsub("Pat", "ASXL1 variant", ASXL1.m$Individual) %>% gsub("Ctrl", "Healthy Control", .)
ASXL1.m <- ASXL1.m[, c("Cohort", "Sample_ID", "Platform_ID", "Tissue", "Age", "Sex", "Disease_Status", "Sentrix_ID", "Sentrix_Position", "Basename")]

allpd <- rbind.fill(GSE112893.m, GSE124366.m, GSE128821.m, GSE137495.m, GSE137502.m, GSE137841.m, GSE137884.m, GSE137894.m, GSE137898.m, GSE137903.m, GSE137904.m, GSE147058.m, GSE150901.m, GSE80261.m, GSE94734.m, ASXL1.m)


allpd_450 <- allpd[allpd$Platform_ID == "GPL13534", ]
allpd_EPIC <- allpd[allpd$Platform_ID %in% c("GPL21145", "GPL23976"), ]

HM450 <- read.metharray.exp("raw/GPL13534/IDAT/", targets = allpd_450, extended = T, recursive = T)
HM450_2 <- sapply(allpd_450$Basename, function(x){
    # print(paste0(x, ":"))
    test <- read.metharray(paste0("raw/GPL13534/IDAT/", x))
    # print(dim(test)[1])
    return(c(x, dim(test)))
}) %>% t() %>% as.data.frame()

EPIC <- read.metharray.exp("raw/GPL21145/IDAT/", targets = allpd_EPIC, extended = T, recursive = T, force = T)
EPIC_2 <- sapply(allpd_EPIC$Basename, function(x){
    # print(paste0(x, ":"))
    test <- read.metharray(paste0("raw/GPL21145/IDAT/", x))
    # print(dim(test)[1])
    return(c(x, dim(test)))
}) %>% t() %>% as.data.frame()

save(HM450, EPIC, file = "RData/803-Buccal_raw.RData")
``` 

### Outlier removal + Normalization

```{r}
### Quality check - remove low quality control probes
MSet.raw <- preprocessRaw(HM450)
MSet.raw <- preprocessRaw(EPIC)
qc.raw <- getQC(MSet.raw)
plotQC(qc.raw, badSampleCutoff = 10.5)

qc <- as.data.frame(qc.raw@listData)
rownames(qc) <- qc.raw@rownames
qc.bad <- qc[qc$mMed < 10.5 & qc$uMed < 10.5, ]
out1 <- rownames(qc.bad)

MSet.raw <- addQC(MSet.raw, qc.raw)
pd <- pData(MSet.raw)
pd$out1 <- 0
pd[pd$mMed < 10.5 & pd$uMed < 10.5, "out1"] <- 1
densityPlot(MSet.raw, sampGroups = pd$out1)

# pData(HM450) <- pd
# HM450 <- HM450[, pd$out1 == 0]
# HM450.FN <- preprocessFunnorm(HM450)
# pData(EPIC) <- pd
# EPIC <- EPIC[, pd$out1 == 0]
# EPIC.FN <- preprocessFunnorm(EPIC)

source(paste0(home, "Functions/DetPvalTestRG.R"))
out2_450k <- DetPBisCTest(HM450)
# Sample(s) GSM3530006_9274651149_R05C02, GSM2122938_9247377165_R06C02 have average detection p-values greater than 0.01!
# Sample(s) GSM3090750_9426020116_R04C01, 100973330053_R06C02, 9343114076_R01C02, 100946150106_R04C02, 9274651104_R01C01, 9285451068_R06C01, 9296930127_R04C02, GSM2122989_9257625053_R03C02 have bisulfite conversion rate lower than 75!
out2_EPIC <- DetPBisCTest(EPIC)
# Sample(s) GSM3686308_201557560027_R02C01 have average detection p-values greater than 0.01!
# Sample(s) 201490030052_R04C01 have bisulfite conversion rate lower than 75!

save(HM450.FN, EPIC.FN, file = "RData/804-Buccal_FN.RData")
save(out2_450k, out2_EPIC, file = "RData/806-outliers2.RData")
```

```{r}
identical(length(allpd$Basename), length(c(colnames(HM450.FN), colnames(EPIC.FN))))
allpd <- allpd[allpd$Basename %in% c(colnames(HM450), colnames(EPIC)), ]
rownames(allpd) <- allpd$Basename

# Combine 450k and EPIC
CpG <- Reduce(intersect, list(rownames(HM450.FN), 
                              rownames(EPIC.FN)))
allbt <- cbind(getBeta(HM450.FN)[CpG, ], getBeta(EPIC.FN)[CpG, ])
allsex <- rbind(getSex(HM450.FN), getSex(EPIC.FN)) %>% as.data.frame()

allbt <- allbt[, colnames(allbt) %in% rownames(allpd)]
allpd <- allpd[colnames(allbt), ]
allsex <- allsex[colnames(allbt), ]
allpd$Estimated_Sex <- allsex$predictedSex

identical(rownames(allpd), colnames(allbt))
table(allpd$Cohort)

PedRef.B <- new("MethyLumiSet",
                betas = as.matrix(allbt), 
                phenoData = as(allpd, "AnnotatedDataFrame"), 
                featureData = TB@featureData[rownames(allbt)], 
                experimentData = TB@experimentData, 
                .__classVersion__ = TB@.__classVersion__)

save(allbt, allpd, PedRef.B, file = "RData/805-All_GEO_reference_buccal.RData")
```

```{r}
load("RData/805-All_GEO_reference_buccal.RData")

MLset <- PedRef.B

# out2 <- gsub("GSM[0-9]{7}_", "", out2)
out3 <- detectOutlier(as.matrix(betas(MLset))) %>% which(.)
save(out3, file = "RData/806-outliers3.RData")
out4 <- outlyx(MLset)
out4 <- rownames(out4)[out4$outliers == T]
save(out4, file = "RData/806-outliers4.RData")
# out5 <- locfdr_LMM(betas(MLset), pData(MLset)) # does not work for some reason???
# save(out5, file = "RData/806-outliers5.RData")

load("RData/04-outliers.RData")
```

```{r}
load("RData/806-outliers2.RData")
load("RData/806-outliers3.RData")
load("RData/806-outliers4.RData")

pd <- pData(PedRef.B)

out2 <- c(names(out2_450k[[1]]), names(out2_450k[[2]]), names(out2_EPIC[[1]]), names(out2_EPIC[[2]]))
allout <- c(out2, names(out3), out4) %>% table(.) %>% sort(.) %>% names(.)
pd$out <- 0
pd[allout, "out"] <- 1
pData(PedRef.B) <- pd
# save(allbt, allpd, PedRef.B, file = "RData/805-All_GEO_reference_buccal.RData") # update PedRef.B with outlier information

BetaPlot(PedRef.B, "out")
```

### Imputation

```{r}
MLset.F <- PedRef.B[, !sampleNames(PedRef.B) %in% allout]
bt <- betas(MLset.F)
pd <- pData(MLset.F)
bt_imputed <- impute.knn(bt)
betas(MLset.F) <- bt_imputed$data
# save(MLset.F, file = "RData/807-Imputed_buccal.RData")
```


### Add more meta info before adjusting value

#### Sex prediction

```{r}
load("RData/807-Imputed_buccal.RData")
source(paste0(home, "Functions/SexAgePrediction.R"))

if (is.null(fData(MLset.F)$CHR)) {
    if (dim(MLset)[[1]] > 600000) {
        load("~/Reference/EPIC_fdat.RData")
        fdat <- merge(fData(MLset.F), fData_EPIC, by = 0, sort = F)
    } else if (dim(MLset)[[1]] > 300000) {
        load("~/Reference/HM450_fdat.RData")
        fdat <- merge(fData(MLset.F), fData_450, by = 0, sort = F)
    }
    rownames(fdat) <- fdat$Row.names
    fData(MLset) <- fdat[rownames(MLset.F), -1]
}
(prediction <- SexPred(MLset.F)) 
pd$Estimated_Sex_LMM <- prediction

prediction2 <- whatsex(as.data.frame(t(bt)))
pd$Estimated_Sex_WS <- prediction2

test <- pd[pd$Sex != pd$Estimated_Sex, ]
test2 <- pd[pd$Sex != pd$Estimated_Sex_LMM, ]
test3 <- pd[unique(c(rownames(test), rownames(test2))), ]
out1 <- rownames(test3)
out2 <- c(rownames(test), rownames(test2)) %>% .[duplicated(.)] # 168 is estimated to have the wrong sex using both methods

MLset.F <- MLset.F[, !sampleNames(MLset.F) %in% out2]

BetaPlot(MLset.F[fData(MLset.F)$CHR == "X", out2], "Sex")
BetaPlot(MLset.F[fData(MLset.F)$CHR == "Y", out2], "Sex") # intersex?

save(MLset.F, file = "RData/807-Imputed_buccal.RData")
```

#### DNAm Age

```{r}
# DNAm Age
bt <- betas(MLset.F) %>% as.data.frame()
bt <- cbind(CpG = rownames(bt), bt)
cpgs.missing <- checkClocks(bt)
DNAmage <- DNAmAge(bt, clocks = "PedBE") %>% as.data.frame()
identical(DNAmage$id, colnames(MLset.F))
rownames(DNAmage) <- DNAmage$id
pd <- cbind(pData(MLset.F), DNAmage[colnames(MLset.F), "PedBE"])
pData(MLset.F) <- pd

save(MLset.F, file = "RData/808-Buccal_DNAm_Age.RData")
```

#### Subset to only test samples - ASXL1 vs Healthy Controls

```{r}
load("RData/808-Buccal_DNAm_Age.RData")

beta <- betas(MLset.F)
meta <- pData(MLset.F)
meta$Disease_Status[meta$Disease_Status == "Control"] <- "Healthy Control"
meta$Sentrix_ID[meta$Cohort == "GSE147058"] <- NA
meta$Sentrix_Position[meta$Cohort == "GSE147058"] <- NA
meta$Sentrix_Position <- as.character(meta$Sentrix_Position)
meta$Sentrix_Row <- strsplit2(as.character(meta$Sentrix_Position), "C")[, 1] %>% gsub("R", "", .) # GSE104812 does not have Sentrix info
meta$Sentrix_Row <- as.numeric(meta$Sentrix_Row)
table(meta$Sentrix_Row)
levels(meta$Sentrix_Row) <- c(1,2,3,4,5,6,7,8)

beta.F <- beta[, meta$Disease_Status %in% c("ASXL1 variant", "Healthy Control")]
meta <- meta[colnames(beta.F), ]
save(beta.F, meta, file = "RData/808-Buccal_DNAm_Age_test.RData")
```

#### Cell type prediction

```{r}
load("RData/808-Buccal_DNAm_Age_test.RData")

# hepidish
data(centEpiFibIC.m)
data(centBloodSub.m)
frac.m <- hepidish(beta.m = beta.F, ref1.m = centEpiFibIC.m, ref2.m = centBloodSub.m, h.CT.idx = 3, method = 'RPC')
meta <- cbind(meta, frac.m)

save(beta.F, meta, file = "RData/808-Buccal_DNAm_Age_test.RData")

# Plot
plt <- meta[meta$Disease_Status %in% c("Healthy Control", "ASXL1 variant"), ]
BloodPlot(mt = plt, 
          vartype = "categorical", 
          varname = "Disease_Status", 
          tissue = "buccal", 
          ct = c("Epi", "Fib", "B", "CD4T", "CD8T", "Mono", "NK", "Neutro", "Eosino")) #+ scale_fill_manual(values = group.colors) 
```

#### Making cell type PCs

```{r}
library(robCompositions)

ct <- meta[, c("Epi", "Fib", "B", "CD4T", "CD8T", "Mono", "NK", "Neutro", "Eosino")]
sum(ct==0) #Are there any proportions predicted to be 0?
ct <- (ct + 0.0001) #If necessary, add the offset of 0.001
min(ct) # Cannot be = or < 0

pca_object <- pcaCoDa(ct)
CTP <- as.data.frame(pca_object$scores)
colnames(CTP) <- c("CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5", "CTP_PC6", "CTP_PC7", "CTP_PC8")
rownames(CTP) <- rownames(ct)
identical(rownames(meta), rownames(CTP))
meta <- cbind(meta, CTP)
pca_object$eigenvalues / sum(pca_object$eigenvalues) # top2 PCs account for > 90% of variance

save(beta.F, meta, file = "RData/808-Buccal_DNAm_Age_test.RData")
```

### Correct for cell type

```{r}
load("RData/808-Buccal_DNAm_Age_test.RData")

beta <- beta.F
residuals <- t(apply(beta, 1, function(x){
    x <- as.numeric(x)
    residuals(summary(lm(x ~ CTP_PC1 + CTP_PC2, data = meta)))
}))
colnames(residuals) <- colnames(beta)
adj.residuals <- residuals + matrix(apply(beta, 1, mean), nrow = nrow(residuals), ncol = ncol(residuals))
adj.residuals[adj.residuals <= 0] <- 0.0001 # convert any values that are less than or equal to zero to 0.0001
adj.residuals[adj.residuals > 1] <- 0.9999

beta.CC <- adj.residuals
save(beta.CC, meta, file = "RData/810-Buccal_celltype_corrected.RData")
```

### Batch correction

#### SVA

```{r}
load("RData/810-Buccal_celltype_corrected.RData")

beta <- beta.CC
dat <- as.matrix(beta2m(beta))

# Use SVA to remove other batch effect (SentrixID and row missing from some cohorts)
mod <- model.matrix(~ as.factor(Disease_Status), data = meta)
mod0 <- model.matrix(~1, data = meta)
(n.sv <- num.sv(dat, mod, method = "leek"))
#n.sv <- 2
svobj <- sva(dat, mod, mod0, n.sv = n.sv)
# save(svobj, file = "RData/810-Buccal_SVA.RData")

sv <- as.data.frame(svobj$sv)
colnames(sv) <- c("sv1", "sv2", "sv3")
meta.CC <- cbind(meta, sv) 

save(beta.CC, meta.CC, file = "RData/810-Buccal_celltype_corrected.RData")
```

#### ComBat

```{r}
load("RData/810-Buccal_celltype_corrected.RData")

beta <- beta.CC
# MLset.BC <- MLset.CC[, !is.na(meta$Sentrix_Row)]
keep <- meta$Sentrix_ID %in% names(table(meta$Sentrix_ID))[table(meta$Sentrix_ID) != 1] & meta$Sentrix_Row %in% names(table(meta$Sentrix_Row))[table(meta$Sentrix_Row) != 0] 
beta <- beta[, keep]
meta <- meta[colnames(beta), ]

dat <- as.matrix(beta2m(beta))
# Subset out only blood & buccal and healthy + mut sample to make batch correction better
# MLset.C <- MLset.C[, sampleNames(MLset.C) %in% rownames(meta[meta$Tissue %in% c("BEC", "PBMC", "WB"), ])]
# MLset.C <- MLset.C[, sampleNames(MLset.C) %in% rownames(meta[meta$Disease_Status %in% c("Healthy", "Crohn's disease", "FASD", "ASXL1"), ])]
# meta <- pData(MLset.C)
# dim(MLset.C)
# Features  Samples 
#   405686      730 

#Combat
# test <- apply(beta, 1, var) # Some value is exactly 0 - problem when changed to Mval
beta[beta == 0] <- 1e-5
# test2 <- apply(dat, 1, var)
# test3 <- lapply(unique(meta$Platform_ID), function(x){
#     out <- apply(dat[, meta$Sentrix_Row == x], 1, var)
#     return(out)
# })

#meta <- meta[meta$Age <=  ]
#sum(apply(beta, 1, var) == 0) # 0
#mod <- model.matrix(~ as.factor(Cell_Type), data = meta)
table(meta$Platform_ID)
M_c1 <- ComBat(dat, batch = meta$Platform_ID, mod = NULL, par.prior = T)
table(meta$Sentrix_ID)
M_c2 <- ComBat(M_c1, batch = meta$Sentrix_ID, mod = NULL, par.prior = T)
table(meta$Sentrix_Row)
M_c3 <- ComBat(M_c2, batch = meta$Sentrix_Row, mod = NULL, par.prior = T)

beta.BC <- m2beta(M_c3)
meta.BC <- meta
save(beta.BC, meta.BC, file = "RData/811-Buccal_batch_corrected.RData")
```

### Plot PCs and DNAm age and other stuff

```{r}
load("RData/810-Buccal_celltype_corrected.RData")

### Plot DNAm Age
plt <- meta.CC[meta$Age >= 6 & meta$Age <= 18]
ggplot(plt, aes(Age, PedBE)) + geom_smooth(method = "lm") + geom_point(aes(color = Disease_Status)) + theme_classic()

# SamplePlots(MLset.F, hmvars = meta[, c("Age", "Cohort", "Platform_ID", "Tissue", "Disease_Status")], hmcor = T, mds = F, hclust = F, SNP = F, hmname = F)

### PCA Plot
source("../../Functions/heatscreesimple.R")

PCA.CC <- prcomp(beta.CC)
meta_cat <- meta.CC[,  c("Cohort", "Platform_ID", "Sentrix_ID", "Sex", "Disease_Status")] 
# colnames(meta_cat) <- c("Predicted_Sex", "Sample_Group")
meta_con <- meta.CC[, c("Sentrix_Row", "Age", 
                     "sv1", "sv2", "sv3", 
                     "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", 
                     "CTP_PC5", "CTP_PC6", "CTP_PC7", "CTP_PC8")] 
heatscreesimple(PCA.CC$rotation, PCA.CC$sdev^2, meta_cat, meta_con, 10) 

load("RData/811-Buccal_batch_corrected.RData")
PCA.BC <- prcomp(beta.BC)
meta_cat <- meta.BC[,  c("Cohort", "Platform_ID", "Sentrix_ID", "Sex", "Disease_Status")] 
# colnames(meta_cat) <- c("Predicted_Sex", "Sample_Group")
meta_con <- meta.BC[, c("Sentrix_Row", "Age", 
                     "sv1", "sv2", "sv3", 
                     "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", 
                     "CTP_PC5", "CTP_PC6", "CTP_PC7", "CTP_PC8")] 
heatscreesimple(PCA.BC$rotation, PCA.BC$sdev^2, meta_cat, meta_con, 10) 
```

### Check for rep correlation

```{r}
source(paste0(home, "Functions/RepCorTest.R"))
load("RData/808-Buccal_DNAm_Age.RData")
load("RData/810-Buccal_celltype_corrected.RData")
load("RData/811-Buccal_batch_corrected.RData")

reps <- c("GSM3530144_5808922041_R03C02", "GSM3530145_5808922041_R04C02")
Rnorm <- betas(MLset.F)[, sampleNames(MLset.F) %in% reps] # normalized, outlier removed
Rcc <- beta.CC[, reps]  # celltype corrected
Rbc <- beta.BC[, reps]  # batch corrected

# Calculate correlation between tech reps after each pre-processing step
Cnorm <- cor(na.omit(Rnorm))[1,2]
Ccc <- cor(na.omit(Rcc))[1,2]
Cbc <- cor(na.omit(Rbc))[1,2]
Cval <- c(Cnorm, Ccc, Cbc)
Cname <- c(1:length(Cval)) %>% as.character(.)
Cdf <- data.frame(Cname, Cval)
    
p1 <- ggplot(Cdf, aes(Cname, Cval)) +
    geom_point(color = "#005C66", alpha = 0.75, size = 3.5) +
    xlab("Pre-Processing and Normalization Step") +
    ylab("Spearman's Correlation (rho)") +
    scale_x_discrete(breaks = c(1:length(Cval)), 
                     labels = c("Normalized", "Celltype Corrected", "Batch Corrected")) +
    theme_bw(base_size = 10)

# Calculate root mean square error between tech reps after each pre-processing step    
Enorm <- rmse(na.omit(Rnorm)[,1], na.omit(Rnorm)[,2])
Ecc <- rmse(na.omit(Rcc)[,1], na.omit(Rcc)[,2])
Ebc <- rmse(na.omit(Rbc)[,1], na.omit(Rbc)[,2])
Eval <- c(Enorm, Ecc, Ebc)
Ename <- c(1:length(Eval)) %>% as.character(.)
Edf <- data.frame(Ename, Eval)
    
p2 <- ggplot(Edf, aes(Ename, Eval)) +
        geom_point(color = "#733556", alpha = 0.75, size = 3.5) +
        xlab("Pre-Processing and Normalization Step") +
        ylab("Root Mean Square Error") +
        scale_x_discrete(breaks = c(1:length(Eval)), 
                         labels = c("Normalized", "Celltype Corrected", "Batch Corrected")) +
        theme_bw(base_size = 10)

grid.arrange(p1, p2)
```

## Test


### Limma

```{r}
load("RData/811-Buccal_batch_corrected.RData")
load("../../Reference/EPIC_fdat_Mreg.RData")

# Check Buccal patient vs HC - Limma
beta.t <- beta.BC
meta.t <- meta[colnames(beta.t), ]
meta.t$Disease_Status <- factor(meta.t$Disease_Status, levels = c("Healthy Control", "ASXL1 variant"))

design <- model.matrix(~Disease_Status + Age + Sex, data = meta.t)
fit <- lmFit(beta.t, design)
fit <- eBayes(fit)
limmaOut <- topTable(fit, coef = 2, adjust = "BH", number = nrow(beta.t))
limmaOut.bu <- merge(limmaOut, 
                     fdat_EPIC_Mreg[, c("CHR", "MAPINFO", "Mreg", "Mreg_plt", 
                                        "UCSC_REFGENE_NAME", "GENCODECOMPV12_NAME")], 
                     by = 0) %>%
    .[order(.$P.Value, decreasing = F), ]
rownames(limmaOut.bu) <- as.character(limmaOut.bu$Row.names)
dim(limmaOut.bu[limmaOut.bu$adj.P.Val < 0.2, ]) # 172  
dim(limmaOut.bu_down <- limmaOut.bu[limmaOut.bu$logFC < 0 & limmaOut.bu$adj.P.Val < 0.2, ]) #0  
dim(limmaOut.bu_up <- limmaOut.bu[limmaOut.bu$logFC > 0 & limmaOut.bu$adj.P.Val < 0.2, ]) #35    

# Save results
save(limmaOut.bu, file = "RData/812-Buccal_Limma_test.RData")
```

#### Plotting Limma's hits

##### Heatmap

```{r}
beta <- beta.BC
meta <- meta.BC

hc_bu <- sample(rownames(meta)[meta$Disease_Status == "Healthy Control"], 30)
p_bu <- rownames(meta)[meta$Disease_Status == "ASXL1 variant"]
plt <- beta[rownames(limmaOut.bu)[limmaOut.bu$adj.P.Val < 0.2], c(hc_bu, p_bu)]
anno <- meta[colnames(plt), c("Disease_Status", "Sex", "Cohort")]

pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         show_colnames = F,
         scale = "row", 
         treeheight_row = 0, 
         cluster_cols = F,
         cluster_rows = T, 
         annotation_col = anno)
pheatmap(plt, # cannot plot all 
         show_rownames = F, 
         show_colnames = F,
         scale = "none", 
         treeheight_row = 0, 
         cluster_cols = F,
         cluster_rows = T, 
         annotation_col = anno) # do not show clustering
```

##### Volcano Plot

```{r}
### Volcano plot
source("../../Functions/makeVolcano2.R")
makeVolcano2(name = rownames(limmaOut.bu), 
             pvalue = limmaOut.bu$P.Value, 
             deltabeta = limmaOut.bu$logFC, 
             ymax = 15, 
             alpha = 0.8, 
             palette = 3)
makeVolcano2(name = rownames(limmaOut.bu), 
             deltabeta = limmaOut.bu$logFC, 
             pvalue = limmaOut.bu$P.Value, 
             adjusted = F,
             ymax = 20, 
             alpha = 0.8, 
             palette = 3)
```

#### ChromHMM enrichment on Limma hits

```{r}
library(missMethyl)
library(dplyr)

load("/mnt/scratch/KoborLab/Personal_Folders/mfu/Reference/EPIC_Anno_ChromeHMM.RData") # This is where I saved an annotation file
anno <- readRDS("/mnt/scratch/KoborLab/Personal_Folders/mfu/Reference/EPICv2_fdat.rds") # You can use this for v2 or the one Beryl created or the one in shared coding resource folder for v1
Chrom_df <- EPIC_Anno_ChromeHMM[rownames(beta), c("Name", "E034_TCells_State")] # You want to look through the colnames and pick a cell type that fit with your samples (e.g. PBMC) for the second column!
colnames(Chrom_df) <- c("Probe_ID", "region")
 
# This part creates an annotation that can be put into missmethyl
ChromHMMset <- lapply(unique(Chrom_df$region), function(x){
    cpgs <- Chrom_df$Probe_ID[Chrom_df$region == x]
    id <- getMappedEntrezIDs(cpgs, array.type = "EPIC", anno = anno)
    return(id$sig.eg)
})
names(ChromHMMset) <- unique(Chrom_df$region)
pathChromHMM <- gsameth(sig.cpg = rownames(limmaOut), # Change this to a vector of names of significantly differentially methylated CpGs
                           all.cpg = rownames(beta), # Change this to all the CpGs you are testing
                           collection = ChromHMMset, 
                           array.type = "EPIC", 
                           anno = anno) %>% .[order(.$FDR), ]
pathChromHMM$Term <- rownames(pathChromHMM) %>% 
    gsub("1_TssA", "Active TSS", .) %>%
    gsub("2_TssFlnk", "Flanking TSS", .) %>% 
    gsub("3_TssFlnkU", "Flanking TSS Upstream", .) %>% 
    gsub("4_TssFlnkD", "Flanking TSS Downstream", .) %>% 
    gsub("5_Tx", "Strong Transcription", .) %>% 
    gsub("6_TxWk", "Weak Transcription", .) %>% 
    gsub("7_EnhG1", "Genic Enhancer 1", .) %>% 
    gsub("8_EnhG2", "Genic Enhancer 2", .) %>% 
    gsub("9_EnhA1", "Active Enhancer 1", .) %>% 
    gsub("10_EnhA2", "Active Enhancer 2", .) %>% 
    gsub("11_EnhWk", "Weak Enhancer", .) %>% 
    gsub("12_ZNF/Rpts", "ZNF genes & repeats", .) %>% 
    gsub("13_Het", "Heterochromatin", .) %>% 
    gsub("14_TssBiv", "Bivalent/Poised TSS", .) %>% 
    gsub("15_EnhBiv", "Bivalent Enhancer", .) %>% 
    gsub("16_ReprPC", "Repressed PolyComb", .) %>% 
    gsub("17_ReprPCWk", "Weak Repressed PolyComb", .) %>% 
    gsub("18_Quies", "	Quiescent/Low", .) 
pathChromHMM$log10Padj <- log10(pathChromHMM$FDR)*-1

# This is for plotting
plt <- pathChromHMM
ggplot(plt, aes(reorder(Term, log10Padj), log10Padj)) + 
    geom_bar(stat = "identity") + 
    labs(y = "Adjusted P-value (-log10)", x = "ChromHMM 18 States Annotation") + 
    geom_hline(yintercept = log10(0.01)*-1, linetype = 'dashed', color = "gray") +
    coord_flip() + 
    theme_bw() + 
    theme(text = element_text(size = 13)) 
```

### Outlier detection test

```{r}
load("RData/811-Buccal_batch_corrected.RData")

# Buccal
beta <- beta.BC
meta <- meta.BC[colnames(beta), ]
betas.pat <- beta[, meta$Disease_Status == "ASXL1 variant"]
betas.ref <- beta[, meta$Disease_Status == "Healthy Control"]

betas.refIQR <- apply(betas.ref, 1, function(x) {quantile(x, c(0.025, 0.25, 0.5, 0.75, 0.975))}) %>% t() %>% as.data.frame()
betas.refIQR$iqr <- betas.refIQR[, 4] - betas.refIQR[, 2]
betas.refIQR$betas.refIQRl <- betas.refIQR[, 2] - 1.5*betas.refIQR$iqr
betas.refIQR$betas.refIQRh <- betas.refIQR[, 4] + 1.5*betas.refIQR$iqr
betas.refSD <- apply(betas.ref, 1, sd)
betas.refM <- apply(betas.ref, 1, mean)

ASXL1.test <- cbind(betas.refIQR, betas.refSD, betas.refM, betas.pat) %>% as.data.frame()
ASXL1.test$betas.refSD2l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD2h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD3l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD3h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD4l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*4)
ASXL1.test$betas.refSD4h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*4)

out <- apply(as.matrix(ASXL1.test), 1, function(x){
    IQR.test.bu <- x["betas.pat"] < x["2.5%"] | 
        x["betas.pat"] > x["97.5%"]
    IQR2.test.bu <- x["betas.pat"] < x["betas.refIQRl"] |
        x["betas.pat"] > x["betas.refIQRh"]
    SD2.test.bu <- x["betas.pat"] < x["betas.refSD2l"] |
        x["betas.pat"] > x["betas.refSD2h"]
    SD3.test.bu <- x["betas.pat"] < x["betas.refSD3l"] |
        x["betas.pat"] > x["betas.refSD3h"]
    SD4.test.bu <- x["betas.pat"] < x["betas.refSD4l"] |
        x["betas.pat"] > x["betas.refSD4h"]
    return(cbind(IQR.test.bu, IQR2.test.bu, SD2.test.bu, SD3.test.bu, SD4.test.bu))
})

out <- t(out)
colnames(out) <- c("IQR.test.bu", "IQR2.test.bu", "SD2.test.bu", "SD3.test.bu", "SD4.test.bu")
ASXL1.test.bu <- cbind(ASXL1.test, as.data.frame(out))
ASXL1.test.bu$meandiff <- ASXL1.test.bu$betas.pat - ASXL1.test.bu$betas.refM

save(ASXL1.test.bu, file = "RData/813-global_test_buccal.RData")

sum(ASXL1.test.bu$IQR.test.bu) # 4189
sum(ASXL1.test.bu$IQR2.test.bu) # 687
sum(ASXL1.test.bu$SD2.test.bu) # 7003
sum(ASXL1.test.bu$SD3.test.bu) # 56
sum(ASXL1.test.bu$SD4.test.bu) # 6

sum(ASXL1.test.bu$IQR.test.bu & abs(ASXL1.test.bu$meandiff) > 0.05) # 1792
sum(ASXL1.test.bu$IQR2.test.bu & abs(ASXL1.test.bu$meandiff) > 0.05) # 337
sum(ASXL1.test.bu$SD2.test.bu & abs(ASXL1.test.bu$meandiff) > 0.05) # 2740
sum(ASXL1.test.bu$SD3.test.bu & abs(ASXL1.test.bu$meandiff) > 0.05) # 34
sum(ASXL1.test.bu$SD4.test.bu & abs(ASXL1.test.bu$meandiff) > 0.05) # 6
```

#### Outlier detection test (buccal) - But with shuffled labels to check test sensitivity / precision

```{r}
load("RData/811-Buccal_batch_corrected.RData")

# Buccal
beta <- beta.BC
meta <- meta.BC[colnames(beta), ]
betas.pat <- beta[, "203077630141_R02C01"]
betas.ref <- beta[, meta$Disease_Status == "Healthy Control" & colnames(beta) != "203077630141_R02C01"]

betas.refIQR <- apply(betas.ref, 1, function(x) {quantile(x, c(0.025, 0.25, 0.5, 0.75, 0.975))}) %>% t() %>% as.data.frame()
betas.refIQR$iqr <- betas.refIQR[, 4] - betas.refIQR[, 2]
betas.refIQR$betas.refIQRl <- betas.refIQR[, 2] - 1.5*betas.refIQR$iqr
betas.refIQR$betas.refIQRh <- betas.refIQR[, 4] + 1.5*betas.refIQR$iqr
betas.refSD <- apply(betas.ref, 1, sd)
betas.refM <- apply(betas.ref, 1, mean)

ASXL1.test <- cbind(betas.refIQR, betas.refSD, betas.refM, betas.pat) %>% as.data.frame()
ASXL1.test$betas.refSD2l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD2h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*2)
ASXL1.test$betas.refSD3l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD3h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*3)
ASXL1.test$betas.refSD4l <- ASXL1.test$betas.refM - (ASXL1.test$betas.refSD*4)
ASXL1.test$betas.refSD4h <- ASXL1.test$betas.refM + (ASXL1.test$betas.refSD*4)

out <- apply(as.matrix(ASXL1.test), 1, function(x){
    IQR.test.bu <- x["betas.pat"] < x["2.5%"] | 
        x["betas.pat"] > x["97.5%"]
    IQR2.test.bu <- x["betas.pat"] < x["betas.refIQRl"] |
        x["betas.pat"] > x["betas.refIQRh"]
    SD2.test.bu <- x["betas.pat"] < x["betas.refSD2l"] |
        x["betas.pat"] > x["betas.refSD2h"]
    SD3.test.bu <- x["betas.pat"] < x["betas.refSD3l"] |
        x["betas.pat"] > x["betas.refSD3h"]
    SD4.test.bu <- x["betas.pat"] < x["betas.refSD4l"] |
        x["betas.pat"] > x["betas.refSD4h"]
    return(cbind(IQR.test.bu, IQR2.test.bu, SD2.test.bu, SD3.test.bu, SD4.test.bu))
})

out <- t(out)
colnames(out) <- c("IQR.test.bu", "IQR2.test.bu", "SD2.test.bu", "SD3.test.bu", "SD4.test.bu")
ASXL1.test.bu_ctr <- cbind(ASXL1.test, as.data.frame(out))
ASXL1.test.bu_ctr$meandiff <- ASXL1.test.bu_ctr$betas.pat - ASXL1.test.bu_ctr$betas.refM

save(ASXL1.test.bu_ctr, file = "RData/813-global_test_buccal_ctr.RData")

sum(ASXL1.test.bu_ctr$IQR.test.bu & abs(ASXL1.test.bu_ctr$meandiff) > 0.05) # 1583 & 435 (w/o and w/ effect size cutoff)
sum(ASXL1.test.bu_ctr$IQR2.test.bu & abs(ASXL1.test.bu_ctr$meandiff) > 0.05) # 130 & 98
sum(ASXL1.test.bu_ctr$SD2.test.bu & abs(ASXL1.test.bu_ctr$meandiff) > 0.05) # 818 & 335 (1.4%)
sum(ASXL1.test.bu_ctr$SD3.test.bu & abs(ASXL1.test.bu_ctr$meandiff) > 0.05) # 7 & 6 (0.3%)
sum(ASXL1.test.bu_ctr$SD4.test.bu & abs(ASXL1.test.bu_ctr$meandiff) > 0.05) # 0 & 0
```



### Permutation Test

```{r}
load("RData/811-Buccal_batch_corrected.RData")
beta <- beta.BC
meta <- meta.BC[colnames(beta), ]

# Set progress bar
set.seed(1234)

# Sign test
permTest <- function(cpg, group, group_var, n) {
    g <- unique(group)
    m0 <- mean(cpg[group == group_var]) # find the average DNAm level for group_var (in this case patient sample)
    m1 <- sapply(n, function(x) {
        ngroup <- sample(group, length(group)) # random shuffling of labels
        mean(cpg[ngroup == group_var]) # find the average DNAm level for randomly labelled samples
    })
    out <- c(sum(m0>=m1), sum(m0<=m1))
    return(out)
}

N <- 2000
cl <- 10

# Buccal - sign test
beta.t <- beta
meta.t <- meta[colnames(beta.t), ]
permOut.bu <- pbapply(beta.t, 1, function(x) permTest(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) %>% t() %>% as.data.frame()
permOut.bu <- permOut.bu/N
colnames(permOut.bu) <- c("Hypo_pval", "Hyper_pval")
permOut.bu$Hypo_qval <- p.adjust(permOut.bu$Hypo_pval)
permOut.bu$Hyper_qval <- p.adjust(permOut.bu$Hyper_pval)
permOut.bu$DeltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])


# Modified rank sum test
permTest2 <- function(cpg, group, group_var, n) {
    g <- unique(group)
    ord0 <- mean(order(cpg)[group == group_var]) # calculate the rank of patient sample's beta value for a given CpG (compared to HCs)
    ord0_d <- (length(group) / 2 - ord0) %>% abs() # calculate the distance from the median rank (the larger the more extreme the value is)
    ord1_d <- sapply(n, function(x) {
        ngroup <- sample(group, length(group)) # random shuffling of labels
        ord1 <- mean(order(cpg)[ngroup %in% group_var])
        ord1_d <- (length(group) / 2 - ord1) %>% abs()
        return(ord1_d)
    })
    out <- sum(ord1_d >= ord0_d) # if the shuffled (permuted) rank is rarely more extreme than the actual rank (ord0), than the patient is very different from HC
    return(out)
} 

# Whole blood - modified rank sum test
beta.t <- beta
meta.t <- meta[colnames(beta.t), ]
permOut.bu2 <- pbapply(beta.t, 1, function(x) permTest2(x, meta.t$Disease_Status, "ASXL1 variant", 1:N), cl = cl) 
deltaB <- beta.t[, meta.t$Disease_Status == "ASXL1 variant"] - rowMeans(beta.t[, meta.t$Disease_Status != "ASXL1 variant"])
permOut.bu2 <- cbind(deltaB = deltaB, pval = permOut.bu2/N) %>% as.data.frame()
permOut.bu2$qval <- p.adjust(permOut.bu2$pval)

save(permOut.bu, permOut.bu2,
     file = "RData/814-Buccal_permutation_test.RData")



# Check overlap
cutoff <- 0.05 # Likelihood for patient sample to get
length(rownames(permOut.bu)[permOut.bu$Hypo_qval < cutoff]) # 1
length(rownames(permOut.bu)[permOut.bu$Hyper_qval < cutoff]) # 17
length(rownames(permOut.bu2)[permOut.bu2$qval < cutoff]) # 11

length(rownames(permOut.bu)[permOut.bu$Hypo_pval < cutoff & abs(permOut.bu$DeltaB) > 0.05]) # 3104
length(rownames(permOut.bu)[permOut.bu$Hyper_pval < cutoff & abs(permOut.bu$DeltaB) > 0.05]) # 4926
length(rownames(permOut.bu2)[permOut.bu2$pval < cutoff & abs(permOut.bu2$deltaB) > 0.05]) # 355


intersect(rownames(permOut.bu2)[permOut.bu2$qval < cutoff], 
          c(rownames(permOut.bu)[permOut.bu$Hypo_qval < cutoff], 
            rownames(permOut.bu)[permOut.bu$Hyper_qval < cutoff])) # 5
```

#### Plotting

```{r}
beta <- beta.BC
meta <- meta.BC
cutoff <- 0.2

# PermTest
samp <- c(sort(sample(1:(ncol(beta)-1), 50)), ncol(beta)) 
plt.hyper <- beta[rownames(permOut.bu)[permOut.bu$Hyper_qval < cutoff & abs(permOut.bu$DeltaB) > 0.05], samp]
plt.hypo <- beta[rownames(permOut.bu)[permOut.bu$Hypo_qval < cutoff & abs(permOut.bu$DeltaB) > 0.05], samp]
plt <- rbind(plt.hyper, plt.hypo)
anno <- meta[colnames(plt), c("Disease_Status", "Sex", "Age", "Cohort")]

pheatmap(plt, 
         show_rownames = F, 
         show_colnames = F,
         scale = "none", 
         treeheight_row = 0, 
         cluster_cols = T,
         cluster_rows = T, 
         annotation_col = anno) # do not show clustering

# PermTest2
samp <- c(sort(sample(1:(ncol(beta)-1), 50)), ncol(beta)) 
plt <- beta[rownames(permOut.bu2)[permOut.bu2$qval < cutoff & abs(permOut.bu2$deltaB) > 0.05], samp]
anno <- meta[colnames(plt), c("Disease_Status", "Sex", "Cohort")]
pheatmap(plt, 
         show_rownames = F, 
         show_colnames = F,
         scale = "none", 
         treeheight_row = 0, 
         cluster_cols = T,
         cluster_rows = T, 
         annotation_col = anno)
```


```{r}
load("~/Reference/EPIC_fdat_Mreg.RData")
source(paste0(home, "Functions/plotmultigeneregion.R"))
source(paste0(home, "Functions/plotGeneRegion.R"))

fdat <- merge(ASXL1.test, fdat_EPIC_Mreg[, c("CHR", "MAPINFO", "Mreg", "UCSC_REFGENE_NAME", "GENCODECOMPV12_NAME")], by = 0)
rownames(fdat) <- fdat$Row.names

# Take the biggest DMRs
#ASXL1.test.dmr <- table(fdat[fdat$SD.test2 == "s", "Mreg"]) %>% sort(decreasing = T) %>% .[.>=1] %>% names()
ASXL1.test.dmr <- table(fdat[test, "Mreg"]) %>% sort(decreasing = T) %>% .[.>=3] %>% names()
ASXL1.test.dmr.gene <- sapply(ASXL1.test.dmr, function(x) {
    return(fdat[fdat$Mreg == x, "UCSC_REFGENE_NAME"] %>% 
               strsplit2(., ";") %>%
               as.vector() %>%
               unique())
}) 
dmr.genes <- do.call(c, ASXL1.test.dmr.gene)
toString(shQuote(dmr.genes))
title <- c('MICALCL', '', 'CLEC9A', 'BAIAP2', 'MAST3', 'LOC401127')
ASXL1.test.dmr.genefdat <- fdat[fdat$Mreg %in% ASXL1.test.dmr, ]

mcolor <- c("#fc8d62", "light gray")
malpha <- c(1, 0.3)
plotmultigeneregion(bt = beta, # [, meta$Disease_Status == "Healthy Control" | rownames(meta) == "203077630141_R07C01"] 
                    fdat = ASXL1.test.dmr.genefdat,
                    plist = ASXL1.test.dmr,
                    colorvar = meta$Disease_Status, # [meta$Disease_Status == "Healthy Control" | rownames(meta) == "203077630141_R07C01"]
                    cscale = mcolor, ascale = malpha, 
                    hide.legend = T, byvar = "Mreg", 
                    title = title, nCol = 4)
```

### Check for rep correlation

```{r}
source(paste0(home, "Functions/RepCorTest.R"))
load("RData/04-Imputed.RData")
load("RData/06-Batch_corrected.RData")
load("RData/07-Celltype_corrected.RData")


reps <- c("203077630141_R01C01", "203077630141_R06C01")
Rnorm <- betas(MLset)[, sampleNames(MLset) %in% reps] # original
Rbc <- betas(MLset.C)[, sampleNames(MLset.C) %in% reps] # batch corrected
Rcc <- betas(MLset.CC)[, sampleNames(MLset.CC) %in% reps] # celltype corrected

# Calculate correlation between tech reps after each pre-processing step
Cnorm <- cor(na.omit(Rnorm))[1,2]
Cbc <- cor(na.omit(Rbc))[1,2]
Ccc <- cor(na.omit(Rcc))[1,2]
Cval <- c(Cnorm, Cbc, Ccc)
Cname <- c(1:length(Cval)) %>% as.character(.)
Cdf <- data.frame(Cname, Cval)
    
p1 <- ggplot(Cdf, aes(Cname, Cval)) +
    geom_point(color = "#005C66", alpha = 0.75, size = 3.5) +
    xlab("Pre-Processing and Normalization Step") +
    ylab("Spearman's Correlation (rho)") +
    scale_x_discrete(breaks = c(1:length(Cval)), 
                     labels = c("Normalized", "Batch Corrected", "Celltype Corrected")) +
    theme_bw(base_size = 10)

# Calculate root mean square error between tech reps after each pre-processing step    
Enorm <- rmse(na.omit(Rnorm)[,1], na.omit(Rnorm)[,2])
Ebc <- rmse(na.omit(Rbc)[,1], na.omit(Rbc)[,2])
Ecc <- rmse(na.omit(Rcc)[,1], na.omit(Rcc)[,2])
Eval <- c(Enorm, Ebc, Ecc)
Ename <- c(1:length(Eval)) %>% as.character(.)
Edf <- data.frame(Ename, Eval)
    
p2 <- ggplot(Edf, aes(Ename, Eval)) +
        geom_point(color = "#733556", alpha = 0.75, size = 3.5) +
        xlab("Pre-Processing and Normalization Step") +
        ylab("Root Mean Square Error") +
        scale_x_discrete(breaks = c(1:length(Eval)), 
                         labels = c("Normalized", "Batch Corrected", "Celltype Corrected")) +
        theme_bw(base_size = 10)

grid.arrange(p1, p2)





```


```{r}
plotmultigeneregion <- function(bt, fdat, plist, colorvar, cscale, ascale, hide.legend = F, byvar = c("Mreg", "UCSC_REFGENE_NAME"), title, nCol = NULL) {
    library(gridExtra)
    source(paste0(home, "Functions/plotGeneRegion.R"))
    
    diffregion <- lapply(1:length(plist), function(p) {
        genefdat <- fdat[grep(plist[p], fdat[, byvar]), ]
        if (length(unique(genefdat$CHR)) > 1) {
            chr <- names(sort(table(genefdat$CHR), decreasing = T))[1]
            genefdat <- genefdat[genefdat$CHR == chr, ]
        } 
        return(plotGeneRegion(x = bt, y = colorvar, z = genefdat, tss = 0, title = title[p], cscale, ascale, HL = hide.legend))
    })
    return(grid.arrange(grobs = diffregion, ncol = nCol))
}

```

```{r}
library(missMethyl)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
ann <- getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)


GE.go <- gometh(rownames(ASXL1.test)[ASXL1.test$SD.test2 == "s"],
                rownames(ASXL1.test), "GO", "450K", T, anno = ann)
GE.go <- gometh(c(rownames(permOut)[permOut$Hypo_qval < 0.05], rownames(permOut)[permOut$Hyper_qval < 0.05]),
                rownames(permOut), "GO", "450K", T, anno = ann)
GE.go.top <- GE.go[GE.go$ONTOLOGY == "BP", ] %>% .[order(.$P.DE), ]
head(GE.go.top)

GE.kegg <- gometh(rownames(ASXL1.test)[ASXL1.test$SD.test2 == "s"],
                  rownames(ASXL1.test), "KEGG", "450K", T, anno = ann)
GE.kegg <- gometh(c(rownames(permOut)[permOut$Hypo_qval < 0.05], rownames(permOut)[permOut$Hyper_qval < 0.05]),
                  rownames(permOut), "KEGG", "450K", T, anno = ann)
GE.kegg.top <- GE.kegg[order(GE.kegg$FDR), ]
head(GE.kegg.top)

write.csv(GE.keff, file = "Enrichment_KEGG_IQRtest.csv")
write.csv(GE.keff, file = "Enrichment_KEGG_SDtest.csv")
write.csv(GE.keff, file = "Enrichment_KEGG_SDtest2.csv")
save(topdiff.topgo, topdiff.topkegg, file = "RData/101-topdiff_enrichment.RData")

test <- fdat[rownames(ASXL1.test)[ASXL1.test$SD.test2 == "s"], ]
test2 <- sapply(test$UCSC_REFGENE_NAME, function(x) {
    return(strsplit2(x, ";") %>%
               as.vector() %>%
               unique())
}) %>% do.call(c, .) %>% unique()

test2 <- ASXL1.test.dmr.gene%>% do.call(c, .) %>% unique()
paste0(test2, collapse = " ")

chea <- read.delim("ChEA_2016_table.txt")
chea$log10Padj <- log10(chea$Adjusted.P.value)*-1
ggplot(chea[1:20, ], aes(reorder(Term, log10Padj), log10Padj)) + geom_bar(stat = "identity") + coord_flip() + theme_bw() + labs(y = "Adjusted P-value (-log10)", x = "CHEA Transcription Factor Targets Dataset (2016)")
GE.go.top$log10Padj <- log10(GE.go.top$FDR)*-1
ggplot(GE.go.top[GE.go.top$FDR<0.05, ], aes(reorder(TERM, log10Padj), log10Padj)) + geom_bar(stat = "identity") + coord_flip() + theme_bw() + labs(y = "Adjusted P-value (-log10)", x = "GO Gene Set Enrichment (Biological Processes)")
GE.kegg.top$log10Padj <- log10(GE.kegg.top$FDR)*-1
ggplot(GE.kegg.top[GE.kegg.top$FDR<0.2, ], aes(reorder(Description, log10Padj), log10Padj)) + geom_bar(stat = "identity") + coord_flip() + theme_bw() + labs(y = "Adjusted P-value (-log10)", x = "KEGG Gene Set Enrichment")

```




#### Check TSDR in blood

```{r}
load("~/Personal_Folders/mfu/Projects/ASXL1/RData/EPIC_FN.RData")
load("~/Personal_Folders/mfu/Projects/ASXL1/RData/10-Biggs_patient_raw.RData")
load("~/Personal_Folders/mfu/Reference/EPIC_Annotation_Complete.RData")
load("RData/07-Celltype_corrected.RData")
test2 <- EPIC_Annotation_Complete[grep("FOXP3", EPIC_Annotation_Complete$UCSC_RefGene_Name), ]
test2 <- test2[test2$MAPINFO == 49117189, ] #cg25442707 

colnames(EPIC.n) <- idat_id <- gsub("GSM[0-9]{7}_", "", colnames(EPIC.n))
EPIC.n <- EPIC.n[, colnames(EPIC.n) %in% sampleNames(MLset.CC)]
pdat <- pData(MLset.CC)[colnames(EPIC.n), ]
pdat <- pdat[pdat$Disease_Status %in% c("Healthy Control", "ASXL1 variant"), ]
EPIC.n <- EPIC.n[, rownames(pdat)]

TSDR <- getBeta(EPIC.n) %>%
    .["cg25442707", , drop = F]

TSDR <- TSDR[, pdat$Cell_Type == "PBMC" & pdat$Gran < 0.1, drop = F]
TSDR_pdat <- pdat[colnames(TSDR), "Disease_Status", drop = F]
ggplot(TSDR)
pheatmap::pheatmap(TSDR, cluster_rows = F, cluster_cols = T, show_rownames = T, annotation_col = TSDR_pdat) +
    coord_flip()
```
